<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Case Study: Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis | NULISAseqR User Guide</title>
  <meta name="description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Case Study: Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis | NULISAseqR User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  <meta name="github-repo" content="Alamar-Biosciences/NULISAseqR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Case Study: Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis | NULISAseqR User Guide" />
  
  <meta name="twitter:description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="outcome-modeling.html"/>
<link rel="next" href="additional-resources.html"/>
<script src="libs/jquery-3.6.1/jquery-3.6.1.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.9/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<link rel="shortcut icon" href="figures/favicon.png" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li style="text-align: left; padding: 20px;">
  <a href="https://www.alamarbio.com" target="_blank">
    <img src="figures/alamar_logo.png" style="height:70px;">
  </a>
</li>
<li><a href="./">NULISAseqR User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-is-nulisaseqr"><i class="fa fa-check"></i>What is NULISAseqR?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i>Installation</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#macos-users"><i class="fa fa-check"></i>macOS Users</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#windows-users"><i class="fa fa-check"></i>Windows Users</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installation-steps"><i class="fa fa-check"></i>Installation Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#getting-help"><i class="fa fa-check"></i>Getting Help</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html"><i class="fa fa-check"></i>Understanding NULISAseq Data</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#what-is-nulisa-technology"><i class="fa fa-check"></i>What is NULISA™ Technology?</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#technical-workflow"><i class="fa fa-check"></i>Technical Workflow</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#learn-more-about-nulisa-technology"><i class="fa fa-check"></i>Learn More About NULISA™ Technology</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#available-nulisaseq-panels"><i class="fa fa-check"></i>Available NULISAseq Panels</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#data-normalization"><i class="fa fa-check"></i>Data Normalization</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#raw-data-sequencing-counts"><i class="fa fa-check"></i>Raw Data: Sequencing Counts</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-1-internal-control-ic-normalization"><i class="fa fa-check"></i>Step 1: Internal Control (IC) Normalization</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-2-inter-plate-control-ipc-normalization"><i class="fa fa-check"></i>Step 2: Inter-Plate Control (IPC) Normalization</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-3-log2-transformation"><i class="fa fa-check"></i>Step 3: Log2 Transformation</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#why-log-transform"><i class="fa fa-check"></i>Why Log Transform?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#npq-vs-fold-change"><i class="fa fa-check"></i>NPQ vs Fold Change</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#understanding-the-relationship"><i class="fa fa-check"></i>Understanding the Relationship</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#working-with-npq-values"><i class="fa fa-check"></i>Working with NPQ Values</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#calculating-fold-changes-from-npq"><i class="fa fa-check"></i>Calculating Fold Changes from NPQ</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#calculating-fold-changes-in-target-ratios"><i class="fa fa-check"></i>Calculating Fold Changes in Target Ratios</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#data-analysis-faq"><i class="fa fa-check"></i>Data Analysis FAQ</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-is-the-limit-of-detection-lod-defined-for-nulisaseq-why-do-some-targets-have-lod-of-0"><i class="fa fa-check"></i>1. How is the limit of detection (LOD) defined for NULISAseq? <br> Why do some targets have LOD of 0?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#should-i-exclude-samples-below-lod"><i class="fa fa-check"></i>2. Should I exclude samples below LOD?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-do-i-calculate-coefficient-of-variation-cv-for-samples"><i class="fa fa-check"></i>3. How do I calculate coefficient of variation (CV) for samples?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#can-i-compare-data-across-different-runs"><i class="fa fa-check"></i>4. Can I compare data across different runs?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-do-you-assess-correlation-between-npq-and-absolute-quantification-data-from-other-platforms"><i class="fa fa-check"></i>5. How do you assess correlation between NPQ and absolute quantification data from other platforms?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="data-import.html"><a href="data-import.html"><i class="fa fa-check"></i><b>1</b> Data Import</a>
<ul>
<li class="chapter" data-level="1.1" data-path="data-import.html"><a href="data-import.html#the-importnulisaseq-function"><i class="fa fa-check"></i><b>1.1</b> The <code>importNULISAseq()</code> Function</a></li>
<li class="chapter" data-level="1.2" data-path="data-import.html"><a href="data-import.html#understanding-the-data-structure"><i class="fa fa-check"></i><b>1.2</b> Understanding the Data Structure</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="data-import.html"><a href="data-import.html#key-components"><i class="fa fa-check"></i><b>1.2.1</b> Key Components</a></li>
<li class="chapter" data-level="1.2.2" data-path="data-import.html"><a href="data-import.html#summary-statistics"><i class="fa fa-check"></i><b>1.2.2</b> Summary Statistics</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="data-import.html"><a href="data-import.html#working-with-your-own-data"><i class="fa fa-check"></i><b>1.3</b> Working with Your Own Data</a></li>
<li class="chapter" data-level="1.4" data-path="data-import.html"><a href="data-import.html#loading-metadata"><i class="fa fa-check"></i><b>1.4</b> Loading Metadata</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="data-import.html"><a href="data-import.html#metadata-structure"><i class="fa fa-check"></i><b>1.4.1</b> Metadata Structure</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="data-import.html"><a href="data-import.html#merging-data-with-metadata"><i class="fa fa-check"></i><b>1.5</b> Merging Data with Metadata</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="data-import.html"><a href="data-import.html#understanding-sample-names"><i class="fa fa-check"></i><b>1.5.1</b> Understanding Sample Names</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-import.html"><a href="data-import.html#merging-options"><i class="fa fa-check"></i><b>1.5.2</b> Merging Options</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-import.html"><a href="data-import.html#filtering-samples"><i class="fa fa-check"></i><b>1.6</b> Filtering Samples</a></li>
<li class="chapter" data-level="1.7" data-path="data-import.html"><a href="data-import.html#complete-import-workflow"><i class="fa fa-check"></i><b>1.7</b> Complete Import Workflow</a></li>
<li class="chapter" data-level="1.8" data-path="data-import.html"><a href="data-import.html#tips-and-best-practices"><i class="fa fa-check"></i><b>1.8</b> Tips and Best Practices</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>2</b> Quality Control</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quality-control.html"><a href="quality-control.html#why-quality-control-matters"><i class="fa fa-check"></i><b>2.1</b> Why Quality Control Matters</a></li>
<li class="chapter" data-level="2.2" data-path="quality-control.html"><a href="quality-control.html#qc-report"><i class="fa fa-check"></i><b>2.2</b> Automated QC Report</a></li>
<li class="chapter" data-level="2.3" data-path="quality-control.html"><a href="quality-control.html#qc-report-structure"><i class="fa fa-check"></i><b>2.3</b> QC Report Structure</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="quality-control.html"><a href="quality-control.html#plate-layout"><i class="fa fa-check"></i><b>2.3.1</b> Plate Layout</a></li>
<li class="chapter" data-level="2.3.2" data-path="quality-control.html"><a href="quality-control.html#run-summary"><i class="fa fa-check"></i><b>2.3.2</b> Run Summary</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#quality-control-metrics"><i class="fa fa-check"></i>Quality Control Metrics</a></li>
<li class="chapter" data-level="2.3.3" data-path="quality-control.html"><a href="quality-control.html#platerun-qc"><i class="fa fa-check"></i><b>2.3.3</b> Plate/Run QC</a></li>
<li class="chapter" data-level="2.3.4" data-path="quality-control.html"><a href="quality-control.html#sample-qc"><i class="fa fa-check"></i><b>2.3.4</b> Sample QC</a></li>
<li class="chapter" data-level="2.3.5" data-path="quality-control.html"><a href="quality-control.html#target-qc"><i class="fa fa-check"></i><b>2.3.5</b> Target QC</a></li>
<li class="chapter" data-level="2.3.6" data-path="quality-control.html"><a href="quality-control.html#coefficient-of-variation-cv"><i class="fa fa-check"></i><b>2.3.6</b> Coefficient of Variation (CV)</a></li>
<li class="chapter" data-level="2.3.7" data-path="quality-control.html"><a href="quality-control.html#batch-effect-assessment"><i class="fa fa-check"></i><b>2.3.7</b> Batch Effect Assessment</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="quality-control.html"><a href="quality-control.html#qc-thresholds-summary"><i class="fa fa-check"></i><b>2.4</b> Summary of QC Thresholds</a></li>
<li class="chapter" data-level="2.5" data-path="quality-control.html"><a href="quality-control.html#manual-qc-checks"><i class="fa fa-check"></i><b>2.5</b> Manual QC Checks</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="quality-control.html"><a href="quality-control.html#check-for-missing-data"><i class="fa fa-check"></i><b>2.5.1</b> Check for Missing Data</a></li>
<li class="chapter" data-level="2.5.2" data-path="quality-control.html"><a href="quality-control.html#low-detectability-targets"><i class="fa fa-check"></i><b>2.5.2</b> Low Detectability Targets</a></li>
<li class="chapter" data-level="2.5.3" data-path="quality-control.html"><a href="quality-control.html#calculate-targets-detectability-by-subgroups"><i class="fa fa-check"></i><b>2.5.3</b> Calculate Targets Detectability by Subgroups</a></li>
<li class="chapter" data-level="2.5.4" data-path="quality-control.html"><a href="quality-control.html#batch-effect-correction"><i class="fa fa-check"></i><b>2.5.4</b> Batch Effect Correction</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="quality-control.html"><a href="quality-control.html#qc-best-practices"><i class="fa fa-check"></i><b>2.6</b> Best Practices</a>
<ul>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#documentation"><i class="fa fa-check"></i>Documentation</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#consistency"><i class="fa fa-check"></i>Consistency</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#collaboration"><i class="fa fa-check"></i>Collaboration</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#when-in-doubt"><i class="fa fa-check"></i>When in Doubt</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="quality-control.html"><a href="quality-control.html#complete-qc-workflow"><i class="fa fa-check"></i><b>2.7</b> Complete QC Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>3</b> Visualization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualization.html"><a href="visualization.html#why-visualize"><i class="fa fa-check"></i><b>3.1</b> Why Visualize?</a></li>
<li class="chapter" data-level="3.2" data-path="visualization.html"><a href="visualization.html#heatmaps"><i class="fa fa-check"></i><b>3.2</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="visualization.html"><a href="visualization.html#basic-heatmap"><i class="fa fa-check"></i><b>3.2.1</b> Basic Heatmap</a></li>
<li class="chapter" data-level="3.2.2" data-path="visualization.html"><a href="visualization.html#heatmap-with-annotations"><i class="fa fa-check"></i><b>3.2.2</b> Heatmap with Annotations</a></li>
<li class="chapter" data-level="3.2.3" data-path="visualization.html"><a href="visualization.html#understanding-the-heatmap"><i class="fa fa-check"></i><b>3.2.3</b> Understanding the Heatmap</a></li>
<li class="chapter" data-level="3.2.4" data-path="visualization.html"><a href="visualization.html#interpreting-clusters"><i class="fa fa-check"></i><b>3.2.4</b> Interpreting Clusters</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="visualization.html"><a href="visualization.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.3</b> Principal Component Analysis (PCA)</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="visualization.html"><a href="visualization.html#pca-with-multiple-visual-encodings"><i class="fa fa-check"></i><b>3.3.1</b> PCA with Multiple Visual Encodings</a></li>
<li class="chapter" data-level="3.3.2" data-path="visualization.html"><a href="visualization.html#understanding-pca-plots"><i class="fa fa-check"></i><b>3.3.2</b> Understanding PCA Plots</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="visualization.html"><a href="visualization.html#custom-visualizations-with-ggplot2"><i class="fa fa-check"></i><b>3.4</b> Custom Visualizations with <code>ggplot2</code></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="visualization.html"><a href="visualization.html#npq-distribution---histogram"><i class="fa fa-check"></i><b>3.4.1</b> NPQ Distribution - Histogram</a></li>
<li class="chapter" data-level="3.4.2" data-path="visualization.html"><a href="visualization.html#npq-distribution-by-group---density-plot"><i class="fa fa-check"></i><b>3.4.2</b> NPQ Distribution by Group - Density Plot</a></li>
<li class="chapter" data-level="3.4.3" data-path="visualization.html"><a href="visualization.html#single-protein-across-groups"><i class="fa fa-check"></i><b>3.4.3</b> Single Protein Across Groups</a></li>
<li class="chapter" data-level="3.4.4" data-path="visualization.html"><a href="visualization.html#multiple-proteins-across-groups"><i class="fa fa-check"></i><b>3.4.4</b> Multiple Proteins Across Groups</a></li>
<li class="chapter" data-level="3.4.5" data-path="visualization.html"><a href="visualization.html#protein-protein-correlation"><i class="fa fa-check"></i><b>3.4.5</b> Protein-Protein Correlation</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="visualization.html"><a href="visualization.html#saving-plots"><i class="fa fa-check"></i><b>3.5</b> Saving Plots</a>
<ul>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-heatmap"><i class="fa fa-check"></i>Save Heatmap</a></li>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-pca"><i class="fa fa-check"></i>Save PCA</a></li>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-ggplot-objects"><i class="fa fa-check"></i>Save ggplot Objects</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="visualization.html"><a href="visualization.html#visualization-best-practices"><i class="fa fa-check"></i><b>3.6</b> Visualization Best Practices</a></li>
<li class="chapter" data-level="3.7" data-path="visualization.html"><a href="visualization.html#complete-visualization-workflow"><i class="fa fa-check"></i><b>3.7</b> Complete Visualization Workflow</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>4</b> Differential Abundance Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#overview"><i class="fa fa-check"></i><b>4.1</b> Overview</a></li>
<li class="chapter" data-level="4.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#Ch4"><i class="fa fa-check"></i><b>4.2</b> Differential Abundance</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#group-comparison"><i class="fa fa-check"></i><b>4.2.1</b> Group Comparison</a></li>
<li class="chapter" data-level="4.2.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#understanding-the-model"><i class="fa fa-check"></i><b>4.2.2</b> Understanding the Model</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#model-results"><i class="fa fa-check"></i><b>4.3</b> Model Results</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#key-columns-in-results"><i class="fa fa-check"></i><b>4.3.1</b> Key Columns in Results</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#de-targets"><i class="fa fa-check"></i><b>4.4</b> Finding Significant Proteins</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#filter-by-fdr-threshold"><i class="fa fa-check"></i><b>4.4.1</b> Filter by FDR Threshold</a></li>
<li class="chapter" data-level="4.4.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#higher-vs-lower-abundance"><i class="fa fa-check"></i><b>4.4.2</b> Higher vs Lower Abundance</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#volcano-plots"><i class="fa fa-check"></i><b>4.5</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#inflammatory-disease-vs-normal"><i class="fa fa-check"></i><b>4.5.1</b> Inflammatory Disease vs Normal</a></li>
<li class="chapter" data-level="4.5.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#understanding-volcano-plots"><i class="fa fa-check"></i><b>4.5.2</b> Understanding Volcano Plots</a></li>
<li class="chapter" data-level="4.5.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#cancer-vs-normal"><i class="fa fa-check"></i><b>4.5.3</b> Cancer vs Normal</a></li>
<li class="chapter" data-level="4.5.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#saving-volcano-plots"><i class="fa fa-check"></i><b>4.5.4</b> Saving Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#visualization-of-significant-targets"><i class="fa fa-check"></i><b>4.6</b> Visualization of Significant Targets</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#inflammatory-disease-vs-normal-1"><i class="fa fa-check"></i><b>4.6.1</b> Inflammatory Disease vs Normal</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#multiple-disease-groups"><i class="fa fa-check"></i><b>4.7</b> Multiple Disease Groups</a></li>
<li class="chapter" data-level="4.8" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#model-considerations"><i class="fa fa-check"></i><b>4.8</b> Model Considerations</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#reference-level"><i class="fa fa-check"></i><b>4.8.1</b> Reference Level</a></li>
<li class="chapter" data-level="4.8.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#including-covariates"><i class="fa fa-check"></i><b>4.8.2</b> Including Covariates</a></li>
<li class="chapter" data-level="4.8.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#model-formula-options"><i class="fa fa-check"></i><b>4.8.3</b> Model Formula Options</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#interpreting-results"><i class="fa fa-check"></i><b>4.9</b> Interpreting Results</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#effect-size-coefficient"><i class="fa fa-check"></i><b>4.9.1</b> Effect Size (Coefficient)</a></li>
<li class="chapter" data-level="4.9.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#p-values"><i class="fa fa-check"></i><b>4.9.2</b> P-values</a></li>
<li class="chapter" data-level="4.9.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#significance-thresholds"><i class="fa fa-check"></i><b>4.9.3</b> Significance Thresholds</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#exporting-results"><i class="fa fa-check"></i><b>4.10</b> Exporting Results</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#save-results-table"><i class="fa fa-check"></i><b>4.10.1</b> Save Results Table</a></li>
<li class="chapter" data-level="4.10.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#create-summary-table"><i class="fa fa-check"></i><b>4.10.2</b> Create Summary Table</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#complete-differential-abundance-workflow"><i class="fa fa-check"></i><b>4.11</b> Complete Differential Abundance Workflow</a></li>
<li class="chapter" data-level="4.12" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#best-practices"><i class="fa fa-check"></i><b>4.12</b> Best Practices</a></li>
<li class="chapter" data-level="4.13" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#common-issues"><i class="fa fa-check"></i><b>4.13</b> Common Issues</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html"><i class="fa fa-check"></i><b>5</b> Longitudinal Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#overview-1"><i class="fa fa-check"></i><b>5.1</b> Overview</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#load-and-prepare-data"><i class="fa fa-check"></i><b>5.1.1</b> Load and Prepare Data</a></li>
<li class="chapter" data-level="5.1.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#create-metadata"><i class="fa fa-check"></i><b>5.1.2</b> Create Metadata</a></li>
<li class="chapter" data-level="5.1.3" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#convert-to-wide-format"><i class="fa fa-check"></i><b>5.1.3</b> Convert to Wide Format</a></li>
<li class="chapter" data-level="5.1.4" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#descriptive-statistics"><i class="fa fa-check"></i><b>5.1.4</b> Descriptive Statistics</a></li>
<li class="chapter" data-level="5.1.5" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#exploratory-visualization"><i class="fa fa-check"></i><b>5.1.5</b> Exploratory Visualization</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#linear-mixed-effects-model"><i class="fa fa-check"></i><b>5.2</b> Linear Mixed-Effects Model</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#group-effect-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Group Effect Analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#understanding-the-model-1"><i class="fa fa-check"></i><b>5.2.2</b> Understanding the Model</a></li>
<li class="chapter" data-level="5.2.3" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#model-results-1"><i class="fa fa-check"></i><b>5.2.3</b> Model Results</a></li>
<li class="chapter" data-level="5.2.4" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#likelihood-ratio-test-lrt-statistics"><i class="fa fa-check"></i><b>5.2.4</b> Likelihood Ratio Test (LRT) Statistics</a></li>
<li class="chapter" data-level="5.2.5" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#model-coefficients"><i class="fa fa-check"></i><b>5.2.5</b> Model Coefficients</a></li>
<li class="chapter" data-level="5.2.6" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#overall-group-effect"><i class="fa fa-check"></i><b>5.2.6</b> Overall Group Effect</a></li>
<li class="chapter" data-level="5.2.7" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#specific-time-comparisons"><i class="fa fa-check"></i><b>5.2.7</b> Specific Time Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#volcano-plots-1"><i class="fa fa-check"></i><b>5.3</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#overall-group-effect-lrt"><i class="fa fa-check"></i><b>5.3.1</b> Overall Group Effect (LRT)</a></li>
<li class="chapter" data-level="5.3.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#group-specific-comparisons"><i class="fa fa-check"></i><b>5.3.2</b> Group-Specific Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#trajectory-visualization"><i class="fa fa-check"></i><b>5.4</b> Trajectory Visualization</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#spaghetti-plot-discrete-time-groups"><i class="fa fa-check"></i><b>5.4.1</b> Spaghetti Plot: Discrete Time Groups</a></li>
<li class="chapter" data-level="5.4.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#spaghetti-plot-continuous-time"><i class="fa fa-check"></i><b>5.4.2</b> Spaghetti Plot: Continuous Time</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#model-considerations-1"><i class="fa fa-check"></i><b>5.5</b> Model Considerations</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#random-effects-structure"><i class="fa fa-check"></i><b>5.5.1</b> Random Effects Structure</a></li>
<li class="chapter" data-level="5.5.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#fixed-effects-design"><i class="fa fa-check"></i><b>5.5.2</b> Fixed Effects Design</a></li>
<li class="chapter" data-level="5.5.3" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#likelihood-ratio-test"><i class="fa fa-check"></i><b>5.5.3</b> Likelihood Ratio Test</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#exporting-results-1"><i class="fa fa-check"></i><b>5.6</b> Exporting Results</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#save-results-tables"><i class="fa fa-check"></i><b>5.6.1</b> Save Results Tables</a></li>
<li class="chapter" data-level="5.6.2" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#save-plots"><i class="fa fa-check"></i><b>5.6.2</b> Save Plots</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#complete-longitudinal-analysis-workflow"><i class="fa fa-check"></i><b>5.7</b> Complete Longitudinal Analysis Workflow</a></li>
<li class="chapter" data-level="5.8" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#best-practices-1"><i class="fa fa-check"></i><b>5.8</b> Best Practices</a></li>
<li class="chapter" data-level="5.9" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#common-issues-1"><i class="fa fa-check"></i><b>5.9</b> Common Issues</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="outcome-modeling.html"><a href="outcome-modeling.html"><i class="fa fa-check"></i><b>6</b> Outcome Modeling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="outcome-modeling.html"><a href="outcome-modeling.html#overview-2"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="outcome-modeling.html"><a href="outcome-modeling.html#example-1-exploring-age-associations-continuous-outcome"><i class="fa fa-check"></i><b>6.2</b> Example 1: Exploring Age Associations (Continuous Outcome)</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="outcome-modeling.html"><a href="outcome-modeling.html#why-model-age"><i class="fa fa-check"></i><b>6.2.1</b> Why Model Age?</a></li>
<li class="chapter" data-level="6.2.2" data-path="outcome-modeling.html"><a href="outcome-modeling.html#age-distribution-in-the-dataset"><i class="fa fa-check"></i><b>6.2.2</b> Age Distribution in the Dataset</a></li>
<li class="chapter" data-level="6.2.3" data-path="outcome-modeling.html"><a href="outcome-modeling.html#building-the-age-regression-model"><i class="fa fa-check"></i><b>6.2.3</b> Building the Age Regression Model</a></li>
<li class="chapter" data-level="6.2.4" data-path="outcome-modeling.html"><a href="outcome-modeling.html#understanding-the-age-model"><i class="fa fa-check"></i><b>6.2.4</b> Understanding the Age Model</a></li>
<li class="chapter" data-level="6.2.5" data-path="outcome-modeling.html"><a href="outcome-modeling.html#examining-age-association-results"><i class="fa fa-check"></i><b>6.2.5</b> Examining Age Association Results</a></li>
<li class="chapter" data-level="6.2.6" data-path="outcome-modeling.html"><a href="outcome-modeling.html#finding-age-associated-proteins"><i class="fa fa-check"></i><b>6.2.6</b> Finding Age-Associated Proteins</a></li>
<li class="chapter" data-level="6.2.7" data-path="outcome-modeling.html"><a href="outcome-modeling.html#volcano-plots-2"><i class="fa fa-check"></i><b>6.2.7</b> Volcano Plots</a></li>
<li class="chapter" data-level="6.2.8" data-path="outcome-modeling.html"><a href="outcome-modeling.html#age-associated-proteins-scatterplots"><i class="fa fa-check"></i><b>6.2.8</b> Age-Associated Proteins Scatterplots</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="outcome-modeling.html"><a href="outcome-modeling.html#example-2-modeling-disease-status-binary-outcome"><i class="fa fa-check"></i><b>6.3</b> Example 2: Modeling Disease Status (Binary Outcome)</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="outcome-modeling.html"><a href="outcome-modeling.html#disease-status-as-binary-outcome"><i class="fa fa-check"></i><b>6.3.1</b> Disease Status as Binary Outcome</a></li>
<li class="chapter" data-level="6.3.2" data-path="outcome-modeling.html"><a href="outcome-modeling.html#preparing-the-disease-data"><i class="fa fa-check"></i><b>6.3.2</b> Preparing the Disease Data</a></li>
<li class="chapter" data-level="6.3.3" data-path="outcome-modeling.html"><a href="outcome-modeling.html#building-the-disease-association-model"><i class="fa fa-check"></i><b>6.3.3</b> Building the Disease Association Model</a></li>
<li class="chapter" data-level="6.3.4" data-path="outcome-modeling.html"><a href="outcome-modeling.html#understanding-the-disease-model"><i class="fa fa-check"></i><b>6.3.4</b> Understanding the Disease Model</a></li>
<li class="chapter" data-level="6.3.5" data-path="outcome-modeling.html"><a href="outcome-modeling.html#examining-disease-association-results"><i class="fa fa-check"></i><b>6.3.5</b> Examining Disease Association Results</a></li>
<li class="chapter" data-level="6.3.6" data-path="outcome-modeling.html"><a href="outcome-modeling.html#finding-disease-associated-proteins"><i class="fa fa-check"></i><b>6.3.6</b> Finding Disease-Associated Proteins</a></li>
<li class="chapter" data-level="6.3.7" data-path="outcome-modeling.html"><a href="outcome-modeling.html#interpreting-odds-ratios"><i class="fa fa-check"></i><b>6.3.7</b> Interpreting Odds Ratios</a></li>
<li class="chapter" data-level="6.3.8" data-path="outcome-modeling.html"><a href="outcome-modeling.html#disease-associated-proteins-volcano-plots"><i class="fa fa-check"></i><b>6.3.8</b> Disease-Associated Proteins Volcano Plots</a></li>
<li class="chapter" data-level="6.3.9" data-path="outcome-modeling.html"><a href="outcome-modeling.html#visualizing-disease-associations"><i class="fa fa-check"></i><b>6.3.9</b> Visualizing Disease Associations</a></li>
<li class="chapter" data-level="6.3.10" data-path="outcome-modeling.html"><a href="outcome-modeling.html#roc-curves-for-disease-associated-proteins"><i class="fa fa-check"></i><b>6.3.10</b> ROC Curves for Disease-Associated Proteins</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="outcome-modeling.html"><a href="outcome-modeling.html#comparing-approaches"><i class="fa fa-check"></i><b>6.4</b> Comparing Approaches</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="outcome-modeling.html"><a href="outcome-modeling.html#when-to-use-each-function"><i class="fa fa-check"></i><b>6.4.1</b> When to Use Each Function</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="outcome-modeling.html"><a href="outcome-modeling.html#exporting-results-2"><i class="fa fa-check"></i><b>6.5</b> Exporting Results</a></li>
<li class="chapter" data-level="6.6" data-path="outcome-modeling.html"><a href="outcome-modeling.html#complete-outcome-modeling-workflow"><i class="fa fa-check"></i><b>6.6</b> Complete Outcome Modeling Workflow</a></li>
<li class="chapter" data-level="6.7" data-path="outcome-modeling.html"><a href="outcome-modeling.html#best-practices-2"><i class="fa fa-check"></i><b>6.7</b> Best Practices</a>
<ul>
<li class="chapter" data-level="" data-path="outcome-modeling.html"><a href="outcome-modeling.html#model-design"><i class="fa fa-check"></i>Model Design</a></li>
<li class="chapter" data-level="" data-path="outcome-modeling.html"><a href="outcome-modeling.html#interpretation"><i class="fa fa-check"></i>Interpretation</a></li>
<li class="chapter" data-level="" data-path="outcome-modeling.html"><a href="outcome-modeling.html#validation"><i class="fa fa-check"></i>Validation</a></li>
<li class="chapter" data-level="" data-path="outcome-modeling.html"><a href="outcome-modeling.html#reporting"><i class="fa fa-check"></i>Reporting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="case-study-1.html"><a href="case-study-1.html"><i class="fa fa-check"></i><b>7</b> Case Study: <br> Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="case-study-1.html"><a href="case-study-1.html#overview-3"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="case-study-1.html"><a href="case-study-1.html#data-preparation"><i class="fa fa-check"></i><b>7.2</b> Data Preparation</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="case-study-1.html"><a href="case-study-1.html#load-required-libraries"><i class="fa fa-check"></i><b>7.2.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="7.2.2" data-path="case-study-1.html"><a href="case-study-1.html#import-data"><i class="fa fa-check"></i><b>7.2.2</b> Import Data</a></li>
<li class="chapter" data-level="7.2.3" data-path="case-study-1.html"><a href="case-study-1.html#prepare-sample-metadata"><i class="fa fa-check"></i><b>7.2.3</b> Prepare Sample Metadata</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="case-study-1.html"><a href="case-study-1.html#apply-detectability-threshold"><i class="fa fa-check"></i><b>7.3</b> Apply Detectability Threshold</a></li>
<li class="chapter" data-level="7.4" data-path="case-study-1.html"><a href="case-study-1.html#global-data-visualization"><i class="fa fa-check"></i><b>7.4</b> Global Data Visualization</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="case-study-1.html"><a href="case-study-1.html#prepare-metadata-for-visualization"><i class="fa fa-check"></i><b>7.4.1</b> Prepare Metadata for Visualization</a></li>
<li class="chapter" data-level="7.4.2" data-path="case-study-1.html"><a href="case-study-1.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>7.4.2</b> Hierarchical Clustering Heatmap</a></li>
<li class="chapter" data-level="7.4.3" data-path="case-study-1.html"><a href="case-study-1.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.4.3</b> Principal Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="case-study-1.html"><a href="case-study-1.html#statistical-testing"><i class="fa fa-check"></i><b>7.5</b> Statistical Testing</a></li>
<li class="chapter" data-level="7.6" data-path="case-study-1.html"><a href="case-study-1.html#systemic-lupus-erythematosus-sle-analysis"><i class="fa fa-check"></i><b>7.6</b> Systemic Lupus Erythematosus (SLE) Analysis</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="case-study-1.html"><a href="case-study-1.html#background"><i class="fa fa-check"></i><b>7.6.1</b> Background</a></li>
<li class="chapter" data-level="7.6.2" data-path="case-study-1.html"><a href="case-study-1.html#volcano-plot"><i class="fa fa-check"></i><b>7.6.2</b> Volcano Plot</a></li>
<li class="chapter" data-level="7.6.3" data-path="case-study-1.html"><a href="case-study-1.html#visualizations"><i class="fa fa-check"></i><b>7.6.3</b> Visualizations</a></li>
<li class="chapter" data-level="7.6.4" data-path="case-study-1.html"><a href="case-study-1.html#directional-analysis"><i class="fa fa-check"></i><b>7.6.4</b> Directional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="case-study-1.html"><a href="case-study-1.html#rheumatoid-arthritis-ra-analysis"><i class="fa fa-check"></i><b>7.7</b> Rheumatoid Arthritis (RA) Analysis</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="case-study-1.html"><a href="case-study-1.html#background-1"><i class="fa fa-check"></i><b>7.7.1</b> Background</a></li>
<li class="chapter" data-level="7.7.2" data-path="case-study-1.html"><a href="case-study-1.html#volcano-plot-1"><i class="fa fa-check"></i><b>7.7.2</b> Volcano Plot</a></li>
<li class="chapter" data-level="7.7.3" data-path="case-study-1.html"><a href="case-study-1.html#visualizations-1"><i class="fa fa-check"></i><b>7.7.3</b> Visualizations</a></li>
<li class="chapter" data-level="7.7.4" data-path="case-study-1.html"><a href="case-study-1.html#directional-analysis-1"><i class="fa fa-check"></i><b>7.7.4</b> Directional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="case-study-1.html"><a href="case-study-1.html#comparing-disease-signatures"><i class="fa fa-check"></i><b>7.8</b> Comparing Disease Signatures</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="case-study-1.html"><a href="case-study-1.html#overlap-analysis"><i class="fa fa-check"></i><b>7.8.1</b> Overlap Analysis</a></li>
<li class="chapter" data-level="7.8.2" data-path="case-study-1.html"><a href="case-study-1.html#differential-abundance-summary-table"><i class="fa fa-check"></i><b>7.8.2</b> Differential Abundance Summary Table</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="case-study-1.html"><a href="case-study-1.html#next-steps"><i class="fa fa-check"></i><b>7.9</b> Next Steps</a></li>
<li class="chapter" data-level="7.10" data-path="case-study-1.html"><a href="case-study-1.html#complete-workflow-example-code"><i class="fa fa-check"></i><b>7.10</b> Complete Workflow Example Code</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i>Additional Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#nulisaseq-resources"><i class="fa fa-check"></i>NULISAseq Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#package-documentation"><i class="fa fa-check"></i>Package Documentation</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#github-repository"><i class="fa fa-check"></i>GitHub Repository</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#support-and-issues"><i class="fa fa-check"></i>Support and Issues</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#nulisa-analysis-software-nas"><i class="fa fa-check"></i>NULISA Analysis Software (NAS)</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#technical-documentation"><i class="fa fa-check"></i>Technical Documentation</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#r-resources"><i class="fa fa-check"></i>R Programming Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#r-install"><i class="fa fa-check"></i>Installing R and RStudio</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#tidyverse-resources"><i class="fa fa-check"></i>Tidyverse Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#visualization-resources"><i class="fa fa-check"></i>Data Visualization Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#ggplot2"><i class="fa fa-check"></i>ggplot2</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#complexheatmap"><i class="fa fa-check"></i>ComplexHeatmap</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#pcatools"><i class="fa fa-check"></i>PCAtools</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citations"><i class="fa fa-check"></i>Citing This Book and Packages</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citing-this-book"><i class="fa fa-check"></i>Citing This Book</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citing-r-and-packages"><i class="fa fa-check"></i>Citing R and Packages</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#key-citations"><i class="fa fa-check"></i>Key Citations</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#happy-analyzing"><i class="fa fa-check"></i><strong>Happy analyzing!</strong> 🚀</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/Alamar-Biosciences/NULISAseqR" target="_blank">GitHub Repository</a></li>
<li><a href="../" target="_blank">NULISAseqR Package</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NULISAseqR User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="case-study-1" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Case Study: <br> Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis<a href="case-study-1.html#case-study-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="overview-3" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Overview<a href="case-study-1.html#overview-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Differential abundance analysis is a powerful tool for proteomic studies, enabling researchers to identify proteins that are differentially abundant between experimental conditions. In this chapter, we demonstrate a complete differential abundance workflow using the NULISAseq Inflammation Panel to compare inflammatory diseases against healthy controls.</p>
<p>We’ll analyze a dataset containing samples from patients with:</p>
<ul>
<li><strong>Systemic Lupus Erythematosus (SLE)</strong>: An autoimmune disease where the immune system attacks its own tissues, causing widespread inflammation and tissue damage</li>
<li><strong>Rheumatoid Arthritis (RA)</strong>: A chronic inflammatory disorder primarily affecting joints, but potentially involving other organ systems</li>
<li><strong>Normal controls</strong>: Healthy individuals without inflammatory disease</li>
</ul>
<p>This workflow will cover data import, target detectability filtering, visualization of target abundance, statistical testing, and interpretation of disease-specific protein signatures.</p>
</div>
<div id="data-preparation" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Data Preparation<a href="case-study-1.html#data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="load-required-libraries" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Load Required Libraries<a href="case-study-1.html#load-required-libraries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="case-study-1.html#cb198-1" tabindex="-1"></a><span class="fu">library</span>(NULISAseqR)</span>
<span id="cb198-2"><a href="case-study-1.html#cb198-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="import-data" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Import Data<a href="case-study-1.html#import-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We begin by importing the NULISAseq data file:</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="case-study-1.html#cb199-1" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package =</span> <span class="st">&quot;NULISAseqR&quot;</span>)</span>
<span id="cb199-2"><a href="case-study-1.html#cb199-2" tabindex="-1"></a>data.a <span class="ot">&lt;-</span> <span class="fu">importNULISAseq</span>(<span class="at">files =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&quot;Inflammation_250_RQ_demo_2025-05-20-Lab-A.xml&quot;</span>))</span></code></pre></div>
<pre><code>#&gt; Using single IC to normalize data.</code></pre>
<div id="preview-npq-long-data-frame" class="section level4 unnumbered hasAnchor">
<h4>Preview NPQ Long Data Frame<a href="case-study-1.html#preview-npq-long-data-frame" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>#&gt; Preview of NPQ long data frame (first 20 rows):</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-c50bad2db936bbe8f430" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c50bad2db936bbe8f430">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of NPQ long data frame<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"],["Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)","Inflammation Panel 250 (V2)"],["2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1","2507701b1"],["Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01"],["SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar","SN79-PROD-Alamar"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0"],["Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample"],["AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER","AGER"],["t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521","t5521"],["Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109","Q15109"],["Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor","Advanced glycosylation end product-specific receptor"],[4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869,4.468532337644869],[8860,8558,8390,12424,4050,16163,10778,11773,10793,16539,9683,9268,6506,8503,11733,12371,5923,7052,5966,7945],[12.81411931282402,13.00856640959298,13.02320981020639,13.37223800444577,11.86795086201432,13.98158567746505,13.25136063837208,13.14354337652858,13.11461299448851,13.79822120391,12.80953582666157,13.06069171791404,12.99343425942618,13.07339292305659,13.40858004072018,13.20581533855051,12.26999505848153,12.65056291433583,12.47502981208736,12.81078424100184],["PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS"],[0.134182291231546,-0.0426308942701309,-0.0709042953132273,0.0801417847211448,-0.000954779266968218,-0.07892444115576031,0.018940433708482,0.199381780424638,0.121817899724307,0.0702478845671866,0.243480647817733,0,-0.264509661172708,-0.0905846829536097,-0.00531095967251071,0.207055818782895,0.105646325890033,0.0111231784601797,-0.0338946639773717,0.019405888601129],[0.971887550200803,0.987951807228916,0.975903614457831,0.979919678714859,0.987951807228916,0.987951807228916,0.983935742971888,0.987951807228916,0.971887550200803,0.983935742971888,0.983935742971888,0.979919678714859,0.975903614457831,0.971887550200803,0.987951807228916,0.991967871485944,0.991967871485944,0.967871485943775,0.979919678714859,0.983935742971888],[95032,80217,77848,90504,83709,77176,85376,100495,93996,89675,104190,83789,61626,76199,83344,101138,92641,84721,80949,85415],[2130160,5184950,1741396,5251261,5925301,6102685,10797892,3829676,3121062,2365266,3683309,6136215,1466317,2109026,2313445,7458913,8789443,2438409,2112522,5900236],["PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS"],["PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS"],["PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS"],["PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS","PASS"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0"],["285582","937904","820176","291216","573743","159184","463904","206816","1011424","966240","303972","19576","138200","92648","91535","301999","506473","282846","814518","497302"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0"],["A","A","A","A","A","A","B","B","B","B","B","B","C","C","C","C","C","C","D","D"],[1,2,3,4,9,11,1,2,3,4,9,11,1,2,3,4,9,11,1,2],[2130160,5184950,1741396,5251261,5925301,6102685,10797892,3829676,3121062,2365266,3683309,6136215,1466317,2109026,2313445,7458913,8789443,2438409,2112522,5900236],[1376511,1999744,743965,1551114,1684981,1324182,2309334,1724234,1318930,1129004,2445577,1437749,756722,1438713,1391185,1998259,1606145,1037932,1043419,1494188],["A1","A2","A3","A4","A9","A11","B1","B2","B3","B4","B9","B11","C1","C2","C3","C4","C9","C11","D1","D2"],["PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA"],["normal","lupusSle","normal","lupusSle","rheumatoidArthritis","rheumatoidArthritis","lupusSle","lupusSle","normal","rheumatoidArthritis","rheumatoidArthritis","lupusSle","normal","rheumatoidArthritis","normal","rheumatoidArthritis","lupusSle","normal","normal","lupusSle"],["a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Panel<\/th>\n      <th>PanelLotNumber<\/th>\n      <th>PlateID<\/th>\n      <th>InstrumentID<\/th>\n      <th>SampleName<\/th>\n      <th>SampleType<\/th>\n      <th>Target<\/th>\n      <th>AlamarTargetID<\/th>\n      <th>UniProtID<\/th>\n      <th>ProteinName<\/th>\n      <th>LOD<\/th>\n      <th>UnnormalizedCount<\/th>\n      <th>NPQ<\/th>\n      <th>Sample_QC_Status<\/th>\n      <th>Sample_QC_IC_Median<\/th>\n      <th>Sample_QC_Detectability<\/th>\n      <th>Sample_QC_ICReads<\/th>\n      <th>Sample_QC_NumReads<\/th>\n      <th>Sample_QC_IC_Median_Status<\/th>\n      <th>Sample_QC_Detectability_Status<\/th>\n      <th>Sample_QC_ICReads_Status<\/th>\n      <th>Sample_QC_NumReads_Status<\/th>\n      <th>sampleID<\/th>\n      <th>sampleBarcode<\/th>\n      <th>sampleName_original<\/th>\n      <th>wellRow<\/th>\n      <th>wellCol<\/th>\n      <th>matching<\/th>\n      <th>non-matching<\/th>\n      <th>AUTO_WELLPOSITION<\/th>\n      <th>SAMPLE_MATRIX<\/th>\n      <th>Disease<\/th>\n      <th>Lab<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"scrollY":"600px","pageLength":20,"searching":false,"info":false,"dom":"t","columnDefs":[{"className":"dt-right","targets":[11,12,13,15,16,17,18,27,28,29]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Panel","targets":1},{"name":"PanelLotNumber","targets":2},{"name":"PlateID","targets":3},{"name":"InstrumentID","targets":4},{"name":"SampleName","targets":5},{"name":"SampleType","targets":6},{"name":"Target","targets":7},{"name":"AlamarTargetID","targets":8},{"name":"UniProtID","targets":9},{"name":"ProteinName","targets":10},{"name":"LOD","targets":11},{"name":"UnnormalizedCount","targets":12},{"name":"NPQ","targets":13},{"name":"Sample_QC_Status","targets":14},{"name":"Sample_QC_IC_Median","targets":15},{"name":"Sample_QC_Detectability","targets":16},{"name":"Sample_QC_ICReads","targets":17},{"name":"Sample_QC_NumReads","targets":18},{"name":"Sample_QC_IC_Median_Status","targets":19},{"name":"Sample_QC_Detectability_Status","targets":20},{"name":"Sample_QC_ICReads_Status","targets":21},{"name":"Sample_QC_NumReads_Status","targets":22},{"name":"sampleID","targets":23},{"name":"sampleBarcode","targets":24},{"name":"sampleName_original","targets":25},{"name":"wellRow","targets":26},{"name":"wellCol","targets":27},{"name":"matching","targets":28},{"name":"non-matching","targets":29},{"name":"AUTO_WELLPOSITION","targets":30},{"name":"SAMPLE_MATRIX","targets":31},{"name":"Disease","targets":32},{"name":"Lab","targets":33}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="preview-npq-wide-data-matrix" class="section level4 unnumbered hasAnchor">
<h4>Preview NPQ Wide Data Matrix<a href="case-study-1.html#preview-npq-wide-data-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>#&gt; Preview of NPQ matrix (first 10 proteins × 10 samples, rounded to 3 digits):</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-d0befa470f4923d38454" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d0befa470f4923d38454">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of NPQ matrix<\/caption>","data":[["AGER","AGRP","ANGPT1","ANGPT2","ANXA1","AREG","BDNF","BMP7","BST2","C1QA"],[12.814,11.774,11.129,12.161,7.346,10.009,10.898,11.537,9.895,12.24],[13.009,12.272,12.617,13.685,8.146000000000001,10.181,14.045,11.568,13.997,15.61],[13.023,11.231,10.909,12.373,7.784,9.914,11.939,10.81,10.467,12.503],[13.372,13.035,12.63,14.229,7.671,10.318,13.947,11.774,12.117,13.064],[11.868,12.307,14.062,13.429,7.4,10.999,15.808,11.968,13.602,12.863],[13.982,12.43,13.991,13.332,7.938,10.296,16.886,12.152,11.717,13.411],[13.251,12.492,14.887,13.726,7.752,10.054,17.174,11.42,15.238,14.984],[13.144,11.588,12.053,12.73,17,10.251,13.251,11.309,11.106,13.179],[13.115,11.915,12.302,12.201,7.456,9.978999999999999,13.867,11.075,10.594,12.764],[13.798,11.326,11.574,12.313,7.352,10.356,13.23,11.007,11.224,12.924]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Z1110029999112524P0<\/th>\n      <th>B1120367366051017P0<\/th>\n      <th>Z1110039645091718P0<\/th>\n      <th>C7121093950071918P0<\/th>\n      <th>R5131013361031325P0<\/th>\n      <th>R5131014732032725P0<\/th>\n      <th>R5110038426010621P0<\/th>\n      <th>B1120863696011118P0<\/th>\n      <th>Z1110040042110424P0<\/th>\n      <th>B1121267322021819P0<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"paging":false,"searching":false,"info":false,"dom":"t","columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Z1110029999112524P0","targets":1},{"name":"B1120367366051017P0","targets":2},{"name":"Z1110039645091718P0","targets":3},{"name":"C7121093950071918P0","targets":4},{"name":"R5131013361031325P0","targets":5},{"name":"R5131014732032725P0","targets":6},{"name":"R5110038426010621P0","targets":7},{"name":"B1120863696011118P0","targets":8},{"name":"Z1110040042110424P0","targets":9},{"name":"B1121267322021819P0","targets":10}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="prepare-sample-metadata" class="section level3 hasAnchor" number="7.2.3">
<h3><span class="header-section-number">7.2.3</span> Prepare Sample Metadata<a href="case-study-1.html#prepare-sample-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we extract and prepare the sample metadata, filtering for experimental samples and setting the reference level for the Disease variable:</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="case-study-1.html#cb203-1" tabindex="-1"></a>metadata.a <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>samples <span class="sc">%&gt;%</span> </span>
<span id="cb203-2"><a href="case-study-1.html#cb203-2" tabindex="-1"></a>  <span class="fu">filter</span>(sampleType <span class="sc">==</span> <span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb203-3"><a href="case-study-1.html#cb203-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Disease =</span> <span class="fu">relevel</span>(Disease, <span class="at">ref =</span> <span class="st">&quot;normal&quot;</span>))</span>
<span id="cb203-4"><a href="case-study-1.html#cb203-4" tabindex="-1"></a></span>
<span id="cb203-5"><a href="case-study-1.html#cb203-5" tabindex="-1"></a><span class="co"># Sample counts per group</span></span>
<span id="cb203-6"><a href="case-study-1.html#cb203-6" tabindex="-1"></a>n_normal <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>)</span>
<span id="cb203-7"><a href="case-study-1.html#cb203-7" tabindex="-1"></a>n_sle <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span>)</span>
<span id="cb203-8"><a href="case-study-1.html#cb203-8" tabindex="-1"></a>n_ra <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span>)</span>
<span id="cb203-9"><a href="case-study-1.html#cb203-9" tabindex="-1"></a></span>
<span id="cb203-10"><a href="case-study-1.html#cb203-10" tabindex="-1"></a><span class="co"># Display sample distribution</span></span>
<span id="cb203-11"><a href="case-study-1.html#cb203-11" tabindex="-1"></a><span class="fu">table</span>(metadata.a<span class="sc">$</span>Disease)</span></code></pre></div>
<pre><code>#&gt; 
#&gt;              normal            lupusSle rheumatoidArthritis 
#&gt;                  19                  12                  12</code></pre>
<p>The <code>relevel()</code> function is important here as it sets “normal” as the reference group for all subsequent statistical comparisons. This means our differential abundance results will show changes in diseased samples relative to healthy controls.</p>
<div id="preview-metadata-data-frame" class="section level4 unnumbered hasAnchor">
<h4>Preview Metadata Data Frame<a href="case-study-1.html#preview-metadata-data-frame" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>#&gt; Preview of metadata data frame:</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f3bbde926fb62ed0436f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f3bbde926fb62ed0436f">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of metadata data frame<\/caption>","data":[["Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01"],["Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01","Plate_01"],["Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample","Sample"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0","R5110042658021519P0","C7122018630081921P0","Z1888195532112024P0","Z1888519631120922P0","R5110042659021519P0","C7122725382012225P0","Z1110042744010324P0","R5130850861101022P0","Z1888664437111324P0","Z1888953492111824P0","Z1110042731012324P0","C7123080004020325P0","Z1110044021112524P0","R5130873936051823P0","Z1888902498111824P0","Z1110044037121123P0","R5130900169010824P0","Z1110044949010524P0","R5131009811022625P0","Z1110044790013024P0","R5130995794012325P0","Z1110046960111824P0","R5131011573030525P0"],["285582","937904","820176","291216","573743","159184","463904","206816","1011424","966240","303972","19576","138200","92648","91535","301999","506473","282846","814518","497302","46042","379301","937249","969061","112590","941126","80406","510874","367340","538988","201432","757848","318552","295768","596568","154462","372318","424862","958254","78628","545572","40740","465700"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0","R5110042658021519P0","C7122018630081921P0","Z1888195532112024P0","Z1888519631120922P0","R5110042659021519P0","C7122725382012225P0","Z1110042744010324P0","R5130850861101022P0","Z1888664437111324P0","Z1888953492111824P0","Z1110042731012324P0","C7123080004020325P0","Z1110044021112524P0","R5130873936051823P0","Z1888902498111824P0","Z1110044037121123P0","R5130900169010824P0","Z1110044949010524P0","R5131009811022625P0","Z1110044790013024P0","R5130995794012325P0","Z1110046960111824P0","R5131011573030525P0"],["Z1110029999112524P0","B1120367366051017P0","Z1110039645091718P0","C7121093950071918P0","R5131013361031325P0","R5131014732032725P0","R5110038426010621P0","B1120863696011118P0","Z1110040042110424P0","B1121267322021819P0","R5131013370031725P0","R5200011218091319P0","Z1110040365092118P0","B1121659107051420P0","Z1110040683092018P0","C7121938817050321P0","R5200034416120121P0","Z1888208171110624P0","Z1110040374110624P0","C7121765124092220P0","R5110042658021519P0","C7122018630081921P0","Z1888195532112024P0","Z1888519631120922P0","R5110042659021519P0","C7122725382012225P0","Z1110042744010324P0","R5130850861101022P0","Z1888664437111324P0","Z1888953492111824P0","Z1110042731012324P0","C7123080004020325P0","Z1110044021112524P0","R5130873936051823P0","Z1888902498111824P0","Z1110044037121123P0","R5130900169010824P0","Z1110044949010524P0","R5131009811022625P0","Z1110044790013024P0","R5130995794012325P0","Z1110046960111824P0","R5131011573030525P0"],["A","A","A","A","A","A","B","B","B","B","B","B","C","C","C","C","C","C","D","D","D","D","D","D","E","E","E","E","E","E","F","F","F","F","F","G","G","G","G","H","H","H","H"],[1,2,3,4,9,11,1,2,3,4,9,11,1,2,3,4,9,11,1,2,3,4,9,11,1,2,3,4,9,11,1,2,3,4,9,1,2,3,4,1,2,3,4],[2130160,5184950,1741396,5251261,5925301,6102685,10797892,3829676,3121062,2365266,3683309,6136215,1466317,2109026,2313445,7458913,8789443,2438409,2112522,5900236,2679744,6010787,3115070,2403503,6576379,4581530,4921490,5688631,6424448,2850029,2358357,11559524,2610852,6816765,2014726,1771338,3836357,1817278,9541509,2462032,3994288,2416511,2262662],[1376511,1999744,743965,1551114,1684981,1324182,2309334,1724234,1318930,1129004,2445577,1437749,756722,1438713,1391185,1998259,1606145,1037932,1043419,1494188,1148144,1484877,1081267,1212244,2511171,1824980,2430318,1291435,1323136,1625288,900380,1560673,1337243,1140190,1146249,807389,1108195,807162,1622726,1519329,3958334,1086748,873786],["A1","A2","A3","A4","A9","A11","B1","B2","B3","B4","B9","B11","C1","C2","C3","C4","C9","C11","D1","D2","D3","D4","D9","D11","E1","E2","E3","E4","E9","E11","F1","F2","F3","F4","F9","G1","G2","G3","G4","H1","H2","H3","H4"],["PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA","PLASMA"],["normal","lupusSle","normal","lupusSle","rheumatoidArthritis","rheumatoidArthritis","lupusSle","lupusSle","normal","rheumatoidArthritis","rheumatoidArthritis","lupusSle","normal","rheumatoidArthritis","normal","rheumatoidArthritis","lupusSle","normal","normal","lupusSle","lupusSle","rheumatoidArthritis","normal","normal","lupusSle","rheumatoidArthritis","normal","lupusSle","normal","normal","normal","rheumatoidArthritis","normal","lupusSle","normal","normal","lupusSle","normal","rheumatoidArthritis","normal","rheumatoidArthritis","normal","rheumatoidArthritis"],["a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a","a"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>plateID<\/th>\n      <th>plateGUID<\/th>\n      <th>sampleType<\/th>\n      <th>sampleID<\/th>\n      <th>sampleBarcode<\/th>\n      <th>sampleName<\/th>\n      <th>sampleName_original<\/th>\n      <th>wellRow<\/th>\n      <th>wellCol<\/th>\n      <th>matching<\/th>\n      <th>non-matching<\/th>\n      <th>AUTO_WELLPOSITION<\/th>\n      <th>SAMPLE_MATRIX<\/th>\n      <th>Disease<\/th>\n      <th>Lab<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"paging":true,"searching":false,"info":true,"dom":"tip","columnDefs":[{"className":"dt-right","targets":[8,9,10]},{"name":"plateID","targets":0},{"name":"plateGUID","targets":1},{"name":"sampleType","targets":2},{"name":"sampleID","targets":3},{"name":"sampleBarcode","targets":4},{"name":"sampleName","targets":5},{"name":"sampleName_original","targets":6},{"name":"wellRow","targets":7},{"name":"wellCol","targets":8},{"name":"matching","targets":9},{"name":"non-matching","targets":10},{"name":"AUTO_WELLPOSITION","targets":11},{"name":"SAMPLE_MATRIX","targets":12},{"name":"Disease","targets":13},{"name":"Lab","targets":14}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="apply-detectability-threshold" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Apply Detectability Threshold<a href="case-study-1.html#apply-detectability-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="preview-detectability-table" class="section level4 unnumbered hasAnchor">
<h4>Preview Detectability Table<a href="case-study-1.html#preview-detectability-table" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The detectability table is stored in <code>data.a$merged$detectability</code>.</p>
<pre><code>#&gt; Preview of detectability table:</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-1b058bf7e95237c99814" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1b058bf7e95237c99814">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of detectability table<\/caption>","data":[["AGER","AGRP","ANGPT1","ANGPT2","ANXA1","AREG","BDNF","BMP7","BST2","C1QA","CALCA","CCL1","CCL11","CCL13","CCL14","CCL15","CCL16","CCL17","CCL19","CCL2","CCL20","CCL21","CCL22","CCL23","CCL24","CCL25","CCL26","CCL27","CCL28","CCL3","CCL4","CCL5","CCL7","CCL8","CD200","CD200R1","CD27","CD274","CD276","CD3E","CD4","CD40","CD40LG","CD46","CD70","CD80","CD83","CD93","CEACAM5","CHI3L1","CLEC4A","CNTF","CRP","CSF1","CSF1R","CSF2","CSF2RB","CSF3","CSF3R","CST7","CTF1","CTLA4","CTSS","CX3CL1","CXADR","CXCL1","CXCL10","CXCL11","CXCL12","CXCL13","CXCL14","CXCL16","CXCL2","CXCL3","CXCL5","CXCL6","CXCL8","CXCL9","EGF","EPO","FASLG","FGF19","FGF2","FGF21","FGF23","FLT1","FLT3LG","FLT4","FTH1","FURIN","GDF15","GDF2","GFAP","GRN","GZMA","GZMB","HAVCR1","HGF","HLA-DRA","ICAM1","ICOSLG","IFNA1; IFNA13","IFNA2","IFNB1","IFNG","IFNL1","IFNL2; IFNL3","IFNW1","IKBKG","IL10","IL10RB","IL11","IL12B","IL12p70","IL12RB1","IL13","IL13RA2","IL15","IL15RA","IL16","IL17A","IL17A|IL17F","IL17B","IL17C","IL17F","IL17RA","IL17RB","IL18","IL18BP","IL18R1","IL19","IL1B","IL1R1","IL1R2","IL1RL1","IL1RN","IL2","IL20","IL22","IL23","IL24","IL27","IL2RA","IL2RB","IL32","IL33","IL34","IL36A","IL36B","IL36G","IL3RA","IL4","IL4R","IL5","IL5RA","IL6","IL6R","IL6ST","IL7","IL7R","IL9","IRAK4","KDR","KITLG","KLRK1","KNG1","LAG3","LAMP3","LCN2","LGALS9","LIF","LILRB2","LTA","MERTK","MICA","MICB","MIF","MMP1","MMP12","MMP3","MMP8","MMP9","MPO","MUC16","NAMPT","NCR1","NGF","NTF3","OSM","OSMR","PDCD1","PDCD1LG2","PDGFA","PDGFB","PGF","PTX3","S100A12","S100A9","SCG2","SDC1","SELE","SELP","SIRPA","SLAMF1","SPP1","TAFA5","TEK","TGFB1","TGFB3","THBS2","THPO","TIMP1","TIMP2","TLR3","TNF","TNFRSF11A","TNFRSF11B","TNFRSF13B","TNFRSF13C","TNFRSF14","TNFRSF17","TNFRSF18","TNFRSF1A","TNFRSF1B","TNFRSF21","TNFRSF4","TNFRSF8","TNFRSF9","TNFSF10","TNFSF11","TNFSF12","TNFSF13","TNFSF14","TNFSF15","TNFSF18","TNFSF4","TNFSF8","TNFSF9","TREM1","TREM2","TSLP","VCAM1","VEGFA","VEGFC","VEGFD","VSNL1","VSTM1","WNT16","WNT7A"],[100,100,100,100,97.7,97.7,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,90.7,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,48.8,100,100,100,97.7,100,100,100,100,100,100,46.5,100,100,100,100,100,93,100,53.5,100,97.7,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,95.3,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,39.5,100,100,100,41.9,100,100,100,95.3,100,100,100,100,100,100,100,100,79.09999999999999,100,100,100,100,100,100,97.7,100,100,100,100,100,100,100,100,90.7,100,100,100,39.5,100,100,100,100,83.7,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,90.7,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,95.3,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Target<\/th>\n      <th>plasma (n = 43)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"paging":true,"searching":false,"info":true,"dom":"tip","columnDefs":[{"className":"dt-right","targets":1},{"name":"Target","targets":0},{"name":"plasma (n = 43)","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br>
To ensure robust statistical analysis, we can filter proteins based on the detectability across samples. We typically use a 50% detectability threshold for plasma samples, meaning a target must be detected in at least 50% of samples to be included:</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="case-study-1.html#cb207-1" tabindex="-1"></a>detectability_threshold <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb207-2"><a href="case-study-1.html#cb207-2" tabindex="-1"></a>targets_passed <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>detectability <span class="sc">%&gt;%</span> </span>
<span id="cb207-3"><a href="case-study-1.html#cb207-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">plasma (n = 43)</span><span class="st">`</span> <span class="sc">&gt;</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb207-4"><a href="case-study-1.html#cb207-4" tabindex="-1"></a>  <span class="fu">pull</span>(Target)</span>
<span id="cb207-5"><a href="case-study-1.html#cb207-5" tabindex="-1"></a></span>
<span id="cb207-6"><a href="case-study-1.html#cb207-6" tabindex="-1"></a><span class="co"># Identify excluded targets</span></span>
<span id="cb207-7"><a href="case-study-1.html#cb207-7" tabindex="-1"></a>excluded_targets <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ), targets_passed)</span>
<span id="cb207-8"><a href="case-study-1.html#cb207-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of targets excluded:&quot;</span>, <span class="fu">length</span>(excluded_targets), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Number of targets excluded: 5</code></pre>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="case-study-1.html#cb209-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Targets excluded:&quot;</span>, <span class="fu">paste</span>(excluded_targets, <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Targets excluded: CNTF, CTSS, IFNW1, IL11, IL32</code></pre>
</div>
</div>
<div id="global-data-visualization" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Global Data Visualization<a href="case-study-1.html#global-data-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before performing differential abundance analysis, we will examine the overall structure of the data using unsupervised visualization methods. These methods are unsupervised in the sense that we do not deliberately segregate data by disease class. Rather, we observe how samples are sorted according to their similarity in protein abundance patterns while we annotate samples by disease class.</p>
<div id="prepare-metadata-for-visualization" class="section level3 hasAnchor" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Prepare Metadata for Visualization<a href="case-study-1.html#prepare-metadata-for-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we clean up the metadata labels for better plot readability:</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="case-study-1.html#cb211-1" tabindex="-1"></a>metadata_rename <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span> </span>
<span id="cb211-2"><a href="case-study-1.html#cb211-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Disease =</span> <span class="fu">case_when</span>(</span>
<span id="cb211-3"><a href="case-study-1.html#cb211-3" tabindex="-1"></a>    Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span> <span class="sc">~</span> <span class="st">&quot;SLE&quot;</span>,</span>
<span id="cb211-4"><a href="case-study-1.html#cb211-4" tabindex="-1"></a>    Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span> <span class="sc">~</span> <span class="st">&quot;RA&quot;</span>,</span>
<span id="cb211-5"><a href="case-study-1.html#cb211-5" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Normal&quot;</span></span>
<span id="cb211-6"><a href="case-study-1.html#cb211-6" tabindex="-1"></a>  ),</span>
<span id="cb211-7"><a href="case-study-1.html#cb211-7" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sample Matrix</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">tolower</span>(SAMPLE_MATRIX)</span>
<span id="cb211-8"><a href="case-study-1.html#cb211-8" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="hierarchical-clustering-heatmap" class="section level3 hasAnchor" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Hierarchical Clustering Heatmap<a href="case-study-1.html#hierarchical-clustering-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The heatmap provides a global view of protein abundance patterns across all samples and identifies natural groupings:</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="case-study-1.html#cb212-1" tabindex="-1"></a>h <span class="ot">&lt;-</span> <span class="fu">generate_heatmap</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb212-2"><a href="case-study-1.html#cb212-2" tabindex="-1"></a>                      <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb212-3"><a href="case-study-1.html#cb212-3" tabindex="-1"></a>                      <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb212-4"><a href="case-study-1.html#cb212-4" tabindex="-1"></a>                      <span class="at">sample_subset =</span> metadata_rename<span class="sc">$</span>sampleName, </span>
<span id="cb212-5"><a href="case-study-1.html#cb212-5" tabindex="-1"></a>                      <span class="at">target_subset =</span> targets_passed,</span>
<span id="cb212-6"><a href="case-study-1.html#cb212-6" tabindex="-1"></a>                      <span class="at">annotate_sample_by =</span> <span class="fu">c</span>(<span class="st">&quot;Disease&quot;</span>, <span class="st">&quot;Sample Matrix&quot;</span>, <span class="st">&quot;Lab&quot;</span>))</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/global-heatmap-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation</strong>: The heatmap shows hierarchical clustering of both samples (columns) and proteins (rows). Look for:</p>
<ul>
<li>Clustering of samples by disease type (indicated by the annotation bars)</li>
<li>Groups of proteins with similar abundance patterns</li>
<li>Elevated levels of inflammatory markers in disease groups relative to controls</li>
</ul>
</div>
<div id="principal-component-analysis" class="section level3 hasAnchor" number="7.4.3">
<h3><span class="header-section-number">7.4.3</span> Principal Component Analysis<a href="case-study-1.html#principal-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>PCA shows the clustering of samples by groups in reduced dimensional space, highlighting variance in the dataset:</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="case-study-1.html#cb213-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">generate_pca</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb213-2"><a href="case-study-1.html#cb213-2" tabindex="-1"></a>                  <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb213-3"><a href="case-study-1.html#cb213-3" tabindex="-1"></a>                  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb213-4"><a href="case-study-1.html#cb213-4" tabindex="-1"></a>                  <span class="at">sample_subset =</span> metadata_rename<span class="sc">$</span>sampleName, </span>
<span id="cb213-5"><a href="case-study-1.html#cb213-5" tabindex="-1"></a>                  <span class="at">target_subset =</span> targets_passed,</span>
<span id="cb213-6"><a href="case-study-1.html#cb213-6" tabindex="-1"></a>                  <span class="at">annotate_sample_by =</span> <span class="st">&quot;Disease&quot;</span>)</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/global-pca-1.png" alt="" width="90%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation</strong>: In the PCA plot:</p>
<ul>
<li>Each point represents a sample</li>
<li>Distance between points reflects similarity in protein abundance patterns</li>
<li>Good separation between disease groups suggests distinct proteomic signatures</li>
<li>PC1 and PC2 percentages indicate how much variation each component explains</li>
</ul>
</div>
</div>
<div id="statistical-testing" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Statistical Testing<a href="case-study-1.html#statistical-testing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next we perform linear modeling to identify proteins that are significantly different in mean abundance between disease groups and controls:</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="case-study-1.html#cb214-1" tabindex="-1"></a>lmTest_da <span class="ot">&lt;-</span> <span class="fu">lmNULISAseq</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[targets_passed, metadata.a<span class="sc">$</span>sampleName],</span>
<span id="cb214-2"><a href="case-study-1.html#cb214-2" tabindex="-1"></a>                         <span class="at">sampleInfo =</span> metadata.a,</span>
<span id="cb214-3"><a href="case-study-1.html#cb214-3" tabindex="-1"></a>                         <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb214-4"><a href="case-study-1.html#cb214-4" tabindex="-1"></a>                         <span class="at">modelFormula =</span> <span class="st">&quot;Disease&quot;</span>)</span></code></pre></div>
<p>The <code>lmNULISAseq()</code> function fits a linear model for each protein with NPQ as outcome and disease group as predictor.</p>
<div id="model-results-2" class="section level4 unnumbered hasAnchor">
<h4>Model Results<a href="case-study-1.html#model-results-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The model results are stored in <code>lmTest_da$modelStats</code>.</p>
<pre><code>#&gt; Preview of `lmTest_da$modelStats` Results Table (rounded to 3 digits):</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-3aa1bea03112cc394f4f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3aa1bea03112cc394f4f">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of Model Results Table<\/caption>","data":[["AGER","AGRP","ANGPT1","ANGPT2","ANXA1","AREG","BDNF","BMP7","BST2","C1QA","CALCA","CCL1","CCL11","CCL13","CCL14","CCL15","CCL16","CCL17","CCL19","CCL2","CCL20","CCL21","CCL22","CCL23","CCL24","CCL25","CCL26","CCL27","CCL28","CCL3","CCL4","CCL5","CCL7","CCL8","CD200","CD200R1","CD27","CD274","CD276","CD3E","CD4","CD40","CD40LG","CD46","CD70","CD80","CD83","CD93","CEACAM5","CHI3L1","CLEC4A","CRP","CSF1","CSF1R","CSF2","CSF2RB","CSF3","CSF3R","CST7","CTF1","CTLA4","CX3CL1","CXADR","CXCL1","CXCL10","CXCL11","CXCL12","CXCL13","CXCL14","CXCL16","CXCL2","CXCL3","CXCL5","CXCL6","CXCL8","CXCL9","EGF","EPO","FASLG","FGF19","FGF2","FGF21","FGF23","FLT1","FLT3LG","FLT4","FTH1","FURIN","GDF15","GDF2","GFAP","GRN","GZMA","GZMB","HAVCR1","HGF","HLA-DRA","ICAM1","ICOSLG","IFNA1; IFNA13","IFNA2","IFNB1","IFNG","IFNL1","IFNL2; IFNL3","IKBKG","IL10","IL10RB","IL12B","IL12p70","IL12RB1","IL13","IL13RA2","IL15","IL15RA","IL16","IL17A","IL17A|IL17F","IL17B","IL17C","IL17F","IL17RA","IL17RB","IL18","IL18BP","IL18R1","IL19","IL1B","IL1R1","IL1R2","IL1RL1","IL1RN","IL2","IL20","IL22","IL23","IL24","IL27","IL2RA","IL2RB","IL33","IL34","IL36A","IL36B","IL36G","IL3RA","IL4","IL4R","IL5","IL5RA","IL6","IL6R","IL6ST","IL7","IL7R","IL9","IRAK4","KDR","KITLG","KLRK1","KNG1","LAG3","LAMP3","LCN2","LGALS9","LIF","LILRB2","LTA","MERTK","MICA","MICB","MIF","MMP1","MMP12","MMP3","MMP8","MMP9","MPO","MUC16","NAMPT","NCR1","NGF","NTF3","OSM","OSMR","PDCD1","PDCD1LG2","PDGFA","PDGFB","PGF","PTX3","S100A12","S100A9","SCG2","SDC1","SELE","SELP","SIRPA","SLAMF1","SPP1","TAFA5","TEK","TGFB1","TGFB3","THBS2","THPO","TIMP1","TIMP2","TLR3","TNF","TNFRSF11A","TNFRSF11B","TNFRSF13B","TNFRSF13C","TNFRSF14","TNFRSF17","TNFRSF18","TNFRSF1A","TNFRSF1B","TNFRSF21","TNFRSF4","TNFRSF8","TNFRSF9","TNFSF10","TNFSF11","TNFSF12","TNFSF13","TNFSF14","TNFSF15","TNFSF18","TNFSF4","TNFSF8","TNFSF9","TREM1","TREM2","TSLP","VCAM1","VEGFA","VEGFC","VEGFD","VSNL1","VSTM1","WNT16","WNT7A"],[12.831,12.048,12.413,12.618,8.94,10.026,13.132,11.364,10.456,12.873,12.732,12.939,13.438,12.795,12.642,12.828,11.581,12.747,11.975,11.859,11.804,13.073,12.481,12.789,13.017,11.992,12.488,12.566,13.48,11.902,12.111,12.503,11.283,13.4,11.084,12.883,12.768,13.133,13.154,11.33,12.647,12.994,13.012,12.798,3.148,12.39,13.04,12.717,12.828,11.986,13.24,14.378,11.853,12.364,12.034,13.26,12.788,12.662,12.676,13.244,12.938,12.959,12.318,11.885,12.678,12.68,9.920999999999999,12.064,11.853,12.643,12.781,13.061,12.715,7.999,5.013,12.735,12.505,12.91,12.978,12.748,13.994,11.603,11.807,13.077,9.266999999999999,12.769,11.417,12.551,13.331,10.892,13.639,12.53,12.734,10.455,12.753,13.267,12.601,12.575,12.696,11.002,11.353,11.335,12.192,13.528,7.45,13.857,11.954,12.714,12.815,12.699,13.119,12.879,11.023,12.833,13.099,12.413,7.647,6.327,13.22,12.98,10.329,12.886,12.823,12.775,12.823,12.536,12.623,8.712999999999999,13.022,12.492,11.986,12.149,9.827999999999999,12.878,12.539,11.72,8.211,12.841,12.7,13.122,9.385,12.143,13.58,8.007999999999999,7.051,12.863,8.558999999999999,12.845,12.455,12.753,12.718,12.584,12.885,12.525,12.619,13.166,13.649,12.294,9.851000000000001,12.61,13.239,12.946,11.431,13.458,12.932,8.680999999999999,12.82,8.101000000000001,12.919,12.811,12.635,14.882,13.606,13.361,12.79,12.675,13.488,11.57,12.597,12.326,12.887,11.793,9.019,14.342,13.107,12.587,12.942,13.228,12.645,13.386,12.848,12.138,13.253,13.745,12.395,12.12,12.503,12.708,12.44,12.427,13.75,12.545,13.059,11.99,12.196,13.105,12.436,9.353,12.358,13.033,13.095,12.648,12.753,12.887,13,12.974,13.83,12.988,12.868,12.322,13.059,12.44,10.797,13.062,12.741,13.032,13.346,12.368,13.917,12.992,13.244,12.819,11.089,13.411,11.921,12.41,12.958,14.214,13.313,7.429,11.507,12.386,13.647,8.804],[88.126,93.925,63.581,90.444,16.024,204.379,37.253,165.384,41.401,82.88,62.909,87.736,101.999,78.145,100.503,49.787,69.00700000000001,35.026,44.856,78.268,35.932,109.536,81.616,110.675,48.411,56.467,61.186,120.884,58.994,30.561,42.779,27.068,44.166,77.07599999999999,12.625,97.59999999999999,100.267,120.77,152.784,52.767,170.296,125.376,48.939,170.402,19.696,107.514,117.419,108.946,74.104,42.449,85.70099999999999,42.382,61.578,71.91800000000001,41.351,72.209,90.998,123.999,62.82,53.618,122.883,181.442,93.04600000000001,32.014,54.574,34.07,18.948,55.846,29.701,126.035,38.076,47.396,37.942,49.62,22.017,71.14400000000001,36.226,83.01600000000001,94.69,51.367,44.64,39.157,57.001,170.964,70.86799999999999,135.331,62.008,106.192,27.814,99.785,71.714,161.78,119.11,68.72199999999999,62.536,76.11,69.679,108.667,139.312,23.319,26.41,38.461,53.428,57.158,36.192,46.727,38.631,146.769,56.896,91.23,110.786,34.558,113.78,108.377,117.212,48.839,17.258,24.177,96.34999999999999,56.742,24.11,187.103,68.297,44.005,98.419,129.144,47.464,18.587,176.378,138.968,87.291,40.202,75.48099999999999,105.154,36.237,57.923,19.362,42.894,94.941,81.812,69.788,60.299,65.31699999999999,54.892,87.375,99.434,36.034,173.646,53.316,65.334,44.473,128.468,224.011,63.628,90.95399999999999,60.729,41.617,30.14,98.212,95.13200000000001,151.252,82.40600000000001,63.626,88.547,112.253,46.847,136.789,32.592,132.711,32.654,74.086,47.115,48.558,72.12,58.364,44.042,33.33,63.592,57.526,30.864,78.791,91.04600000000001,21.719,50.674,131.519,91.696,151.013,59.514,55.181,126.324,74.43000000000001,23.449,76.29900000000001,82.089,102.935,64.089,80.078,107.381,75.63800000000001,53.552,83.752,181.176,127.019,63.831,85.191,133.087,66.73399999999999,100.118,73.28400000000001,46.485,101.838,98.602,100.534,91.696,120.813,110.247,123.474,108.348,39.867,125.316,89.411,85.261,68.02,101.44,62.975,120.167,57.902,58.935,143.936,136.493,136.489,119.869,95.61,84.946,69.102,85.126,122.194,138.412,119.289,80.952,128.908,42.401,86.401,71.123],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.233,0.226,0.506,0.651,1.562,0.249,1.447,0.208,1.435,0.862,-0.107,1.018,0.414,0.779,0.412,0.729,0.417,1.51,1.118,0.897,0.369,0.75,1.005,0.203,0.521,0.722,0.975,0.527,0.601,1.246,0.864,-0.57,1.202,0.514,2,-0.003,0.76,0.648,0.137,0.066,0.466,0.722,1.34,0.467,0.301,0.748,0.879,0.351,0.054,0.757,0.032,-1.162,0.93,0.796,0.593,0.39,0.08400000000000001,0.492,1.159,0.897,0.831,0.303,0.423,0.5,0.955,2.485,1.119,1.315,0.546,0.239,-1.272,-0.845,-0.723,0.929,0.768,0.625,0.819,-0.07199999999999999,0.628,-0.455,0.82,0.618,0.946,-0.235,0.252,0.188,0.282,0.749,-0.498,0.254,0.396,0.231,0.676,1.11,0.893,1.462,0.292,0.428,0.025,1.455,1.191,0.345,0.422,0.5669999999999999,0.341,-0.9350000000000001,0.925,0.48,1.324,0.612,0.293,0.109,0.221,-0.13,0.599,1.911,1.009,0.908,0.351,-0.35,0.723,0.326,0.105,2.476,0.617,0.506,0.648,3.589,-0.003,0.192,0.503,2.372,0.441,0.06900000000000001,-0.053,0.781,0.398,0.5629999999999999,0.773,0.639,0.511,0.519,0.124,0.091,0.305,0.327,1.516,0.21,-0.023,-0.756,0.753,0.43,0.214,0.11,0.423,0.6909999999999999,1.562,0.516,0.341,0.526,0.042,0.66,0.627,0.867,1.184,1.059,0.407,0.253,0.352,0.065,0.764,2.024,0.9409999999999999,0.539,-0.138,2.249,2.721,1.052,0.483,3.01,0.775,0.251,1.009,2.231,0.533,0.921,0.027,0.889,1.228,0.3,1.218,4.436,0.364,-0.019,0.416,0.536,0.179,0.167,0.63,0.547,0.776,0.159,0.544,0.662,0.614,0.308,0.47,0.526,0.155,0.945,0.611,0.339,0.154,0.6909999999999999,0.628,0.284,0.532,0.628,0.795,0.572,1.099,0.788,0.887,-0.72,0.223,0.551,1.092,0.778,0.134,0.694,0.413,0.506,-0.142,0.614,1.064,0.633,0.352,0.5659999999999999,0.423,0.424,0.203,1.303,-0.029,0.288],[0.994,1.095,1.613,2.902,1.742,3.158,2.554,1.886,3.535,3.452,-0.33,4.295,1.957,2.96,2.04,1.761,1.546,2.581,2.606,3.682,0.698,3.91,4.088,1.093,1.206,2.115,2.971,3.156,1.636,1.991,1.9,-0.768,2.926,1.838,1.417,-0.014,3.715,3.705,0.99,0.191,3.905,4.337,3.136,3.871,1.17,4.038,4.926,1.873,0.195,1.667,0.129,-2.132,3.005,2.881,1.267,1.322,0.373,3.001,3.574,2.26,4.913,2.642,1.989,0.838,2.559,4.154,1.33,3.789,0.851,1.479,-2.358,-1.907,-1.342,3.585,2.098,2.171,1.475,-0.287,2.852,-1.141,1.628,1.298,2.843,-1.912,1.197,1.24,0.953,3.945,-0.647,1.449,1.296,1.856,3.935,4.54,2.725,5.218,1.003,2.302,0.167,1.919,1.724,0.729,1.151,1.491,1.031,-1.962,1.861,3.449,3.657,2.736,1.541,0.182,1.42,-0.6820000000000001,3.336,4.679,1.416,2.159,1.59,-0.952,1.05,2.942,0.347,5.307,2.944,3.241,1.516,4.764,-0.023,1.332,2.279,4.883,2.107,0.35,-0.095,2.4,0.584,1.171,3.597,2.477,2.365,1.602,0.373,0.389,2.348,1.574,3.971,1.765,-0.061,-2.41,1.639,2.729,2.315,0.347,1.899,1.984,2.964,0.788,2.114,2.469,0.297,2.614,2.172,3.55,6.393,3.555,2.703,0.633,2.25,0.103,2.789,3.986,2.09,1.809,-0.391,4.862,4.183,3.596,1.373,4.689,2.95,1.204,1.511,4.905,3.33,4.172,0.196,2.488,3.334,1.762,4.392,5.332,1.304,-0.06900000000000001,2.151,1.765,0.713,0.876,2.384,1.466,2.939,1.432,3.292,2.193,2.669,1.944,1.568,3.504,0.572,2.098,2.955,1.642,0.754,3.059,3.631,1.499,2.956,3.259,1.533,3.618,4.682,3.358,3.477,-3.481,0.6860000000000001,3.159,2.947,2.306,0.864,4.539,2.646,2.946,-0.763,2.418,3.837,2.7,2.068,3.428,2.357,2.878,1.417,2.776,-0.114,1.449],[0.326,0.28,0.115,0.006,0.089,0.003,0.015,0.067,0.001,0.001,0.743,0,0.057,0.005,0.048,0.08599999999999999,0.13,0.014,0.013,0.001,0.489,0,0,0.281,0.235,0.041,0.005,0.003,0.11,0.053,0.065,0.447,0.006,0.073,0.164,0.989,0.001,0.001,0.328,0.85,0,0,0.003,0,0.249,0,0,0.068,0.846,0.103,0.898,0.039,0.005,0.006,0.212,0.194,0.711,0.005,0.001,0.029,0,0.012,0.054,0.407,0.014,0,0.191,0,0.4,0.147,0.023,0.064,0.187,0.001,0.042,0.036,0.148,0.775,0.007,0.261,0.111,0.202,0.007,0.063,0.238,0.222,0.347,0,0.521,0.155,0.202,0.07099999999999999,0,0,0.008999999999999999,0,0.322,0.027,0.868,0.062,0.092,0.47,0.256,0.144,0.309,0.057,0.07000000000000001,0.001,0.001,0.008999999999999999,0.131,0.857,0.163,0.499,0.002,0,0.164,0.037,0.12,0.347,0.3,0.005,0.731,0,0.005,0.002,0.137,0,0.981,0.19,0.028,0,0.041,0.728,0.924,0.021,0.5629999999999999,0.249,0.001,0.018,0.023,0.117,0.711,0.699,0.024,0.123,0,0.08500000000000001,0.952,0.021,0.109,0.008999999999999999,0.026,0.73,0.065,0.054,0.005,0.436,0.041,0.018,0.768,0.013,0.036,0.001,0,0.001,0.01,0.531,0.03,0.918,0.008,0,0.043,0.078,0.698,0,0,0.001,0.177,0,0.005,0.236,0.139,0,0.002,0,0.845,0.017,0.002,0.08599999999999999,0,0,0.2,0.945,0.038,0.08500000000000001,0.48,0.386,0.022,0.15,0.005,0.16,0.002,0.034,0.011,0.059,0.125,0.001,0.571,0.042,0.005,0.108,0.455,0.004,0.001,0.142,0.005,0.002,0.133,0.001,0,0.002,0.001,0.001,0.497,0.003,0.005,0.026,0.393,0,0.012,0.005,0.45,0.02,0,0.01,0.045,0.001,0.023,0.006,0.164,0.008,0.91,0.155],[0.396,0.348,0.178,0.019,0.145,0.012,0.037,0.116,0.006,0.007,0.792,0.001,0.105,0.017,0.091,0.141,0.196,0.035,0.033,0.005,0.555,0.003,0.002,0.348,0.303,0.082,0.017,0.012,0.173,0.101,0.114,0.519,0.018,0.124,0.225,0.989,0.005,0.005,0.396,0.886,0.003,0.001,0.013,0.003,0.315,0.003,0,0.118,0.886,0.166,0.925,0.08,0.017,0.02,0.277,0.258,0.772,0.017,0.005,0.063,0,0.031,0.101,0.477,0.037,0.002,0.256,0.004,0.471,0.212,0.053,0.114,0.254,0.005,0.082,0.075,0.212,0.819,0.021,0.326,0.174,0.265,0.021,0.113,0.304,0.288,0.415,0.003,0.584,0.219,0.265,0.121,0.003,0.001,0.027,0,0.393,0.058,0.897,0.112,0.149,0.539,0.323,0.209,0.379,0.105,0.12,0.007,0.005,0.027,0.196,0.89,0.225,0.5610000000000001,0.008,0.001,0.225,0.076,0.184,0.415,0.37,0.017,0.782,0,0.017,0.01,0.203,0.001,0.985,0.256,0.061,0,0.082,0.782,0.9399999999999999,0.05,0.624,0.315,0.005,0.043,0.053,0.181,0.772,0.765,0.054,0.188,0.003,0.141,0.96,0.049,0.173,0.027,0.058,0.782,0.114,0.101,0.017,0.508,0.082,0.044,0.8149999999999999,0.033,0.075,0.006,0,0.006,0.028,0.591,0.064,0.9379999999999999,0.024,0.003,0.083,0.131,0.765,0,0.002,0.005,0.242,0.001,0.017,0.303,0.204,0,0.008,0.002,0.886,0.043,0.008,0.141,0.001,0,0.265,0.957,0.077,0.141,0.548,0.46,0.051,0.215,0.017,0.224,0.008999999999999999,0.073,0.03,0.107,0.189,0.006,0.63,0.082,0.017,0.173,0.524,0.015,0.005,0.207,0.017,0.01,0.198,0.005,0.001,0.008,0.006,0.006,0.5610000000000001,0.012,0.017,0.058,0.465,0.001,0.031,0.017,0.52,0.049,0.003,0.028,0.08699999999999999,0.007,0.053,0.02,0.225,0.024,0.9330000000000001,0.219],[1,1,1,1,1,0.738,1,1,0.255,0.324,1,0.026,1,1,1,1,1,1,1,0.167,1,0.08500000000000001,0.05,1,1,1,1,0.742,1,1,1,1,1,1,1,1,0.151,0.156,1,1,0.08599999999999999,0.023,0.783,0.095,1,0.058,0.004,1,1,1,1,1,1,1,1,1,1,1,0.229,1,0.004,1,1,1,1,0.041,1,0.122,1,1,1,1,1,0.221,1,1,1,1,1,1,1,1,1,1,1,1,1,0.077,1,1,1,1,0.079,0.012,1,0.001,1,1,1,1,1,1,1,1,1,1,1,0.327,0.18,1,1,1,1,1,0.45,0.008,1,1,1,1,1,1,1,0.001,1,0.586,1,0.006,1,1,1,0.004,1,1,1,1,1,1,0.214,1,1,1,1,1,1,1,0.07099999999999999,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.245,0,0.241,1,1,1,1,1,0.068,1,1,1,0.004,0.037,0.214,1,0.008,1,1,1,0.004,0.457,0.038,1,1,0.452,1,0.02,0.001,1,1,1,1,1,1,1,1,1,1,0.508,1,1,1,1,0.279,1,1,1,1,1,0.965,0.194,1,1,0.5570000000000001,1,0.201,0.008,0.422,0.301,0.298,1,0.734,1,1,1,0.012,1,1,1,1,0.106,1,1,0.346,1,1,1,1,1,1],[0.155,0.288,0.295,0.462,-0.107,0.367,1.041,0.523,0.766,0.551,0.395,1.045,0.791,0.6879999999999999,0.042,0.046,0.249,0.642,0.791,1.005,-0.162,0.6909999999999999,0.86,0.106,0.9429999999999999,0.48,0.998,0.409,0.416,0.569,0.6830000000000001,-0.474,1.619,0.596,1.244,0.195,0.766,0.658,0.287,0.347,0.368,0.584,0.906,0.586,0.238,0.826,0.839,0.339,0.492,1.444,0.008999999999999999,-0.595,0.773,0.696,0.575,-0.199,-0.222,0.427,0.7,0.637,0.794,0.08400000000000001,0.415,-0.365,0.331,1.816,0.662,0.9419999999999999,0.769,0.449,-1.044,-0.781,-0.855,1.149,0.516,0.218,0.636,0.335,0.855,-0.06,0.113,0.258,-0.129,-0.226,0.222,0.189,0.964,0.619,0.874,0.375,0.028,0.365,0.523,0.9419999999999999,0.795,1.192,0.638,0.386,0.002,1.712,1.652,0.796,0.322,0.703,0.745,-0.868,0.253,0.5580000000000001,0.962,0.493,0.315,-0.13,0.467,-0.075,0.728,1.542,-0.26,0.773,0.282,-0.325,0.428,0.38,0.699,2.351,0.7,0.661,0.599,3.13,0.281,0.312,0.73,1.749,0.36,0.251,0.283,0.395,1.049,-0.719,0.58,0.526,0.234,0.367,-0.307,0.118,0.349,0.538,0.921,0.224,-0.11,0.046,0.967,0.349,0.359,-0.22,0.025,0.191,0.745,-0.17,0.118,0.74,0.163,0.589,0.468,0.905,0.853,1.21,0.393,1.142,0.327,0.576,0.574,1.173,0.511,0.928,0.004,2.406,2.208,1.121,0.461,2.131,0.905,0.422,0.468,1.888,0.464,0.991,0.354,0.542,1.213,0.456,1.237,4.303,0.433,0.443,0.489,0.779,0.385,0.389,0.724,0.6929999999999999,0.459,0.359,0.342,0.661,0.592,0.131,0.544,0.352,0.276,1.532,0.636,0.5590000000000001,0.269,0.551,0.593,0.262,0.474,0.544,1.743,0.519,0.995,0.8169999999999999,0.891,-0.713,-0.302,0.218,0.945,0.5,0.179,0.582,0.389,0.849,0.089,0.626,1.017,0.742,0.362,0.308,0.597,0.324,0.313,0.181,0.031,0.24],[0.662,1.397,0.9389999999999999,2.06,-0.119,4.661,1.837,4.737,1.888,2.207,1.213,4.409,3.734,2.616,0.208,0.111,0.922,1.097,1.843,4.125,-0.306,3.6,3.497,0.573,2.183,1.406,3.043,2.445,1.133,0.909,1.501,-0.638,3.943,2.134,0.881,0.919,3.741,3.762,2.076,1.006,3.079,3.505,2.12,4.856,0.925,4.46,4.702,1.807,1.768,3.181,0.035,-1.09,2.498,2.519,1.23,-0.674,-0.982,2.599,2.158,1.603,4.692,0.728,1.949,-0.611,0.887,3.035,0.787,2.713,1.199,2.785,-1.934,-1.764,-1.588,4.433,1.411,0.759,1.146,1.342,3.882,-0.15,0.225,0.542,-0.387,-1.836,1.059,1.247,3.256,3.258,1.135,2.139,0.093,2.929,3.046,3.853,2.425,4.256,2.194,2.077,0.017,2.257,2.391,1.681,0.877,1.848,2.253,-1.822,0.508,4.005,2.657,2.205,1.653,-0.218,2.996,-0.395,4.051,3.776,-0.366,1.838,1.276,-0.885,0.622,3.429,2.317,5.039,3.342,4.234,1.401,4.155,2.366,2.16,3.309,3.602,1.721,1.277,0.509,1.213,1.539,-1.493,2.697,2.039,1.083,1.133,-0.92,0.504,2.694,2.589,2.412,1.881,-0.292,0.148,2.103,2.217,3.887,-0.695,0.113,0.548,1.413,-0.259,0.733,3.474,1.156,2.333,1.62,3.706,4.609,4.061,2.607,2.859,2.088,0.914,2.094,2.31,1.135,3.115,0.013,5.202,3.394,3.832,1.31,3.319,3.442,2.025,0.702,4.151,2.896,4.49,2.568,1.516,3.292,2.677,4.459,5.172,1.549,1.648,2.529,2.564,1.535,2.045,2.738,1.858,1.738,3.224,2.067,2.189,2.574,0.828,1.817,2.346,1.019,3.4,3.079,2.709,1.318,2.44,3.429,1.385,2.632,2.824,3.359,3.286,4.238,3.484,3.494,-3.445,-0.928,1.251,2.552,1.482,1.15,3.807,2.497,4.938,0.48,2.467,3.67,3.166,2.126,1.866,3.33,2.198,2.184,0.385,0.121,1.206],[0.512,0.17,0.353,0.046,0.906,0,0.074,0,0.066,0.033,0.232,0,0.001,0.012,0.836,0.912,0.362,0.279,0.073,0,0.761,0.001,0.001,0.57,0.035,0.168,0.004,0.019,0.264,0.369,0.141,0.527,0,0.039,0.383,0.363,0.001,0.001,0.044,0.32,0.004,0.001,0.04,0,0.36,0,0,0.078,0.08500000000000001,0.003,0.972,0.282,0.017,0.016,0.226,0.504,0.332,0.013,0.037,0.117,0,0.471,0.058,0.545,0.38,0.004,0.436,0.01,0.238,0.008,0.06,0.08500000000000001,0.12,0,0.166,0.453,0.259,0.187,0,0.882,0.823,0.591,0.701,0.074,0.296,0.22,0.002,0.002,0.263,0.039,0.926,0.006,0.004,0,0.02,0,0.034,0.044,0.987,0.03,0.022,0.1,0.385,0.07199999999999999,0.03,0.076,0.614,0,0.011,0.033,0.106,0.829,0.005,0.695,0,0.001,0.716,0.073,0.209,0.382,0.537,0.001,0.026,0,0.002,0,0.169,0,0.023,0.037,0.002,0.001,0.093,0.209,0.614,0.232,0.132,0.143,0.01,0.048,0.285,0.264,0.363,0.617,0.01,0.013,0.021,0.067,0.772,0.883,0.042,0.032,0,0.491,0.911,0.587,0.165,0.797,0.468,0.001,0.254,0.025,0.113,0.001,0,0,0.013,0.007,0.043,0.366,0.043,0.026,0.263,0.003,0.99,0,0.002,0,0.198,0.002,0.001,0.05,0.487,0,0.006,0,0.014,0.137,0.002,0.011,0,0,0.129,0.107,0.015,0.014,0.133,0.047,0.008999999999999999,0.07099999999999999,0.09,0.003,0.045,0.034,0.014,0.412,0.077,0.024,0.314,0.002,0.004,0.01,0.195,0.019,0.001,0.174,0.012,0.007,0.002,0.002,0,0.001,0.001,0.001,0.359,0.218,0.015,0.146,0.257,0,0.017,0,0.634,0.018,0.001,0.003,0.04,0.06900000000000001,0.002,0.034,0.035,0.702,0.904,0.235],[0.583,0.25,0.445,0.08799999999999999,0.927,0.001,0.128,0.001,0.122,0.074,0.32,0.001,0.004,0.037,0.872,0.927,0.446,0.364,0.128,0.002,0.8110000000000001,0.006,0.007,0.638,0.075,0.249,0.015,0.049,0.346,0.448,0.217,0.598,0.003,0.081,0.459,0.446,0.004,0.004,0.08699999999999999,0.407,0.014,0.007,0.082,0.001,0.446,0.001,0.001,0.133,0.143,0.012,0.98,0.366,0.044,0.043,0.315,0.578,0.42,0.038,0.078,0.186,0.001,0.547,0.109,0.612,0.459,0.015,0.514,0.032,0.324,0.027,0.111,0.143,0.19,0.001,0.248,0.531,0.346,0.272,0.003,0.913,0.866,0.656,0.755,0.128,0.38,0.308,0.01,0.01,0.346,0.081,0.9379999999999999,0.02,0.015,0.004,0.05,0.002,0.075,0.08699999999999999,0.99,0.06900000000000001,0.053,0.165,0.459,0.128,0.06900000000000001,0.131,0.675,0.003,0.034,0.074,0.173,0.868,0.017,0.754,0.002,0.004,0.767,0.128,0.297,0.459,0.607,0.007,0.061,0.001,0.008999999999999999,0.002,0.25,0.002,0.056,0.078,0.008999999999999999,0.006,0.153,0.297,0.675,0.32,0.206,0.218,0.032,0.091,0.368,0.346,0.446,0.675,0.032,0.038,0.051,0.122,0.819,0.913,0.08400000000000001,0.074,0.003,0.5649999999999999,0.927,0.654,0.248,0.842,0.546,0.007,0.345,0.059,0.182,0.004,0.001,0.002,0.038,0.023,0.08599999999999999,0.447,0.08500000000000001,0.061,0.346,0.013,0.99,0.001,0.008,0.004,0.284,0.008999999999999999,0.007,0.093,0.5629999999999999,0.002,0.021,0.001,0.039,0.212,0.008999999999999999,0.033,0.001,0.001,0.203,0.173,0.042,0.039,0.206,0.091,0.03,0.127,0.149,0.01,0.08799999999999999,0.075,0.039,0.488,0.131,0.058,0.401,0.008,0.014,0.032,0.282,0.049,0.007,0.254,0.036,0.025,0.008,0.008999999999999999,0.002,0.007,0.007,0.007,0.446,0.308,0.04,0.222,0.346,0.004,0.044,0.001,0.6909999999999999,0.047,0.005,0.012,0.081,0.125,0.008999999999999999,0.075,0.075,0.755,0.927,0.322],[1,1,1,1,1,0.008,1,0.007,1,1,1,0.019,0.143,1,1,1,1,1,1,0.044,1,0.212,0.285,1,1,1,1,1,1,1,1,1,0.077,1,1,1,0.14,0.132,1,1,0.914,0.278,1,0.005,1,0.016,0.007,1,1,0.6919999999999999,1,1,1,1,1,1,1,1,1,1,0.008,1,1,1,1,1,1,1,1,1,1,1,1,0.017,1,1,1,1,0.092,1,1,1,1,1,1,1,0.5629999999999999,0.5590000000000001,1,1,1,1,0.999,0.101,1,0.03,1,1,1,1,1,1,1,1,1,1,1,0.064,1,1,1,1,1,1,0.056,0.127,1,1,1,1,1,0.346,1,0.003,0.442,0.032,1,0.041,1,1,0.486,0.211,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.091,1,1,1,1,1,1,0.304,1,1,1,0.155,0.01,0.054,1,1,1,1,1,1,1,0.828,1,0.002,0.382,0.107,1,0.471,0.334,1,1,0.041,1,0.014,1,1,0.508,1,0.016,0.002,1,1,1,1,1,1,1,1,1,0.615,1,1,1,1,1,1,1,0.375,0.914,1,1,1,0.346,1,1,1,0.421,0.517,0.031,0.296,0.287,0.331,1,1,1,1,1,0.116,1,0.004,1,1,0.173,0.721,1,1,0.457,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>target<\/th>\n      <th>intercept_coef<\/th>\n      <th>intercept_tstat<\/th>\n      <th>intercept_pval_unadj<\/th>\n      <th>intercept_pval_FDR<\/th>\n      <th>intercept_pval_bonf<\/th>\n      <th>DiseaselupusSle_coef<\/th>\n      <th>DiseaselupusSle_tstat<\/th>\n      <th>DiseaselupusSle_pval_unadj<\/th>\n      <th>DiseaselupusSle_pval_FDR<\/th>\n      <th>DiseaselupusSle_pval_bonf<\/th>\n      <th>DiseaserheumatoidArthritis_coef<\/th>\n      <th>DiseaserheumatoidArthritis_tstat<\/th>\n      <th>DiseaserheumatoidArthritis_pval_unadj<\/th>\n      <th>DiseaserheumatoidArthritis_pval_FDR<\/th>\n      <th>DiseaserheumatoidArthritis_pval_bonf<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"paging":true,"searching":false,"info":true,"dom":"tip","columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]},{"name":"target","targets":0},{"name":"intercept_coef","targets":1},{"name":"intercept_tstat","targets":2},{"name":"intercept_pval_unadj","targets":3},{"name":"intercept_pval_FDR","targets":4},{"name":"intercept_pval_bonf","targets":5},{"name":"DiseaselupusSle_coef","targets":6},{"name":"DiseaselupusSle_tstat","targets":7},{"name":"DiseaselupusSle_pval_unadj","targets":8},{"name":"DiseaselupusSle_pval_FDR","targets":9},{"name":"DiseaselupusSle_pval_bonf","targets":10},{"name":"DiseaserheumatoidArthritis_coef","targets":11},{"name":"DiseaserheumatoidArthritis_tstat","targets":12},{"name":"DiseaserheumatoidArthritis_pval_unadj","targets":13},{"name":"DiseaserheumatoidArthritis_pval_FDR","targets":14},{"name":"DiseaserheumatoidArthritis_pval_bonf","targets":15}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>For comparing SLE to Normal:</p>
<ul>
<li><strong><code>target</code></strong>: Target name</li>
<li><strong><code>DiseaselupusSle_coef</code></strong>: Difference in mean NPQ (log2 fold change) between SLE and Normal</li>
<li><strong><code>DiseaselupusSle_pval</code></strong>: Unadjusted p-value</li>
<li><strong><code>DiseaselupusSle_pval_FDR</code></strong>: FDR-adjusted p-value</li>
<li><strong><code>DiseaselupusSle_pval_bonf</code></strong>: Bonferroni-adjusted p-value
<br></li>
</ul>
</div>
</div>
<div id="systemic-lupus-erythematosus-sle-analysis" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Systemic Lupus Erythematosus (SLE) Analysis<a href="case-study-1.html#systemic-lupus-erythematosus-sle-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="background" class="section level3 hasAnchor" number="7.6.1">
<h3><span class="header-section-number">7.6.1</span> Background<a href="case-study-1.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Systemic Lupus Erythematosus is characterized by chronic inflammation affecting multiple organ systems. The immune dysregulation in SLE involves abnormal B and T cell activation, autoantibody production, and inflammatory cytokine release. The proteomic signature of SLE typically shows:</p>
<ul>
<li>Elevated inflammatory cytokines and chemokines</li>
<li>Increased acute phase proteins</li>
<li>Markers of immune activation</li>
<li>Proteins involved in tissue damage and repair</li>
</ul>
</div>
<div id="volcano-plot" class="section level3 hasAnchor" number="7.6.2">
<h3><span class="header-section-number">7.6.2</span> Volcano Plot<a href="case-study-1.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The volcano plot visualizes both the magnitude (fold change) and statistical significance of protein changes:</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="case-study-1.html#cb216-1" tabindex="-1"></a><span class="fu">volcanoPlot</span>(<span class="at">coefs =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaselupusSle_coef,</span>
<span id="cb216-2"><a href="case-study-1.html#cb216-2" tabindex="-1"></a>            <span class="at">p_vals =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaselupusSle_pval_FDR,</span>
<span id="cb216-3"><a href="case-study-1.html#cb216-3" tabindex="-1"></a>            <span class="at">target_labels =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>target,</span>
<span id="cb216-4"><a href="case-study-1.html#cb216-4" tabindex="-1"></a>            <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Systemic Lupus Erythematosus (SLE) (n=&quot;</span>, n_sle, <span class="st">&quot;) </span><span class="sc">\n</span><span class="st"> vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>))</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/sle-volcano-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation</strong>:</p>
<ul>
<li><strong>X-axis (coefficient)</strong>: Log2 fold change; positive values indicate elevated levels in SLE</li>
<li><strong>Y-axis (-log10 p-value)</strong>: Statistical significance; higher values = more significant</li>
<li><strong>Colored points</strong>: Proteins passing the FDR &lt; 0.05 threshold</li>
<li><strong>Upper left/right corners</strong>: Most significantly changed proteins</li>
</ul>
</div>
<div id="visualizations" class="section level3 hasAnchor" number="7.6.3">
<h3><span class="header-section-number">7.6.3</span> Visualizations<a href="case-study-1.html#visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we examine only the samples and proteins relevant to the SLE comparison:</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="case-study-1.html#cb217-1" tabindex="-1"></a><span class="co"># Select SLE and normal samples</span></span>
<span id="cb217-2"><a href="case-study-1.html#cb217-2" tabindex="-1"></a>sle_samples <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span></span>
<span id="cb217-3"><a href="case-study-1.html#cb217-3" tabindex="-1"></a>  <span class="fu">filter</span>(Disease <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>, <span class="st">&quot;lupusSle&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb217-4"><a href="case-study-1.html#cb217-4" tabindex="-1"></a>  <span class="fu">pull</span>(sampleName)</span>
<span id="cb217-5"><a href="case-study-1.html#cb217-5" tabindex="-1"></a></span>
<span id="cb217-6"><a href="case-study-1.html#cb217-6" tabindex="-1"></a><span class="co"># Identify significantly differential proteins</span></span>
<span id="cb217-7"><a href="case-study-1.html#cb217-7" tabindex="-1"></a>sig_targets_sle <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb217-8"><a href="case-study-1.html#cb217-8" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb217-9"><a href="case-study-1.html#cb217-9" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb217-10"><a href="case-study-1.html#cb217-10" tabindex="-1"></a></span>
<span id="cb217-11"><a href="case-study-1.html#cb217-11" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of significant targets in SLE:&quot;</span>, <span class="fu">length</span>(sig_targets_sle), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Number of significant targets in SLE: 102</code></pre>
<div id="sle-specific-heatmap" class="section level4 hasAnchor" number="7.6.3.1">
<h4><span class="header-section-number">7.6.3.1</span> SLE-Specific Heatmap<a href="case-study-1.html#sle-specific-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="case-study-1.html#cb219-1" tabindex="-1"></a>h_sle <span class="ot">&lt;-</span> <span class="fu">generate_heatmap</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb219-2"><a href="case-study-1.html#cb219-2" tabindex="-1"></a>                          <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb219-3"><a href="case-study-1.html#cb219-3" tabindex="-1"></a>                          <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb219-4"><a href="case-study-1.html#cb219-4" tabindex="-1"></a>                          <span class="at">sample_subset =</span> sle_samples, </span>
<span id="cb219-5"><a href="case-study-1.html#cb219-5" tabindex="-1"></a>                          <span class="at">target_subset =</span> sig_targets_sle,</span>
<span id="cb219-6"><a href="case-study-1.html#cb219-6" tabindex="-1"></a>                          <span class="at">annotate_sample_by =</span> <span class="fu">c</span>(<span class="st">&quot;Disease&quot;</span>, <span class="st">&quot;Sample Matrix&quot;</span>, <span class="st">&quot;Lab&quot;</span>),</span>
<span id="cb219-7"><a href="case-study-1.html#cb219-7" tabindex="-1"></a>                          <span class="at">row_fontsize =</span> <span class="dv">6</span>,</span>
<span id="cb219-8"><a href="case-study-1.html#cb219-8" tabindex="-1"></a>                          <span class="at">col_fontsize =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/sle-heatmap-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
<p>This heatmap focuses exclusively on proteins we found to be significantly altered in SLE, making disease-specific patterns more apparent.</p>
</div>
<div id="sle-specific-pca" class="section level4 hasAnchor" number="7.6.3.2">
<h4><span class="header-section-number">7.6.3.2</span> SLE-Specific PCA<a href="case-study-1.html#sle-specific-pca" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="case-study-1.html#cb220-1" tabindex="-1"></a>p_sle <span class="ot">&lt;-</span> <span class="fu">generate_pca</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb220-2"><a href="case-study-1.html#cb220-2" tabindex="-1"></a>                      <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb220-3"><a href="case-study-1.html#cb220-3" tabindex="-1"></a>                      <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb220-4"><a href="case-study-1.html#cb220-4" tabindex="-1"></a>                      <span class="at">sample_subset =</span> sle_samples, </span>
<span id="cb220-5"><a href="case-study-1.html#cb220-5" tabindex="-1"></a>                      <span class="at">target_subset =</span> sig_targets_sle,</span>
<span id="cb220-6"><a href="case-study-1.html#cb220-6" tabindex="-1"></a>                      <span class="at">annotate_sample_by =</span> <span class="st">&quot;Disease&quot;</span>)</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/sle-pca-1.png" alt="" width="90%" style="display: block; margin: auto;" /></p>
<p>Using only significantly differential proteins often enhances separation between groups in PCA.</p>
</div>
</div>
<div id="directional-analysis" class="section level3 hasAnchor" number="7.6.4">
<h3><span class="header-section-number">7.6.4</span> Directional Analysis<a href="case-study-1.html#directional-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can separate elevated and reduced proteins to understand the nature of proteomic changes:</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="case-study-1.html#cb221-1" tabindex="-1"></a><span class="co"># Elevated proteins (higher in SLE)</span></span>
<span id="cb221-2"><a href="case-study-1.html#cb221-2" tabindex="-1"></a>sig_targets_sle_up <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb221-3"><a href="case-study-1.html#cb221-3" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, DiseaselupusSle_coef <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb221-4"><a href="case-study-1.html#cb221-4" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb221-5"><a href="case-study-1.html#cb221-5" tabindex="-1"></a></span>
<span id="cb221-6"><a href="case-study-1.html#cb221-6" tabindex="-1"></a><span class="co"># Reduced proteins (lower in SLE)</span></span>
<span id="cb221-7"><a href="case-study-1.html#cb221-7" tabindex="-1"></a>sig_targets_sle_down <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb221-8"><a href="case-study-1.html#cb221-8" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, DiseaselupusSle_coef <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb221-9"><a href="case-study-1.html#cb221-9" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb221-10"><a href="case-study-1.html#cb221-10" tabindex="-1"></a></span>
<span id="cb221-11"><a href="case-study-1.html#cb221-11" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Elevated targets:&quot;</span>, <span class="fu">length</span>(sig_targets_sle_up), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Elevated targets: 100</code></pre>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="case-study-1.html#cb223-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Reduced targets:&quot;</span>, <span class="fu">length</span>(sig_targets_sle_down), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Reduced targets: 2</code></pre>
<div id="prepare-long-format-data" class="section level5 unnumbered hasAnchor">
<h5>Prepare Long Format Data<a href="case-study-1.html#prepare-long-format-data" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We clean up the covariate names in the long data frame for better boxplot readability:</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="case-study-1.html#cb225-1" tabindex="-1"></a>data.long <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ_long <span class="sc">%&gt;%</span> </span>
<span id="cb225-2"><a href="case-study-1.html#cb225-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb225-3"><a href="case-study-1.html#cb225-3" tabindex="-1"></a>    <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">case_when</span>(</span>
<span id="cb225-4"><a href="case-study-1.html#cb225-4" tabindex="-1"></a>      Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span> <span class="sc">~</span> <span class="st">&quot;SLE&quot;</span>,</span>
<span id="cb225-5"><a href="case-study-1.html#cb225-5" tabindex="-1"></a>      Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span> <span class="sc">~</span> <span class="st">&quot;RA&quot;</span>,</span>
<span id="cb225-6"><a href="case-study-1.html#cb225-6" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Normal&quot;</span></span>
<span id="cb225-7"><a href="case-study-1.html#cb225-7" tabindex="-1"></a>    ))</span></code></pre></div>
</div>
<div id="elevated-proteins-in-sle" class="section level4 hasAnchor" number="7.6.4.1">
<h4><span class="header-section-number">7.6.4.1</span> Elevated Proteins in SLE<a href="case-study-1.html#elevated-proteins-in-sle" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="case-study-1.html#cb226-1" tabindex="-1"></a>data.long <span class="sc">%&gt;%</span></span>
<span id="cb226-2"><a href="case-study-1.html#cb226-2" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_sle_up, </span>
<span id="cb226-3"><a href="case-study-1.html#cb226-3" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> sle_samples) <span class="sc">%&gt;%</span></span>
<span id="cb226-4"><a href="case-study-1.html#cb226-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb226-5"><a href="case-study-1.html#cb226-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb226-6"><a href="case-study-1.html#cb226-6" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb226-7"><a href="case-study-1.html#cb226-7" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Elevated Significant Differential Abundance Targets - SLE (n=&quot;</span>, n_sle, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb226-8"><a href="case-study-1.html#cb226-8" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb226-9"><a href="case-study-1.html#cb226-9" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb226-10"><a href="case-study-1.html#cb226-10" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),      </span>
<span id="cb226-11"><a href="case-study-1.html#cb226-11" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),                   </span>
<span id="cb226-12"><a href="case-study-1.html#cb226-12" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),                    </span>
<span id="cb226-13"><a href="case-study-1.html#cb226-13" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),                    </span>
<span id="cb226-14"><a href="case-study-1.html#cb226-14" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>), </span>
<span id="cb226-15"><a href="case-study-1.html#cb226-15" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),                     </span>
<span id="cb226-16"><a href="case-study-1.html#cb226-16" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),                      </span>
<span id="cb226-17"><a href="case-study-1.html#cb226-17" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),                    </span>
<span id="cb226-18"><a href="case-study-1.html#cb226-18" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)                     </span>
<span id="cb226-19"><a href="case-study-1.html#cb226-19" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/sle-boxplots-up-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation:</strong> Elevation of these proteins in SLE reflects multiple aspects of autoimmune pathology:</p>
<ul>
<li><strong>Immune cell activation and co-stimulation</strong>: Elevated CD40, CD80, CD83, CTLA4, and PD-1 indicate activated T cells and antigen-presenting cells</li>
<li><strong>Pro-inflammatory cytokine signaling</strong>: Increased IL-1β, IL-6R, IL-12, IL-18, and TNF family members drive systemic inflammation</li>
<li><strong>Chemokine-mediated cell trafficking</strong>: CCL and CXCL chemokines (e.g., CXCL10, CCL2) recruit immune cells to sites of inflammation</li>
<li><strong>Type I interferon response</strong>: BST2 and other interferon-stimulated genes reflect the characteristic interferon signature of lupus</li>
<li><strong>Tissue damage markers</strong>: MMP8, MMP9, and S100A12 indicate ongoing tissue degradation and neutrophil activation</li>
<li><strong>Immune checkpoint dysregulation</strong>: Elevated LAG3, PDCD1, and CTLA4 suggest attempted but inadequate negative regulation</li>
</ul>
</div>
<div id="reduced-proteins-in-sle" class="section level4 hasAnchor" number="7.6.4.2">
<h4><span class="header-section-number">7.6.4.2</span> Reduced Proteins in SLE<a href="case-study-1.html#reduced-proteins-in-sle" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="case-study-1.html#cb227-1" tabindex="-1"></a>data.long <span class="sc">%&gt;%</span></span>
<span id="cb227-2"><a href="case-study-1.html#cb227-2" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_sle_down, </span>
<span id="cb227-3"><a href="case-study-1.html#cb227-3" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> sle_samples) <span class="sc">%&gt;%</span></span>
<span id="cb227-4"><a href="case-study-1.html#cb227-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb227-5"><a href="case-study-1.html#cb227-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb227-6"><a href="case-study-1.html#cb227-6" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb227-7"><a href="case-study-1.html#cb227-7" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Reduced Significant Differential Abundance Targets - </span><span class="sc">\n</span><span class="st">SLE (n=&quot;</span>, n_sle, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb227-8"><a href="case-study-1.html#cb227-8" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb227-9"><a href="case-study-1.html#cb227-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>))</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/sle-boxplots-down-1.png" alt="" width="90%" height="90%" /></p>
<p><strong>Interpretation:</strong> Decreased abundance of IL5RA and TNFSF10 in SLE samples may indicate:</p>
<ul>
<li><strong>Altered immune regulation</strong>: TNFSF10 (TRAIL) promotes apoptosis of autoreactive cells; reduced levels may impair clearance of self-reactive lymphocytes</li>
<li><strong>Dysregulated cytokine signaling</strong>: IL5RA reduced levels suggests disrupted eosinophil and type 2 immune responses</li>
<li><strong>Chronic inflammation effects</strong>: Prolonged autoimmune activation can suppress certain immune regulatory pathways</li>
<li><strong>Loss of tolerance mechanisms</strong>: Reduced pro-apoptotic signaling may contribute to persistence of autoreactive immune cells</li>
</ul>
</div>
</div>
</div>
<div id="rheumatoid-arthritis-ra-analysis" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Rheumatoid Arthritis (RA) Analysis<a href="case-study-1.html#rheumatoid-arthritis-ra-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="background-1" class="section level3 hasAnchor" number="7.7.1">
<h3><span class="header-section-number">7.7.1</span> Background<a href="case-study-1.html#background-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Rheumatoid Arthritis is a chronic inflammatory disorder that primarily affects synovial joints but can have systemic manifestations. The pathophysiology involves:</p>
<ul>
<li>Autoimmune targeting of joint tissues</li>
<li>Synovial inflammation and proliferation</li>
<li>Cartilage and bone destruction</li>
<li>Systemic inflammation</li>
</ul>
<p>The proteomic profile of RA often shows markers of:</p>
<ul>
<li>Joint inflammation and damage</li>
<li>Systemic inflammatory response</li>
<li>Matrix degradation</li>
<li>Immune activation</li>
</ul>
</div>
<div id="volcano-plot-1" class="section level3 hasAnchor" number="7.7.2">
<h3><span class="header-section-number">7.7.2</span> Volcano Plot<a href="case-study-1.html#volcano-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="case-study-1.html#cb228-1" tabindex="-1"></a><span class="fu">volcanoPlot</span>(<span class="at">coefs =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaserheumatoidArthritis_coef,</span>
<span id="cb228-2"><a href="case-study-1.html#cb228-2" tabindex="-1"></a>            <span class="at">p_vals =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaserheumatoidArthritis_pval_FDR,</span>
<span id="cb228-3"><a href="case-study-1.html#cb228-3" tabindex="-1"></a>            <span class="at">target_labels =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>target,</span>
<span id="cb228-4"><a href="case-study-1.html#cb228-4" tabindex="-1"></a>            <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Rheumatoid Arthritis (RA) (n=&quot;</span>, n_ra, <span class="st">&quot;) </span><span class="sc">\n</span><span class="st"> vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>))</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/ra-volcano-1.png" alt="" width="100%" height="100%" /></p>
<p>Compare this volcano plot to the SLE plot above. Are there:</p>
<ul>
<li>Proteins significantly changed in both conditions (shared inflammatory pathways)?</li>
<li>Disease-specific signatures unique to RA or SLE?</li>
<li>Similar magnitude of changes between the two diseases?</li>
</ul>
</div>
<div id="visualizations-1" class="section level3 hasAnchor" number="7.7.3">
<h3><span class="header-section-number">7.7.3</span> Visualizations<a href="case-study-1.html#visualizations-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="case-study-1.html#cb229-1" tabindex="-1"></a><span class="co"># Select RA and normal samples</span></span>
<span id="cb229-2"><a href="case-study-1.html#cb229-2" tabindex="-1"></a>ra_samples <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span></span>
<span id="cb229-3"><a href="case-study-1.html#cb229-3" tabindex="-1"></a>  <span class="fu">filter</span>(Disease <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>, <span class="st">&quot;rheumatoidArthritis&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb229-4"><a href="case-study-1.html#cb229-4" tabindex="-1"></a>  <span class="fu">pull</span>(sampleName)</span>
<span id="cb229-5"><a href="case-study-1.html#cb229-5" tabindex="-1"></a></span>
<span id="cb229-6"><a href="case-study-1.html#cb229-6" tabindex="-1"></a><span class="co"># Identify significantly differential proteins</span></span>
<span id="cb229-7"><a href="case-study-1.html#cb229-7" tabindex="-1"></a>sig_targets_ra <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb229-8"><a href="case-study-1.html#cb229-8" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb229-9"><a href="case-study-1.html#cb229-9" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb229-10"><a href="case-study-1.html#cb229-10" tabindex="-1"></a></span>
<span id="cb229-11"><a href="case-study-1.html#cb229-11" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of significant targets in RA:&quot;</span>, <span class="fu">length</span>(sig_targets_ra), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Number of significant targets in RA: 96</code></pre>
<div id="ra-specific-heatmap" class="section level4 hasAnchor" number="7.7.3.1">
<h4><span class="header-section-number">7.7.3.1</span> RA-Specific Heatmap<a href="case-study-1.html#ra-specific-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="case-study-1.html#cb231-1" tabindex="-1"></a>h_ra <span class="ot">&lt;-</span> <span class="fu">generate_heatmap</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb231-2"><a href="case-study-1.html#cb231-2" tabindex="-1"></a>                         <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb231-3"><a href="case-study-1.html#cb231-3" tabindex="-1"></a>                         <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb231-4"><a href="case-study-1.html#cb231-4" tabindex="-1"></a>                         <span class="at">sample_subset =</span> ra_samples, </span>
<span id="cb231-5"><a href="case-study-1.html#cb231-5" tabindex="-1"></a>                         <span class="at">target_subset =</span> sig_targets_ra,</span>
<span id="cb231-6"><a href="case-study-1.html#cb231-6" tabindex="-1"></a>                         <span class="at">annotate_sample_by =</span> <span class="fu">c</span>(<span class="st">&quot;Disease&quot;</span>, <span class="st">&quot;Sample Matrix&quot;</span>, <span class="st">&quot;Lab&quot;</span>),</span>
<span id="cb231-7"><a href="case-study-1.html#cb231-7" tabindex="-1"></a>                         <span class="at">row_fontsize =</span> <span class="dv">6</span>,</span>
<span id="cb231-8"><a href="case-study-1.html#cb231-8" tabindex="-1"></a>                         <span class="at">col_fontsize =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/ra-heatmap-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="ra-specific-pca" class="section level4 hasAnchor" number="7.7.3.2">
<h4><span class="header-section-number">7.7.3.2</span> RA-Specific PCA<a href="case-study-1.html#ra-specific-pca" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="case-study-1.html#cb232-1" tabindex="-1"></a>p_ra <span class="ot">&lt;-</span> <span class="fu">generate_pca</span>(<span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb232-2"><a href="case-study-1.html#cb232-2" tabindex="-1"></a>                     <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb232-3"><a href="case-study-1.html#cb232-3" tabindex="-1"></a>                     <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb232-4"><a href="case-study-1.html#cb232-4" tabindex="-1"></a>                     <span class="at">sample_subset =</span> ra_samples, </span>
<span id="cb232-5"><a href="case-study-1.html#cb232-5" tabindex="-1"></a>                     <span class="at">target_subset =</span> sig_targets_ra,</span>
<span id="cb232-6"><a href="case-study-1.html#cb232-6" tabindex="-1"></a>                     <span class="at">annotate_sample_by =</span> <span class="st">&quot;Disease&quot;</span>)</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/ra-pca-1.png" alt="" width="90%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="directional-analysis-1" class="section level3 hasAnchor" number="7.7.4">
<h3><span class="header-section-number">7.7.4</span> Directional Analysis<a href="case-study-1.html#directional-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="case-study-1.html#cb233-1" tabindex="-1"></a><span class="co"># Elevated proteins (higher in RA)</span></span>
<span id="cb233-2"><a href="case-study-1.html#cb233-2" tabindex="-1"></a>sig_targets_ra_up <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb233-3"><a href="case-study-1.html#cb233-3" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, </span>
<span id="cb233-4"><a href="case-study-1.html#cb233-4" tabindex="-1"></a>         DiseaserheumatoidArthritis_coef <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb233-5"><a href="case-study-1.html#cb233-5" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb233-6"><a href="case-study-1.html#cb233-6" tabindex="-1"></a></span>
<span id="cb233-7"><a href="case-study-1.html#cb233-7" tabindex="-1"></a><span class="co"># Reduced proteins (lower in RA)</span></span>
<span id="cb233-8"><a href="case-study-1.html#cb233-8" tabindex="-1"></a>sig_targets_ra_down <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb233-9"><a href="case-study-1.html#cb233-9" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, </span>
<span id="cb233-10"><a href="case-study-1.html#cb233-10" tabindex="-1"></a>         DiseaserheumatoidArthritis_coef <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb233-11"><a href="case-study-1.html#cb233-11" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb233-12"><a href="case-study-1.html#cb233-12" tabindex="-1"></a></span>
<span id="cb233-13"><a href="case-study-1.html#cb233-13" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Elevated targets:&quot;</span>, <span class="fu">length</span>(sig_targets_ra_up), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Elevated targets: 95</code></pre>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="case-study-1.html#cb235-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Reduced targets:&quot;</span>, <span class="fu">length</span>(sig_targets_ra_down), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Reduced targets: 1</code></pre>
<div id="elevated-proteins-in-ra" class="section level4 hasAnchor" number="7.7.4.1">
<h4><span class="header-section-number">7.7.4.1</span> Elevated Proteins in RA<a href="case-study-1.html#elevated-proteins-in-ra" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="case-study-1.html#cb237-1" tabindex="-1"></a>data.long <span class="sc">%&gt;%</span></span>
<span id="cb237-2"><a href="case-study-1.html#cb237-2" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_ra_up, </span>
<span id="cb237-3"><a href="case-study-1.html#cb237-3" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> ra_samples) <span class="sc">%&gt;%</span></span>
<span id="cb237-4"><a href="case-study-1.html#cb237-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb237-5"><a href="case-study-1.html#cb237-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb237-6"><a href="case-study-1.html#cb237-6" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb237-7"><a href="case-study-1.html#cb237-7" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Elevated Significant Differential Abundance Targets - RA (n=&quot;</span>, n_ra, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb237-8"><a href="case-study-1.html#cb237-8" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb237-9"><a href="case-study-1.html#cb237-9" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb237-10"><a href="case-study-1.html#cb237-10" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),      </span>
<span id="cb237-11"><a href="case-study-1.html#cb237-11" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),                  </span>
<span id="cb237-12"><a href="case-study-1.html#cb237-12" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),                   </span>
<span id="cb237-13"><a href="case-study-1.html#cb237-13" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),                   </span>
<span id="cb237-14"><a href="case-study-1.html#cb237-14" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>), </span>
<span id="cb237-15"><a href="case-study-1.html#cb237-15" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),                    </span>
<span id="cb237-16"><a href="case-study-1.html#cb237-16" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),                     </span>
<span id="cb237-17"><a href="case-study-1.html#cb237-17" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),                   </span>
<span id="cb237-18"><a href="case-study-1.html#cb237-18" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)                    </span>
<span id="cb237-19"><a href="case-study-1.html#cb237-19" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/ra-boxplots-up-1.png" alt="" width="100%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation:</strong> Elevated proteins in RA reflect multiple aspects of autoimmune joint inflammation:</p>
<ul>
<li><strong>Immune cell activation and co-stimulation</strong>: Elevated CD40, CD80, CD83, CTLA4, and PD-1 indicate activated T cells and antigen-presenting cells</li>
<li><strong>Pro-inflammatory cytokine signaling</strong>: Increased IL-1β, TNF, IL-18, IL-6ST, and TNF receptor family members drive chronic inflammation</li>
<li><strong>Chemokine-mediated cell trafficking</strong>: CCL and CXCL chemokines (e.g., CCL2, CXCL13) recruit immune cells to inflamed synovial tissue</li>
<li><strong>Joint destruction markers</strong>: Matrix metalloproteinases (MMP8, MMP9, MMP12) and cartilage degradation protein CHI3L1 indicate ongoing tissue damage</li>
<li><strong>Neutrophil activation</strong>: S100A12, MPO, and LCN2 reflect neutrophil infiltration characteristic of RA synovitis</li>
<li><strong>Angiogenesis and tissue remodeling</strong>: VEGFC, HGF, and growth factors support pannus formation and joint destruction</li>
<li><strong>Immune checkpoint dysregulation</strong>: Elevated LAG3, PDCD1, CTLA4, and PD-L2 suggest attempted but inadequate immune regulation</li>
</ul>
</div>
<div id="reduced-proteins-in-ra" class="section level4 hasAnchor" number="7.7.4.2">
<h4><span class="header-section-number">7.7.4.2</span> Reduced Proteins in RA<a href="case-study-1.html#reduced-proteins-in-ra" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="case-study-1.html#cb238-1" tabindex="-1"></a>data.long <span class="sc">%&gt;%</span></span>
<span id="cb238-2"><a href="case-study-1.html#cb238-2" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_ra_down, </span>
<span id="cb238-3"><a href="case-study-1.html#cb238-3" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> ra_samples) <span class="sc">%&gt;%</span></span>
<span id="cb238-4"><a href="case-study-1.html#cb238-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb238-5"><a href="case-study-1.html#cb238-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb238-6"><a href="case-study-1.html#cb238-6" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb238-7"><a href="case-study-1.html#cb238-7" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Reduced Significant Differential Abundance Targets - </span><span class="sc">\n</span><span class="st">RA (n=&quot;</span>, n_ra, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb238-8"><a href="case-study-1.html#cb238-8" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb238-9"><a href="case-study-1.html#cb238-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>))</span></code></pre></div>
<p><img src="NULISAseqR-guide_files/figure-html/ra-boxplots-down-1.png" alt="" width="80%" style="display: block; margin: auto;" /></p>
<p><strong>Interpretation:</strong> Reduced TNFSF10 (TRAIL) abundance in RA may reflect:</p>
<ul>
<li><strong>Impaired apoptosis regulation</strong>: TRAIL promotes death of activated immune cells; reduced levels may allow persistence of autoreactive lymphocytes in synovial tissue</li>
<li><strong>Loss of immune tolerance</strong>: Decreased pro-apoptotic signaling contributes to accumulation of inflammatory cells in joints</li>
<li><strong>Synovial hyperplasia</strong>: Reduced TRAIL may fail to limit proliferation of synovial fibroblasts, contributing to pannus formation</li>
</ul>
</div>
</div>
</div>
<div id="comparing-disease-signatures" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> Comparing Disease Signatures<a href="case-study-1.html#comparing-disease-signatures" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="overlap-analysis" class="section level3 hasAnchor" number="7.8.1">
<h3><span class="header-section-number">7.8.1</span> Overlap Analysis<a href="case-study-1.html#overlap-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To understand shared versus disease-specific biology, we can examine the overlap of significant proteins:</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="case-study-1.html#cb239-1" tabindex="-1"></a><span class="co"># Proteins significant in both diseases</span></span>
<span id="cb239-2"><a href="case-study-1.html#cb239-2" tabindex="-1"></a>shared_targets <span class="ot">&lt;-</span> <span class="fu">intersect</span>(sig_targets_sle, sig_targets_ra)</span>
<span id="cb239-3"><a href="case-study-1.html#cb239-3" tabindex="-1"></a></span>
<span id="cb239-4"><a href="case-study-1.html#cb239-4" tabindex="-1"></a><span class="co"># Disease-specific proteins</span></span>
<span id="cb239-5"><a href="case-study-1.html#cb239-5" tabindex="-1"></a>sle_specific <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(sig_targets_sle, sig_targets_ra)</span>
<span id="cb239-6"><a href="case-study-1.html#cb239-6" tabindex="-1"></a>ra_specific <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(sig_targets_ra, sig_targets_sle)</span>
<span id="cb239-7"><a href="case-study-1.html#cb239-7" tabindex="-1"></a></span>
<span id="cb239-8"><a href="case-study-1.html#cb239-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Shared significant targets:&quot;</span>, <span class="fu">length</span>(shared_targets), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; Shared significant targets: 73</code></pre>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="case-study-1.html#cb241-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;SLE-specific targets:&quot;</span>, <span class="fu">length</span>(sle_specific), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; SLE-specific targets: 29</code></pre>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="case-study-1.html#cb243-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;RA-specific targets:&quot;</span>, <span class="fu">length</span>(ra_specific), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>#&gt; RA-specific targets: 23</code></pre>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="case-study-1.html#cb245-1" tabindex="-1"></a><span class="co"># Display shared and specific targets</span></span>
<span id="cb245-2"><a href="case-study-1.html#cb245-2" tabindex="-1"></a><span class="co"># Create a data frame with comma-separated strings and counts</span></span>
<span id="cb245-3"><a href="case-study-1.html#cb245-3" tabindex="-1"></a>overlap_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb245-4"><a href="case-study-1.html#cb245-4" tabindex="-1"></a>  <span class="st">&quot;Category&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Shared (SLE and RA)&quot;</span>, <span class="st">&quot;SLE-Specific&quot;</span>, <span class="st">&quot;RA-Specific&quot;</span>),</span>
<span id="cb245-5"><a href="case-study-1.html#cb245-5" tabindex="-1"></a>  <span class="st">&quot;Count&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">length</span>(shared_targets), <span class="fu">length</span>(sle_specific), <span class="fu">length</span>(ra_specific)),</span>
<span id="cb245-6"><a href="case-study-1.html#cb245-6" tabindex="-1"></a>  <span class="st">&quot;Targets&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb245-7"><a href="case-study-1.html#cb245-7" tabindex="-1"></a>    <span class="fu">paste</span>(shared_targets, <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb245-8"><a href="case-study-1.html#cb245-8" tabindex="-1"></a>    <span class="fu">paste</span>(sle_specific, <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb245-9"><a href="case-study-1.html#cb245-9" tabindex="-1"></a>    <span class="fu">paste</span>(ra_specific, <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>)</span>
<span id="cb245-10"><a href="case-study-1.html#cb245-10" tabindex="-1"></a>  ),</span>
<span id="cb245-11"><a href="case-study-1.html#cb245-11" tabindex="-1"></a>  <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb245-12"><a href="case-study-1.html#cb245-12" tabindex="-1"></a>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-82ede84a1dd987845a88" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-82ede84a1dd987845a88">{"x":{"filter":"none","vertical":false,"caption":"<caption>Comparison of Significant Protein Targets Across Diseases<\/caption>","data":[["Shared (SLE and RA)","SLE-Specific","RA-Specific"],[73,29,23],["AREG, CCL1, CCL13, CCL2, CCL21, CCL22, CCL26, CCL27, CCL7, CD27, CD274, CD4, CD40, CD46, CD80, CD83, CSF1, CSF1R, CSF3R, CTLA4, CXCL11, CXCL13, CXCL6, FASLG, FURIN, GZMA, GZMB, HGF, IL10RB, IL12B, IL15RA, IL16, IL17RA, IL18, IL18BP, IL18R1, IL1B, IL1RN, IL2RA, KLRK1, LCN2, LGALS9, LIF, LILRB2, MMP8, MMP9, MPO, NAMPT, NCR1, OSM, OSMR, PDCD1, PDGFB, PTX3, S100A12, THBS2, TNFRSF11A, TNFRSF13C, TNFRSF14, TNFRSF18, TNFRSF1A, TNFRSF21, TNFRSF4, TNFRSF8, TNFRSF9, TNFSF10, TNFSF13, TNFSF18, TNFSF4, TNFSF8, TREM1, TREM2, TSLP","ANGPT2, BDNF, BST2, C1QA, CCL17, CCL19, CD40LG, CST7, CX3CL1, CXCL10, FGF23, HAVCR1, IL12p70, IL2RB, IL4, IL5RA, IL6R, IRAK4, LAG3, MICB, MIF, PDGFA, TAFA5, TGFB1, TIMP2, TNFSF12, VEGFA, VEGFD, VSTM1","BMP7, CCL11, CHI3L1, CXCL16, FTH1, GRN, IL13RA2, IL1RL1, IL36G, IL3RA, IL6ST, LTA, MMP12, PDCD1LG2, PGF, SDC1, SELE, SLAMF1, TEK, TNF, TNFRSF11B, TNFRSF1B, VEGFC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Category<\/th>\n      <th>Count<\/th>\n      <th>Targets<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","pageLength":3,"autoWidth":true,"columnDefs":[{"width":"15%","targets":0},{"width":"10%","targets":1},{"width":"75%","targets":2},{"className":"dt-right","targets":1},{"name":"Category","targets":0},{"name":"Count","targets":1},{"name":"Targets","targets":2}],"order":[],"orderClasses":false,"lengthMenu":[3,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><strong>Biological Interpretation:</strong></p>
<ul>
<li><strong>Shared targets</strong> highlight core autoimmune inflammatory mechanisms common to both SLE and RA, including cytokine signaling, immune cell activation, and tissue remodeling pathways</li>
<li><strong>SLE-specific targets</strong> may reflect systemic immune dysregulation, interferon signaling, and multi-organ involvement characteristic of lupus</li>
<li><strong>RA-specific targets</strong> likely represent joint-specific processes such as synovial inflammation, cartilage degradation, and localized tissue destruction</li>
<li><strong>Directional concordance</strong>: Most shared proteins show directional concordance (elevated in both diseases), indicating similar inflammatory activation patterns</li>
</ul>
</div>
<div id="differential-abundance-summary-table" class="section level3 hasAnchor" number="7.8.2">
<h3><span class="header-section-number">7.8.2</span> Differential Abundance Summary Table<a href="case-study-1.html#differential-abundance-summary-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="case-study-1.html#cb246-1" tabindex="-1"></a>summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb246-2"><a href="case-study-1.html#cb246-2" tabindex="-1"></a>  <span class="at">Condition =</span> <span class="fu">c</span>(<span class="st">&quot;SLE vs Normal&quot;</span>, <span class="st">&quot;RA vs Normal&quot;</span>),</span>
<span id="cb246-3"><a href="case-study-1.html#cb246-3" tabindex="-1"></a>  <span class="at">Total_Significant =</span> <span class="fu">c</span>(<span class="fu">length</span>(sig_targets_sle), <span class="fu">length</span>(sig_targets_ra)),</span>
<span id="cb246-4"><a href="case-study-1.html#cb246-4" tabindex="-1"></a>  <span class="at">Elevated =</span> <span class="fu">c</span>(<span class="fu">length</span>(sig_targets_sle_up), <span class="fu">length</span>(sig_targets_ra_up)),</span>
<span id="cb246-5"><a href="case-study-1.html#cb246-5" tabindex="-1"></a>  <span class="at">Reduced =</span> <span class="fu">c</span>(<span class="fu">length</span>(sig_targets_sle_down), <span class="fu">length</span>(sig_targets_ra_down))</span>
<span id="cb246-6"><a href="case-study-1.html#cb246-6" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="case-study-1.html#cb247-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary, </span>
<span id="cb247-2"><a href="case-study-1.html#cb247-2" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;Summary of Differential Abundance Results&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:1">Table 7.1: </span>Summary of Differential Abundance Results</caption>
<thead>
<tr class="header">
<th align="left">Condition</th>
<th align="right">Total_Significant</th>
<th align="right">Elevated</th>
<th align="right">Reduced</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SLE vs Normal</td>
<td align="right">102</td>
<td align="right">100</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">RA vs Normal</td>
<td align="right">96</td>
<td align="right">95</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="next-steps" class="section level2 hasAnchor" number="7.9">
<h2><span class="header-section-number">7.9</span> Next Steps<a href="case-study-1.html#next-steps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After identifying differentially abundant proteins, typical follow-up analyses include:</p>
<ul>
<li><strong>Pathway enrichment analysis</strong>: Identify biological pathways over-represented in significant proteins</li>
<li><strong>Protein network analysis</strong>: Understand functional relationships between differential proteins</li>
<li><strong>Biomarker evaluation</strong>: Assess individual proteins or combinations for diagnostic potential</li>
<li><strong>Correlation with clinical variables</strong>: Link protein changes to disease severity or outcomes</li>
<li><strong>Validation studies</strong>: Confirm findings in independent cohorts or using orthogonal methods</li>
</ul>
</div>
<div id="complete-workflow-example-code" class="section level2 hasAnchor" number="7.10">
<h2><span class="header-section-number">7.10</span> Complete Workflow Example Code<a href="case-study-1.html#complete-workflow-example-code" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="case-study-1.html#cb248-1" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-2"><a href="case-study-1.html#cb248-2" tabindex="-1"></a><span class="co"># COMPLETE DIFFERENTIAL ABUNDANCE WORKFLOW: SLE and RA Analysis</span></span>
<span id="cb248-3"><a href="case-study-1.html#cb248-3" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-4"><a href="case-study-1.html#cb248-4" tabindex="-1"></a></span>
<span id="cb248-5"><a href="case-study-1.html#cb248-5" tabindex="-1"></a><span class="co"># 1. Load libraries and import data</span></span>
<span id="cb248-6"><a href="case-study-1.html#cb248-6" tabindex="-1"></a><span class="fu">library</span>(NULISAseqR)</span>
<span id="cb248-7"><a href="case-study-1.html#cb248-7" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb248-8"><a href="case-study-1.html#cb248-8" tabindex="-1"></a></span>
<span id="cb248-9"><a href="case-study-1.html#cb248-9" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package =</span> <span class="st">&quot;NULISAseqR&quot;</span>)</span>
<span id="cb248-10"><a href="case-study-1.html#cb248-10" tabindex="-1"></a>data.a <span class="ot">&lt;-</span> <span class="fu">importNULISAseq</span>(<span class="at">files =</span> <span class="fu">file.path</span>(data_dir,<span class="st">&quot;Inflammation_250_RQ_demo_2025-05-20-Lab-A.xml&quot;</span>))</span>
<span id="cb248-11"><a href="case-study-1.html#cb248-11" tabindex="-1"></a></span>
<span id="cb248-12"><a href="case-study-1.html#cb248-12" tabindex="-1"></a><span class="co"># 2. Prepare metadata</span></span>
<span id="cb248-13"><a href="case-study-1.html#cb248-13" tabindex="-1"></a>metadata.a <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>samples <span class="sc">%&gt;%</span> </span>
<span id="cb248-14"><a href="case-study-1.html#cb248-14" tabindex="-1"></a>  <span class="fu">filter</span>(sampleType <span class="sc">==</span> <span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb248-15"><a href="case-study-1.html#cb248-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Disease =</span> <span class="fu">relevel</span>(Disease, <span class="at">ref =</span> <span class="st">&quot;normal&quot;</span>))</span>
<span id="cb248-16"><a href="case-study-1.html#cb248-16" tabindex="-1"></a></span>
<span id="cb248-17"><a href="case-study-1.html#cb248-17" tabindex="-1"></a>metadata_rename <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span> </span>
<span id="cb248-18"><a href="case-study-1.html#cb248-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Disease =</span> <span class="fu">case_when</span>(</span>
<span id="cb248-19"><a href="case-study-1.html#cb248-19" tabindex="-1"></a>    Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span> <span class="sc">~</span> <span class="st">&quot;SLE&quot;</span>,</span>
<span id="cb248-20"><a href="case-study-1.html#cb248-20" tabindex="-1"></a>    Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span> <span class="sc">~</span> <span class="st">&quot;RA&quot;</span>,</span>
<span id="cb248-21"><a href="case-study-1.html#cb248-21" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Normal&quot;</span></span>
<span id="cb248-22"><a href="case-study-1.html#cb248-22" tabindex="-1"></a>  ),</span>
<span id="cb248-23"><a href="case-study-1.html#cb248-23" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sample Matrix</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">tolower</span>(SAMPLE_MATRIX))</span>
<span id="cb248-24"><a href="case-study-1.html#cb248-24" tabindex="-1"></a></span>
<span id="cb248-25"><a href="case-study-1.html#cb248-25" tabindex="-1"></a><span class="co"># Sample counts per group</span></span>
<span id="cb248-26"><a href="case-study-1.html#cb248-26" tabindex="-1"></a>n_normal <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>)</span>
<span id="cb248-27"><a href="case-study-1.html#cb248-27" tabindex="-1"></a>n_sle <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span>)</span>
<span id="cb248-28"><a href="case-study-1.html#cb248-28" tabindex="-1"></a>n_ra <span class="ot">&lt;-</span> <span class="fu">sum</span>(metadata.a<span class="sc">$</span>Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span>)</span>
<span id="cb248-29"><a href="case-study-1.html#cb248-29" tabindex="-1"></a></span>
<span id="cb248-30"><a href="case-study-1.html#cb248-30" tabindex="-1"></a><span class="co"># Display sample distribution</span></span>
<span id="cb248-31"><a href="case-study-1.html#cb248-31" tabindex="-1"></a><span class="fu">table</span>(metadata.a<span class="sc">$</span>Disease)</span>
<span id="cb248-32"><a href="case-study-1.html#cb248-32" tabindex="-1"></a></span>
<span id="cb248-33"><a href="case-study-1.html#cb248-33" tabindex="-1"></a><span class="co"># 3. Target detectability filtering</span></span>
<span id="cb248-34"><a href="case-study-1.html#cb248-34" tabindex="-1"></a>detectability_threshold <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb248-35"><a href="case-study-1.html#cb248-35" tabindex="-1"></a>targets_passed <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>detectability <span class="sc">%&gt;%</span> </span>
<span id="cb248-36"><a href="case-study-1.html#cb248-36" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">plasma (n = 43)</span><span class="st">`</span> <span class="sc">&gt;</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb248-37"><a href="case-study-1.html#cb248-37" tabindex="-1"></a>  <span class="fu">pull</span>(Target)</span>
<span id="cb248-38"><a href="case-study-1.html#cb248-38" tabindex="-1"></a></span>
<span id="cb248-39"><a href="case-study-1.html#cb248-39" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of targets passing QC:&quot;</span>, <span class="fu">length</span>(targets_passed), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-40"><a href="case-study-1.html#cb248-40" tabindex="-1"></a></span>
<span id="cb248-41"><a href="case-study-1.html#cb248-41" tabindex="-1"></a><span class="co"># 4. Run differential abundance analysis</span></span>
<span id="cb248-42"><a href="case-study-1.html#cb248-42" tabindex="-1"></a>lmTest_da <span class="ot">&lt;-</span> <span class="fu">lmNULISAseq</span>(</span>
<span id="cb248-43"><a href="case-study-1.html#cb248-43" tabindex="-1"></a>  <span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[targets_passed, metadata.a<span class="sc">$</span>sampleName],</span>
<span id="cb248-44"><a href="case-study-1.html#cb248-44" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata.a,</span>
<span id="cb248-45"><a href="case-study-1.html#cb248-45" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb248-46"><a href="case-study-1.html#cb248-46" tabindex="-1"></a>  <span class="at">modelFormula =</span> <span class="st">&quot;Disease&quot;</span></span>
<span id="cb248-47"><a href="case-study-1.html#cb248-47" tabindex="-1"></a>)</span>
<span id="cb248-48"><a href="case-study-1.html#cb248-48" tabindex="-1"></a></span>
<span id="cb248-49"><a href="case-study-1.html#cb248-49" tabindex="-1"></a><span class="co"># Create output directories</span></span>
<span id="cb248-50"><a href="case-study-1.html#cb248-50" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;figures&quot;</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb248-51"><a href="case-study-1.html#cb248-51" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;results&quot;</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb248-52"><a href="case-study-1.html#cb248-52" tabindex="-1"></a></span>
<span id="cb248-53"><a href="case-study-1.html#cb248-53" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-54"><a href="case-study-1.html#cb248-54" tabindex="-1"></a><span class="co"># SLE ANALYSIS</span></span>
<span id="cb248-55"><a href="case-study-1.html#cb248-55" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-56"><a href="case-study-1.html#cb248-56" tabindex="-1"></a></span>
<span id="cb248-57"><a href="case-study-1.html#cb248-57" tabindex="-1"></a><span class="co"># 5a. Filter significant SLE results</span></span>
<span id="cb248-58"><a href="case-study-1.html#cb248-58" tabindex="-1"></a>sig_targets_sle <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-59"><a href="case-study-1.html#cb248-59" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-60"><a href="case-study-1.html#cb248-60" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-61"><a href="case-study-1.html#cb248-61" tabindex="-1"></a></span>
<span id="cb248-62"><a href="case-study-1.html#cb248-62" tabindex="-1"></a>sig_targets_sle_up <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-63"><a href="case-study-1.html#cb248-63" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, DiseaselupusSle_coef <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-64"><a href="case-study-1.html#cb248-64" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-65"><a href="case-study-1.html#cb248-65" tabindex="-1"></a></span>
<span id="cb248-66"><a href="case-study-1.html#cb248-66" tabindex="-1"></a>sig_targets_sle_down <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-67"><a href="case-study-1.html#cb248-67" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaselupusSle_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, DiseaselupusSle_coef <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-68"><a href="case-study-1.html#cb248-68" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-69"><a href="case-study-1.html#cb248-69" tabindex="-1"></a></span>
<span id="cb248-70"><a href="case-study-1.html#cb248-70" tabindex="-1"></a><span class="co"># 6a. Get SLE and normal samples</span></span>
<span id="cb248-71"><a href="case-study-1.html#cb248-71" tabindex="-1"></a>sle_samples <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span></span>
<span id="cb248-72"><a href="case-study-1.html#cb248-72" tabindex="-1"></a>  <span class="fu">filter</span>(Disease <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>, <span class="st">&quot;lupusSle&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb248-73"><a href="case-study-1.html#cb248-73" tabindex="-1"></a>  <span class="fu">pull</span>(sampleName)</span>
<span id="cb248-74"><a href="case-study-1.html#cb248-74" tabindex="-1"></a></span>
<span id="cb248-75"><a href="case-study-1.html#cb248-75" tabindex="-1"></a><span class="co"># 7a. Create SLE volcano plot and save as PDF</span></span>
<span id="cb248-76"><a href="case-study-1.html#cb248-76" tabindex="-1"></a><span class="fu">volcanoPlot</span>(</span>
<span id="cb248-77"><a href="case-study-1.html#cb248-77" tabindex="-1"></a>  <span class="at">coefs =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaselupusSle_coef,</span>
<span id="cb248-78"><a href="case-study-1.html#cb248-78" tabindex="-1"></a>  <span class="at">p_vals =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaselupusSle_pval_FDR,</span>
<span id="cb248-79"><a href="case-study-1.html#cb248-79" tabindex="-1"></a>  <span class="at">target_labels =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>target,</span>
<span id="cb248-80"><a href="case-study-1.html#cb248-80" tabindex="-1"></a>  <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Systemic Lupus Erythematosus (SLE) (n=&quot;</span>, n_sle, <span class="st">&quot;) </span><span class="sc">\n</span><span class="st"> vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-81"><a href="case-study-1.html#cb248-81" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;volcano_plot_sle_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-82"><a href="case-study-1.html#cb248-82" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-83"><a href="case-study-1.html#cb248-83" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">6</span>,</span>
<span id="cb248-84"><a href="case-study-1.html#cb248-84" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">5</span></span>
<span id="cb248-85"><a href="case-study-1.html#cb248-85" tabindex="-1"></a>)</span>
<span id="cb248-86"><a href="case-study-1.html#cb248-86" tabindex="-1"></a></span>
<span id="cb248-87"><a href="case-study-1.html#cb248-87" tabindex="-1"></a><span class="co"># 8a. Create SLE heatmap and save as PDF</span></span>
<span id="cb248-88"><a href="case-study-1.html#cb248-88" tabindex="-1"></a>h_sle <span class="ot">&lt;-</span> <span class="fu">generate_heatmap</span>(</span>
<span id="cb248-89"><a href="case-study-1.html#cb248-89" tabindex="-1"></a>  <span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb248-90"><a href="case-study-1.html#cb248-90" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb248-91"><a href="case-study-1.html#cb248-91" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb248-92"><a href="case-study-1.html#cb248-92" tabindex="-1"></a>  <span class="at">sample_subset =</span> sle_samples,</span>
<span id="cb248-93"><a href="case-study-1.html#cb248-93" tabindex="-1"></a>  <span class="at">target_subset =</span> sig_targets_sle,</span>
<span id="cb248-94"><a href="case-study-1.html#cb248-94" tabindex="-1"></a>  <span class="at">annotate_sample_by =</span> <span class="fu">c</span>(<span class="st">&quot;Disease&quot;</span>, <span class="st">&quot;Sample Matrix&quot;</span>, <span class="st">&quot;Lab&quot;</span>),</span>
<span id="cb248-95"><a href="case-study-1.html#cb248-95" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-96"><a href="case-study-1.html#cb248-96" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;heatmap_sle_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-97"><a href="case-study-1.html#cb248-97" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">8</span>,</span>
<span id="cb248-98"><a href="case-study-1.html#cb248-98" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">6</span></span>
<span id="cb248-99"><a href="case-study-1.html#cb248-99" tabindex="-1"></a>)</span>
<span id="cb248-100"><a href="case-study-1.html#cb248-100" tabindex="-1"></a></span>
<span id="cb248-101"><a href="case-study-1.html#cb248-101" tabindex="-1"></a><span class="co"># 9a. Create SLE PCA and save as PDF</span></span>
<span id="cb248-102"><a href="case-study-1.html#cb248-102" tabindex="-1"></a>p_sle <span class="ot">&lt;-</span> <span class="fu">generate_pca</span>(</span>
<span id="cb248-103"><a href="case-study-1.html#cb248-103" tabindex="-1"></a>  <span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb248-104"><a href="case-study-1.html#cb248-104" tabindex="-1"></a>  <span class="at">plot_title =</span> <span class="st">&quot;PCA: SLE vs Normal</span><span class="sc">\n</span><span class="st">Significant DA Targets&quot;</span>,</span>
<span id="cb248-105"><a href="case-study-1.html#cb248-105" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb248-106"><a href="case-study-1.html#cb248-106" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb248-107"><a href="case-study-1.html#cb248-107" tabindex="-1"></a>  <span class="at">sample_subset =</span> sle_samples,</span>
<span id="cb248-108"><a href="case-study-1.html#cb248-108" tabindex="-1"></a>  <span class="at">target_subset =</span> sig_targets_sle,</span>
<span id="cb248-109"><a href="case-study-1.html#cb248-109" tabindex="-1"></a>  <span class="at">annotate_sample_by =</span> <span class="st">&quot;Disease&quot;</span>,</span>
<span id="cb248-110"><a href="case-study-1.html#cb248-110" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-111"><a href="case-study-1.html#cb248-111" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;pca_plot_sle_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-112"><a href="case-study-1.html#cb248-112" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">5</span>,</span>
<span id="cb248-113"><a href="case-study-1.html#cb248-113" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">4</span></span>
<span id="cb248-114"><a href="case-study-1.html#cb248-114" tabindex="-1"></a>)</span>
<span id="cb248-115"><a href="case-study-1.html#cb248-115" tabindex="-1"></a></span>
<span id="cb248-116"><a href="case-study-1.html#cb248-116" tabindex="-1"></a><span class="co"># 10a. Prepare long format data for boxplots</span></span>
<span id="cb248-117"><a href="case-study-1.html#cb248-117" tabindex="-1"></a>data.long <span class="ot">&lt;-</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ_long <span class="sc">%&gt;%</span> </span>
<span id="cb248-118"><a href="case-study-1.html#cb248-118" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb248-119"><a href="case-study-1.html#cb248-119" tabindex="-1"></a>    <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">case_when</span>(</span>
<span id="cb248-120"><a href="case-study-1.html#cb248-120" tabindex="-1"></a>      Disease <span class="sc">==</span> <span class="st">&quot;lupusSle&quot;</span> <span class="sc">~</span> <span class="st">&quot;SLE&quot;</span>,</span>
<span id="cb248-121"><a href="case-study-1.html#cb248-121" tabindex="-1"></a>      Disease <span class="sc">==</span> <span class="st">&quot;rheumatoidArthritis&quot;</span> <span class="sc">~</span> <span class="st">&quot;RA&quot;</span>,</span>
<span id="cb248-122"><a href="case-study-1.html#cb248-122" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Normal&quot;</span></span>
<span id="cb248-123"><a href="case-study-1.html#cb248-123" tabindex="-1"></a>    ))</span>
<span id="cb248-124"><a href="case-study-1.html#cb248-124" tabindex="-1"></a></span>
<span id="cb248-125"><a href="case-study-1.html#cb248-125" tabindex="-1"></a><span class="co"># 11a. Create elevated SLE boxplots and save as PDF</span></span>
<span id="cb248-126"><a href="case-study-1.html#cb248-126" tabindex="-1"></a>boxplot_sle_up <span class="ot">&lt;-</span> data.long <span class="sc">%&gt;%</span></span>
<span id="cb248-127"><a href="case-study-1.html#cb248-127" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_sle_up, </span>
<span id="cb248-128"><a href="case-study-1.html#cb248-128" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> sle_samples) <span class="sc">%&gt;%</span></span>
<span id="cb248-129"><a href="case-study-1.html#cb248-129" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb248-130"><a href="case-study-1.html#cb248-130" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb248-131"><a href="case-study-1.html#cb248-131" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-132"><a href="case-study-1.html#cb248-132" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Elevated Significant Differential Abundance Targets - SLE (n=&quot;</span>, n_sle, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-133"><a href="case-study-1.html#cb248-133" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-134"><a href="case-study-1.html#cb248-134" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb248-135"><a href="case-study-1.html#cb248-135" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb248-136"><a href="case-study-1.html#cb248-136" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb248-137"><a href="case-study-1.html#cb248-137" tabindex="-1"></a></span>
<span id="cb248-138"><a href="case-study-1.html#cb248-138" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb248-139"><a href="case-study-1.html#cb248-139" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;boxplot_sle_elevated.pdf&quot;</span>,</span>
<span id="cb248-140"><a href="case-study-1.html#cb248-140" tabindex="-1"></a>  <span class="at">plot =</span> boxplot_sle_up,</span>
<span id="cb248-141"><a href="case-study-1.html#cb248-141" tabindex="-1"></a>  <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>,</span>
<span id="cb248-142"><a href="case-study-1.html#cb248-142" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-143"><a href="case-study-1.html#cb248-143" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb248-144"><a href="case-study-1.html#cb248-144" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">10</span></span>
<span id="cb248-145"><a href="case-study-1.html#cb248-145" tabindex="-1"></a>)</span>
<span id="cb248-146"><a href="case-study-1.html#cb248-146" tabindex="-1"></a></span>
<span id="cb248-147"><a href="case-study-1.html#cb248-147" tabindex="-1"></a><span class="co"># 12a. Create reduced SLE boxplots and save as PDF</span></span>
<span id="cb248-148"><a href="case-study-1.html#cb248-148" tabindex="-1"></a>boxplot_sle_down <span class="ot">&lt;-</span> data.long <span class="sc">%&gt;%</span></span>
<span id="cb248-149"><a href="case-study-1.html#cb248-149" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_sle_down, </span>
<span id="cb248-150"><a href="case-study-1.html#cb248-150" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> sle_samples) <span class="sc">%&gt;%</span></span>
<span id="cb248-151"><a href="case-study-1.html#cb248-151" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb248-152"><a href="case-study-1.html#cb248-152" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb248-153"><a href="case-study-1.html#cb248-153" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-154"><a href="case-study-1.html#cb248-154" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Reduced Significant Differential Abundance Targets - </span><span class="sc">\n</span><span class="st">SLE (n=&quot;</span>, n_sle, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-155"><a href="case-study-1.html#cb248-155" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-156"><a href="case-study-1.html#cb248-156" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb248-157"><a href="case-study-1.html#cb248-157" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>))</span>
<span id="cb248-158"><a href="case-study-1.html#cb248-158" tabindex="-1"></a></span>
<span id="cb248-159"><a href="case-study-1.html#cb248-159" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb248-160"><a href="case-study-1.html#cb248-160" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;boxplot_sle_reduced.pdf&quot;</span>,</span>
<span id="cb248-161"><a href="case-study-1.html#cb248-161" tabindex="-1"></a>  <span class="at">plot =</span> boxplot_sle_down,</span>
<span id="cb248-162"><a href="case-study-1.html#cb248-162" tabindex="-1"></a>  <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>,</span>
<span id="cb248-163"><a href="case-study-1.html#cb248-163" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-164"><a href="case-study-1.html#cb248-164" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb248-165"><a href="case-study-1.html#cb248-165" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">3</span></span>
<span id="cb248-166"><a href="case-study-1.html#cb248-166" tabindex="-1"></a>)</span>
<span id="cb248-167"><a href="case-study-1.html#cb248-167" tabindex="-1"></a></span>
<span id="cb248-168"><a href="case-study-1.html#cb248-168" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-169"><a href="case-study-1.html#cb248-169" tabindex="-1"></a><span class="co"># RA ANALYSIS</span></span>
<span id="cb248-170"><a href="case-study-1.html#cb248-170" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-171"><a href="case-study-1.html#cb248-171" tabindex="-1"></a></span>
<span id="cb248-172"><a href="case-study-1.html#cb248-172" tabindex="-1"></a><span class="co"># 5b. Filter significant RA results</span></span>
<span id="cb248-173"><a href="case-study-1.html#cb248-173" tabindex="-1"></a>sig_targets_ra <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-174"><a href="case-study-1.html#cb248-174" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-175"><a href="case-study-1.html#cb248-175" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-176"><a href="case-study-1.html#cb248-176" tabindex="-1"></a></span>
<span id="cb248-177"><a href="case-study-1.html#cb248-177" tabindex="-1"></a>sig_targets_ra_up <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-178"><a href="case-study-1.html#cb248-178" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, </span>
<span id="cb248-179"><a href="case-study-1.html#cb248-179" tabindex="-1"></a>         DiseaserheumatoidArthritis_coef <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-180"><a href="case-study-1.html#cb248-180" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-181"><a href="case-study-1.html#cb248-181" tabindex="-1"></a></span>
<span id="cb248-182"><a href="case-study-1.html#cb248-182" tabindex="-1"></a>sig_targets_ra_down <span class="ot">&lt;-</span> lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb248-183"><a href="case-study-1.html#cb248-183" tabindex="-1"></a>  <span class="fu">filter</span>(DiseaserheumatoidArthritis_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, </span>
<span id="cb248-184"><a href="case-study-1.html#cb248-184" tabindex="-1"></a>         DiseaserheumatoidArthritis_coef <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb248-185"><a href="case-study-1.html#cb248-185" tabindex="-1"></a>  <span class="fu">pull</span>(target)</span>
<span id="cb248-186"><a href="case-study-1.html#cb248-186" tabindex="-1"></a></span>
<span id="cb248-187"><a href="case-study-1.html#cb248-187" tabindex="-1"></a><span class="co"># 6b. Get RA and normal samples</span></span>
<span id="cb248-188"><a href="case-study-1.html#cb248-188" tabindex="-1"></a>ra_samples <span class="ot">&lt;-</span> metadata.a <span class="sc">%&gt;%</span></span>
<span id="cb248-189"><a href="case-study-1.html#cb248-189" tabindex="-1"></a>  <span class="fu">filter</span>(Disease <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>, <span class="st">&quot;rheumatoidArthritis&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb248-190"><a href="case-study-1.html#cb248-190" tabindex="-1"></a>  <span class="fu">pull</span>(sampleName)</span>
<span id="cb248-191"><a href="case-study-1.html#cb248-191" tabindex="-1"></a></span>
<span id="cb248-192"><a href="case-study-1.html#cb248-192" tabindex="-1"></a><span class="co"># 7b. Create RA volcano plot and save as PDF</span></span>
<span id="cb248-193"><a href="case-study-1.html#cb248-193" tabindex="-1"></a><span class="fu">volcanoPlot</span>(</span>
<span id="cb248-194"><a href="case-study-1.html#cb248-194" tabindex="-1"></a>  <span class="at">coefs =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaserheumatoidArthritis_coef,</span>
<span id="cb248-195"><a href="case-study-1.html#cb248-195" tabindex="-1"></a>  <span class="at">p_vals =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>DiseaserheumatoidArthritis_pval_FDR,</span>
<span id="cb248-196"><a href="case-study-1.html#cb248-196" tabindex="-1"></a>  <span class="at">target_labels =</span> lmTest_da<span class="sc">$</span>modelStats<span class="sc">$</span>target,</span>
<span id="cb248-197"><a href="case-study-1.html#cb248-197" tabindex="-1"></a>  <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Rheumatoid Arthritis (RA) (n=&quot;</span>, n_ra, <span class="st">&quot;) </span><span class="sc">\n</span><span class="st"> vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-198"><a href="case-study-1.html#cb248-198" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;volcano_plot_ra_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-199"><a href="case-study-1.html#cb248-199" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-200"><a href="case-study-1.html#cb248-200" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">6</span>,</span>
<span id="cb248-201"><a href="case-study-1.html#cb248-201" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">5</span></span>
<span id="cb248-202"><a href="case-study-1.html#cb248-202" tabindex="-1"></a>)</span>
<span id="cb248-203"><a href="case-study-1.html#cb248-203" tabindex="-1"></a></span>
<span id="cb248-204"><a href="case-study-1.html#cb248-204" tabindex="-1"></a><span class="co"># 8b. Create RA heatmap and save as PDF</span></span>
<span id="cb248-205"><a href="case-study-1.html#cb248-205" tabindex="-1"></a>h_ra <span class="ot">&lt;-</span> <span class="fu">generate_heatmap</span>(</span>
<span id="cb248-206"><a href="case-study-1.html#cb248-206" tabindex="-1"></a>  <span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb248-207"><a href="case-study-1.html#cb248-207" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb248-208"><a href="case-study-1.html#cb248-208" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb248-209"><a href="case-study-1.html#cb248-209" tabindex="-1"></a>  <span class="at">sample_subset =</span> ra_samples,</span>
<span id="cb248-210"><a href="case-study-1.html#cb248-210" tabindex="-1"></a>  <span class="at">target_subset =</span> sig_targets_ra,</span>
<span id="cb248-211"><a href="case-study-1.html#cb248-211" tabindex="-1"></a>  <span class="at">annotate_sample_by =</span> <span class="fu">c</span>(<span class="st">&quot;Disease&quot;</span>, <span class="st">&quot;Sample Matrix&quot;</span>, <span class="st">&quot;Lab&quot;</span>),</span>
<span id="cb248-212"><a href="case-study-1.html#cb248-212" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-213"><a href="case-study-1.html#cb248-213" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;heatmap_ra_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-214"><a href="case-study-1.html#cb248-214" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">8</span>,</span>
<span id="cb248-215"><a href="case-study-1.html#cb248-215" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">6</span></span>
<span id="cb248-216"><a href="case-study-1.html#cb248-216" tabindex="-1"></a>)</span>
<span id="cb248-217"><a href="case-study-1.html#cb248-217" tabindex="-1"></a></span>
<span id="cb248-218"><a href="case-study-1.html#cb248-218" tabindex="-1"></a><span class="co"># 9b. Create RA PCA and save as PDF</span></span>
<span id="cb248-219"><a href="case-study-1.html#cb248-219" tabindex="-1"></a>p_ra <span class="ot">&lt;-</span> <span class="fu">generate_pca</span>(</span>
<span id="cb248-220"><a href="case-study-1.html#cb248-220" tabindex="-1"></a>  <span class="at">data =</span> data.a<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ,</span>
<span id="cb248-221"><a href="case-study-1.html#cb248-221" tabindex="-1"></a>  <span class="at">plot_title =</span> <span class="st">&quot;PCA: RA vs Normal</span><span class="sc">\n</span><span class="st">Significant DA Targets&quot;</span>,</span>
<span id="cb248-222"><a href="case-study-1.html#cb248-222" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata_rename,</span>
<span id="cb248-223"><a href="case-study-1.html#cb248-223" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;sampleName&quot;</span>,</span>
<span id="cb248-224"><a href="case-study-1.html#cb248-224" tabindex="-1"></a>  <span class="at">sample_subset =</span> ra_samples,</span>
<span id="cb248-225"><a href="case-study-1.html#cb248-225" tabindex="-1"></a>  <span class="at">target_subset =</span> sig_targets_ra,</span>
<span id="cb248-226"><a href="case-study-1.html#cb248-226" tabindex="-1"></a>  <span class="at">annotate_sample_by =</span> <span class="st">&quot;Disease&quot;</span>,</span>
<span id="cb248-227"><a href="case-study-1.html#cb248-227" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-228"><a href="case-study-1.html#cb248-228" tabindex="-1"></a>  <span class="at">plot_name =</span> <span class="st">&quot;pca_plot_ra_vs_normal.pdf&quot;</span>,</span>
<span id="cb248-229"><a href="case-study-1.html#cb248-229" tabindex="-1"></a>  <span class="at">plot_width =</span> <span class="dv">5</span>,</span>
<span id="cb248-230"><a href="case-study-1.html#cb248-230" tabindex="-1"></a>  <span class="at">plot_height =</span> <span class="dv">4</span></span>
<span id="cb248-231"><a href="case-study-1.html#cb248-231" tabindex="-1"></a>)</span>
<span id="cb248-232"><a href="case-study-1.html#cb248-232" tabindex="-1"></a></span>
<span id="cb248-233"><a href="case-study-1.html#cb248-233" tabindex="-1"></a><span class="co"># 10b. Create elevated RA boxplots and save as PDF</span></span>
<span id="cb248-234"><a href="case-study-1.html#cb248-234" tabindex="-1"></a>boxplot_ra_up <span class="ot">&lt;-</span> data.long <span class="sc">%&gt;%</span></span>
<span id="cb248-235"><a href="case-study-1.html#cb248-235" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_ra_up, </span>
<span id="cb248-236"><a href="case-study-1.html#cb248-236" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> ra_samples) <span class="sc">%&gt;%</span></span>
<span id="cb248-237"><a href="case-study-1.html#cb248-237" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb248-238"><a href="case-study-1.html#cb248-238" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb248-239"><a href="case-study-1.html#cb248-239" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-240"><a href="case-study-1.html#cb248-240" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Elevated Significant Differential Abundance Targets - RA (n=&quot;</span>, n_ra, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-241"><a href="case-study-1.html#cb248-241" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-242"><a href="case-study-1.html#cb248-242" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb248-243"><a href="case-study-1.html#cb248-243" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>))</span>
<span id="cb248-244"><a href="case-study-1.html#cb248-244" tabindex="-1"></a></span>
<span id="cb248-245"><a href="case-study-1.html#cb248-245" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb248-246"><a href="case-study-1.html#cb248-246" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;boxplot_ra_elevated.pdf&quot;</span>,</span>
<span id="cb248-247"><a href="case-study-1.html#cb248-247" tabindex="-1"></a>  <span class="at">plot =</span> boxplot_ra_up,</span>
<span id="cb248-248"><a href="case-study-1.html#cb248-248" tabindex="-1"></a>  <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>,</span>
<span id="cb248-249"><a href="case-study-1.html#cb248-249" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-250"><a href="case-study-1.html#cb248-250" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb248-251"><a href="case-study-1.html#cb248-251" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">10</span></span>
<span id="cb248-252"><a href="case-study-1.html#cb248-252" tabindex="-1"></a>)</span>
<span id="cb248-253"><a href="case-study-1.html#cb248-253" tabindex="-1"></a></span>
<span id="cb248-254"><a href="case-study-1.html#cb248-254" tabindex="-1"></a><span class="co"># 11b. Create reduced RA boxplots and save as PDF</span></span>
<span id="cb248-255"><a href="case-study-1.html#cb248-255" tabindex="-1"></a>boxplot_ra_down <span class="ot">&lt;-</span> data.long <span class="sc">%&gt;%</span></span>
<span id="cb248-256"><a href="case-study-1.html#cb248-256" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">%in%</span> sig_targets_ra_down, </span>
<span id="cb248-257"><a href="case-study-1.html#cb248-257" tabindex="-1"></a>         SampleName <span class="sc">%in%</span> ra_samples) <span class="sc">%&gt;%</span></span>
<span id="cb248-258"><a href="case-study-1.html#cb248-258" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>, <span class="at">y =</span> NPQ, <span class="at">fill =</span> <span class="st">`</span><span class="at">Disease Type</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb248-259"><a href="case-study-1.html#cb248-259" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb248-260"><a href="case-study-1.html#cb248-260" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Target, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-261"><a href="case-study-1.html#cb248-261" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Reduced Significant Differential Abundance Targets - </span><span class="sc">\n</span><span class="st">RA (n=&quot;</span>, n_ra, <span class="st">&quot;) vs Normal (n=&quot;</span>, n_normal, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb248-262"><a href="case-study-1.html#cb248-262" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;with FDR-adjusted p &lt; 0.05&quot;</span>) <span class="sc">+</span></span>
<span id="cb248-263"><a href="case-study-1.html#cb248-263" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb248-264"><a href="case-study-1.html#cb248-264" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb248-265"><a href="case-study-1.html#cb248-265" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb248-266"><a href="case-study-1.html#cb248-266" tabindex="-1"></a></span>
<span id="cb248-267"><a href="case-study-1.html#cb248-267" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb248-268"><a href="case-study-1.html#cb248-268" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;boxplot_ra_reduced.pdf&quot;</span>,</span>
<span id="cb248-269"><a href="case-study-1.html#cb248-269" tabindex="-1"></a>  <span class="at">plot =</span> boxplot_ra_down,</span>
<span id="cb248-270"><a href="case-study-1.html#cb248-270" tabindex="-1"></a>  <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>,</span>
<span id="cb248-271"><a href="case-study-1.html#cb248-271" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;figures&quot;</span>,</span>
<span id="cb248-272"><a href="case-study-1.html#cb248-272" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb248-273"><a href="case-study-1.html#cb248-273" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">3</span></span>
<span id="cb248-274"><a href="case-study-1.html#cb248-274" tabindex="-1"></a>)</span>
<span id="cb248-275"><a href="case-study-1.html#cb248-275" tabindex="-1"></a></span>
<span id="cb248-276"><a href="case-study-1.html#cb248-276" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-277"><a href="case-study-1.html#cb248-277" tabindex="-1"></a><span class="co"># COMPARATIVE ANALYSIS AND EXPORT</span></span>
<span id="cb248-278"><a href="case-study-1.html#cb248-278" tabindex="-1"></a><span class="co"># ============================================================================</span></span>
<span id="cb248-279"><a href="case-study-1.html#cb248-279" tabindex="-1"></a></span>
<span id="cb248-280"><a href="case-study-1.html#cb248-280" tabindex="-1"></a><span class="co"># 13. Analyze overlap between diseases</span></span>
<span id="cb248-281"><a href="case-study-1.html#cb248-281" tabindex="-1"></a>shared_targets <span class="ot">&lt;-</span> <span class="fu">intersect</span>(sig_targets_sle, sig_targets_ra)</span>
<span id="cb248-282"><a href="case-study-1.html#cb248-282" tabindex="-1"></a>sle_specific <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(sig_targets_sle, sig_targets_ra)</span>
<span id="cb248-283"><a href="case-study-1.html#cb248-283" tabindex="-1"></a>ra_specific <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(sig_targets_ra, sig_targets_sle)</span>
<span id="cb248-284"><a href="case-study-1.html#cb248-284" tabindex="-1"></a></span>
<span id="cb248-285"><a href="case-study-1.html#cb248-285" tabindex="-1"></a><span class="co"># 14. Export all results to CSV</span></span>
<span id="cb248-286"><a href="case-study-1.html#cb248-286" tabindex="-1"></a><span class="fu">write_csv</span>(lmTest_da<span class="sc">$</span>modelStats, <span class="st">&quot;results/all_da_results.csv&quot;</span>)</span>
<span id="cb248-287"><a href="case-study-1.html#cb248-287" tabindex="-1"></a></span>
<span id="cb248-288"><a href="case-study-1.html#cb248-288" tabindex="-1"></a><span class="co"># Export SLE results</span></span>
<span id="cb248-289"><a href="case-study-1.html#cb248-289" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb248-290"><a href="case-study-1.html#cb248-290" tabindex="-1"></a>  lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(target <span class="sc">%in%</span> sig_targets_sle),</span>
<span id="cb248-291"><a href="case-study-1.html#cb248-291" tabindex="-1"></a>  <span class="st">&quot;results/sig_sle_proteins.csv&quot;</span></span>
<span id="cb248-292"><a href="case-study-1.html#cb248-292" tabindex="-1"></a>)</span>
<span id="cb248-293"><a href="case-study-1.html#cb248-293" tabindex="-1"></a></span>
<span id="cb248-294"><a href="case-study-1.html#cb248-294" tabindex="-1"></a><span class="co"># Export RA results</span></span>
<span id="cb248-295"><a href="case-study-1.html#cb248-295" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb248-296"><a href="case-study-1.html#cb248-296" tabindex="-1"></a>  lmTest_da<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(target <span class="sc">%in%</span> sig_targets_ra),</span>
<span id="cb248-297"><a href="case-study-1.html#cb248-297" tabindex="-1"></a>  <span class="st">&quot;results/sig_ra_proteins.csv&quot;</span></span>
<span id="cb248-298"><a href="case-study-1.html#cb248-298" tabindex="-1"></a>)</span>
<span id="cb248-299"><a href="case-study-1.html#cb248-299" tabindex="-1"></a></span>
<span id="cb248-300"><a href="case-study-1.html#cb248-300" tabindex="-1"></a><span class="co"># Export overlap results</span></span>
<span id="cb248-301"><a href="case-study-1.html#cb248-301" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb248-302"><a href="case-study-1.html#cb248-302" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">target =</span> shared_targets),</span>
<span id="cb248-303"><a href="case-study-1.html#cb248-303" tabindex="-1"></a>  <span class="st">&quot;results/shared_sle_ra_proteins.csv&quot;</span></span>
<span id="cb248-304"><a href="case-study-1.html#cb248-304" tabindex="-1"></a>)</span>
<span id="cb248-305"><a href="case-study-1.html#cb248-305" tabindex="-1"></a></span>
<span id="cb248-306"><a href="case-study-1.html#cb248-306" tabindex="-1"></a><span class="co"># 15. Print comprehensive summary</span></span>
<span id="cb248-307"><a href="case-study-1.html#cb248-307" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">========================================</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-308"><a href="case-study-1.html#cb248-308" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;DIFFERENTIAL ABUNDANCE SUMMARY</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-309"><a href="case-study-1.html#cb248-309" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;========================================</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-310"><a href="case-study-1.html#cb248-310" tabindex="-1"></a></span>
<span id="cb248-311"><a href="case-study-1.html#cb248-311" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Quality Control:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-312"><a href="case-study-1.html#cb248-312" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Targets passing detectability threshold:&quot;</span>, <span class="fu">length</span>(targets_passed), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-313"><a href="case-study-1.html#cb248-313" tabindex="-1"></a></span>
<span id="cb248-314"><a href="case-study-1.html#cb248-314" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;SLE vs Normal:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-315"><a href="case-study-1.html#cb248-315" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Total significant proteins:&quot;</span>, <span class="fu">length</span>(sig_targets_sle), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-316"><a href="case-study-1.html#cb248-316" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Elevated:&quot;</span>, <span class="fu">length</span>(sig_targets_sle_up), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-317"><a href="case-study-1.html#cb248-317" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Reduced:&quot;</span>, <span class="fu">length</span>(sig_targets_sle_down), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-318"><a href="case-study-1.html#cb248-318" tabindex="-1"></a></span>
<span id="cb248-319"><a href="case-study-1.html#cb248-319" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;RA vs Normal:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-320"><a href="case-study-1.html#cb248-320" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Total significant proteins:&quot;</span>, <span class="fu">length</span>(sig_targets_ra), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-321"><a href="case-study-1.html#cb248-321" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Elevated:&quot;</span>, <span class="fu">length</span>(sig_targets_ra_up), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-322"><a href="case-study-1.html#cb248-322" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Reduced:&quot;</span>, <span class="fu">length</span>(sig_targets_ra_down), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-323"><a href="case-study-1.html#cb248-323" tabindex="-1"></a></span>
<span id="cb248-324"><a href="case-study-1.html#cb248-324" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Disease Comparison:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-325"><a href="case-study-1.html#cb248-325" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Shared significant targets:&quot;</span>, <span class="fu">length</span>(shared_targets), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-326"><a href="case-study-1.html#cb248-326" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  SLE-specific targets:&quot;</span>, <span class="fu">length</span>(sle_specific), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-327"><a href="case-study-1.html#cb248-327" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  RA-specific targets:&quot;</span>, <span class="fu">length</span>(ra_specific), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-328"><a href="case-study-1.html#cb248-328" tabindex="-1"></a></span>
<span id="cb248-329"><a href="case-study-1.html#cb248-329" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Output files created in:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-330"><a href="case-study-1.html#cb248-330" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  - figures/  (all plots as PDFs)</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-331"><a href="case-study-1.html#cb248-331" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  - results/  (all statistical results as CSVs)</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb248-332"><a href="case-study-1.html#cb248-332" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;========================================</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="outcome-modeling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="additional-resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": {},
    "google": false
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  },
  "toolbar": {
    "position": "fixed"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
