<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Predictive Modeling | NULISAseqR User Guide</title>
  <meta name="description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Predictive Modeling | NULISAseqR User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  <meta name="github-repo" content="Alamar-Biosciences/NULISAseqR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Predictive Modeling | NULISAseqR User Guide" />
  
  <meta name="twitter:description" content="A comprehensive guide to analyzing NULISAseq proteomic data with the NULISAseqR R package" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="longitudinal-analysis.html"/>
<link rel="next" href="complete-workflow-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li style="text-align: center; padding: 20px;">
  <a href="https://www.alamarbio.com" target="_blank">
    <img src="figures/alamar_logo.png" style="height:70px;">
  </a>
</li>
<li><a href="./">NULISAseqR User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-is-nulisaseqr"><i class="fa fa-check"></i>What is NULISAseqR?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i>Installation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#getting-help"><i class="fa fa-check"></i>Getting Help</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html"><i class="fa fa-check"></i>Understanding NULISAseq Data</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#what-is-nulisa-technology"><i class="fa fa-check"></i>What is NULISA™ Technology?</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#technical-workflow"><i class="fa fa-check"></i>Technical Workflow</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#learn-more-about-nulisa-technology"><i class="fa fa-check"></i>Learn More About NULISA™ Technology</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#available-nulisaseq-panels"><i class="fa fa-check"></i>Available NULISAseq Panels</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#data-normalization"><i class="fa fa-check"></i>Data Normalization</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#raw-data-sequencing-counts"><i class="fa fa-check"></i>Raw Data: Sequencing Counts</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-1-internal-control-ic-normalization"><i class="fa fa-check"></i>Step 1: Internal Control (IC) Normalization</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-2-inter-plate-control-ipc-normalization"><i class="fa fa-check"></i>Step 2: Inter-Plate Control (IPC) Normalization</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#step-3-log2-transformation"><i class="fa fa-check"></i>Step 3: Log2 Transformation</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#why-log-transform"><i class="fa fa-check"></i>Why Log Transform?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#npq-vs-fold-change"><i class="fa fa-check"></i>NPQ vs Fold Change</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#understanding-the-relationship"><i class="fa fa-check"></i>Understanding the Relationship</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#working-with-npq-values"><i class="fa fa-check"></i>Working with NPQ Values</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#calculating-fold-changes-from-npq"><i class="fa fa-check"></i>Calculating Fold Changes from NPQ</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#comparing-across-targets"><i class="fa fa-check"></i>Comparing Across Targets</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#data-analysis-faq"><i class="fa fa-check"></i>Data Analysis FAQ</a>
<ul>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-is-the-limit-of-detection-lod-defined-for-nulisaseq-why-do-some-targets-have-lod-of-0"><i class="fa fa-check"></i>1. How is the limit of detection (LOD) defined for NULISAseq? <br> Why do some targets have LOD of 0?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#should-i-exclude-samples-below-lod"><i class="fa fa-check"></i>2. Should I exclude samples below LOD?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-do-i-calculate-coefficient-of-variation-cv-for-samples"><i class="fa fa-check"></i>3. How do I calculate coefficient of variation (CV) for samples?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#can-i-compare-data-across-different-runs"><i class="fa fa-check"></i>4. Can I compare data across different runs?</a></li>
<li class="chapter" data-level="" data-path="understanding-nulisaseq-data.html"><a href="understanding-nulisaseq-data.html#how-do-you-assess-correlation-between-npq-and-absolute-quantification-data-from-other-platforms"><i class="fa fa-check"></i>5. How do you assess correlation between NPQ and absolute quantification data from other platforms?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="data-import.html"><a href="data-import.html"><i class="fa fa-check"></i><b>1</b> Data Import</a>
<ul>
<li class="chapter" data-level="1.1" data-path="data-import.html"><a href="data-import.html#the-importnulisaseq-function"><i class="fa fa-check"></i><b>1.1</b> The <code>importNULISAseq()</code> Function</a></li>
<li class="chapter" data-level="1.2" data-path="data-import.html"><a href="data-import.html#understanding-the-data-structure"><i class="fa fa-check"></i><b>1.2</b> Understanding the Data Structure</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="data-import.html"><a href="data-import.html#key-components"><i class="fa fa-check"></i><b>1.2.1</b> Key Components</a></li>
<li class="chapter" data-level="1.2.2" data-path="data-import.html"><a href="data-import.html#summary-statistics"><i class="fa fa-check"></i><b>1.2.2</b> Summary Statistics</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="data-import.html"><a href="data-import.html#working-with-your-own-data"><i class="fa fa-check"></i><b>1.3</b> Working with Your Own Data</a></li>
<li class="chapter" data-level="1.4" data-path="data-import.html"><a href="data-import.html#loading-metadata"><i class="fa fa-check"></i><b>1.4</b> Loading Metadata</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="data-import.html"><a href="data-import.html#metadata-structure"><i class="fa fa-check"></i><b>1.4.1</b> Metadata Structure</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="data-import.html"><a href="data-import.html#merging-data-with-metadata"><i class="fa fa-check"></i><b>1.5</b> Merging Data with Metadata</a></li>
<li class="chapter" data-level="1.6" data-path="data-import.html"><a href="data-import.html#filtering-samples"><i class="fa fa-check"></i><b>1.6</b> Filtering Samples</a></li>
<li class="chapter" data-level="1.7" data-path="data-import.html"><a href="data-import.html#complete-import-workflow"><i class="fa fa-check"></i><b>1.7</b> Complete Import Workflow</a></li>
<li class="chapter" data-level="1.8" data-path="data-import.html"><a href="data-import.html#tips-and-best-practices"><i class="fa fa-check"></i><b>1.8</b> Tips and Best Practices</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>2</b> Quality Control</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quality-control.html"><a href="quality-control.html#why-quality-control-matters"><i class="fa fa-check"></i><b>2.1</b> Why Quality Control Matters</a></li>
<li class="chapter" data-level="2.2" data-path="quality-control.html"><a href="quality-control.html#qc-report"><i class="fa fa-check"></i><b>2.2</b> Automated QC Report</a></li>
<li class="chapter" data-level="2.3" data-path="quality-control.html"><a href="quality-control.html#qc-report-structure"><i class="fa fa-check"></i><b>2.3</b> QC Report Structure</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="quality-control.html"><a href="quality-control.html#plate-layout"><i class="fa fa-check"></i><b>2.3.1</b> Plate Layout</a></li>
<li class="chapter" data-level="2.3.2" data-path="quality-control.html"><a href="quality-control.html#run-summary"><i class="fa fa-check"></i><b>2.3.2</b> Run Summary</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#quality-control-metrics"><i class="fa fa-check"></i>Quality Control Metrics</a></li>
<li class="chapter" data-level="2.3.3" data-path="quality-control.html"><a href="quality-control.html#platerun-qc"><i class="fa fa-check"></i><b>2.3.3</b> Plate/Run QC</a></li>
<li class="chapter" data-level="2.3.4" data-path="quality-control.html"><a href="quality-control.html#sample-qc"><i class="fa fa-check"></i><b>2.3.4</b> Sample QC</a></li>
<li class="chapter" data-level="2.3.5" data-path="quality-control.html"><a href="quality-control.html#target-qc"><i class="fa fa-check"></i><b>2.3.5</b> Target QC</a></li>
<li class="chapter" data-level="2.3.6" data-path="quality-control.html"><a href="quality-control.html#coefficient-of-variation-cv"><i class="fa fa-check"></i><b>2.3.6</b> Coefficient of Variation (CV)</a></li>
<li class="chapter" data-level="2.3.7" data-path="quality-control.html"><a href="quality-control.html#plate-effect-assessment"><i class="fa fa-check"></i><b>2.3.7</b> Plate Effect Assessment</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="quality-control.html"><a href="quality-control.html#qc-thresholds-summary"><i class="fa fa-check"></i><b>2.4</b> Summary of QC Thresholds</a></li>
<li class="chapter" data-level="2.5" data-path="quality-control.html"><a href="quality-control.html#manual-qc-checks"><i class="fa fa-check"></i><b>2.5</b> Manual QC Checks</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="quality-control.html"><a href="quality-control.html#check-for-missing-data"><i class="fa fa-check"></i><b>2.5.1</b> Check for Missing Data</a></li>
<li class="chapter" data-level="2.5.2" data-path="quality-control.html"><a href="quality-control.html#low-detectability-targets"><i class="fa fa-check"></i><b>2.5.2</b> Low Detectability Targets</a></li>
<li class="chapter" data-level="2.5.3" data-path="quality-control.html"><a href="quality-control.html#calculate-targets-detectability-by-subgroups"><i class="fa fa-check"></i><b>2.5.3</b> Calculate Targets Detectability by Subgroups</a></li>
<li class="chapter" data-level="2.5.4" data-path="quality-control.html"><a href="quality-control.html#batch-effect-correction"><i class="fa fa-check"></i><b>2.5.4</b> Batch Effect Correction</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="quality-control.html"><a href="quality-control.html#qc-best-practices"><i class="fa fa-check"></i><b>2.6</b> Best Practices</a>
<ul>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#documentation"><i class="fa fa-check"></i>Documentation</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#consistency"><i class="fa fa-check"></i>Consistency</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#collaboration"><i class="fa fa-check"></i>Collaboration</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#when-in-doubt"><i class="fa fa-check"></i>When in Doubt</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="quality-control.html"><a href="quality-control.html#complete-qc-workflow"><i class="fa fa-check"></i><b>2.7</b> Complete QC Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>3</b> Visualization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualization.html"><a href="visualization.html#why-visualize"><i class="fa fa-check"></i><b>3.1</b> Why Visualize?</a></li>
<li class="chapter" data-level="3.2" data-path="visualization.html"><a href="visualization.html#heatmaps"><i class="fa fa-check"></i><b>3.2</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="visualization.html"><a href="visualization.html#basic-heatmap"><i class="fa fa-check"></i><b>3.2.1</b> Basic Heatmap</a></li>
<li class="chapter" data-level="3.2.2" data-path="visualization.html"><a href="visualization.html#heatmap-with-annotations"><i class="fa fa-check"></i><b>3.2.2</b> Heatmap with Annotations</a></li>
<li class="chapter" data-level="3.2.3" data-path="visualization.html"><a href="visualization.html#understanding-the-heatmap"><i class="fa fa-check"></i><b>3.2.3</b> Understanding the Heatmap</a></li>
<li class="chapter" data-level="3.2.4" data-path="visualization.html"><a href="visualization.html#interpreting-clusters"><i class="fa fa-check"></i><b>3.2.4</b> Interpreting Clusters</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="visualization.html"><a href="visualization.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.3</b> Principal Component Analysis (PCA)</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="visualization.html"><a href="visualization.html#pca-with-multiple-visual-encodings"><i class="fa fa-check"></i><b>3.3.1</b> PCA with Multiple Visual Encodings</a></li>
<li class="chapter" data-level="3.3.2" data-path="visualization.html"><a href="visualization.html#understanding-pca-plots"><i class="fa fa-check"></i><b>3.3.2</b> Understanding PCA Plots</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="visualization.html"><a href="visualization.html#custom-visualizations-with-ggplot2"><i class="fa fa-check"></i><b>3.4</b> Custom Visualizations with <code>ggplot2</code></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="visualization.html"><a href="visualization.html#npq-distribution---histogram"><i class="fa fa-check"></i><b>3.4.1</b> NPQ Distribution - Histogram</a></li>
<li class="chapter" data-level="3.4.2" data-path="visualization.html"><a href="visualization.html#npq-distribution-by-group---density-plot"><i class="fa fa-check"></i><b>3.4.2</b> NPQ Distribution by Group - Density Plot</a></li>
<li class="chapter" data-level="3.4.3" data-path="visualization.html"><a href="visualization.html#single-protein-across-groups"><i class="fa fa-check"></i><b>3.4.3</b> Single Protein Across Groups</a></li>
<li class="chapter" data-level="3.4.4" data-path="visualization.html"><a href="visualization.html#multiple-proteins-across-groups"><i class="fa fa-check"></i><b>3.4.4</b> Multiple Proteins Across Groups</a></li>
<li class="chapter" data-level="3.4.5" data-path="visualization.html"><a href="visualization.html#protein-protein-correlation"><i class="fa fa-check"></i><b>3.4.5</b> Protein-Protein Correlation</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="visualization.html"><a href="visualization.html#saving-plots"><i class="fa fa-check"></i><b>3.5</b> Saving Plots</a>
<ul>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-heatmap"><i class="fa fa-check"></i>Save Heatmap</a></li>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-pca"><i class="fa fa-check"></i>Save PCA</a></li>
<li class="chapter" data-level="" data-path="visualization.html"><a href="visualization.html#save-ggplot-objects"><i class="fa fa-check"></i>Save ggplot Objects</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="visualization.html"><a href="visualization.html#visualization-best-practices"><i class="fa fa-check"></i><b>3.6</b> Visualization Best Practices</a></li>
<li class="chapter" data-level="3.7" data-path="visualization.html"><a href="visualization.html#complete-visualization-workflow"><i class="fa fa-check"></i><b>3.7</b> Complete Visualization Workflow</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>4</b> Differential Expression Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#overview"><i class="fa fa-check"></i><b>4.1</b> Overview</a></li>
<li class="chapter" data-level="4.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#basic-differential-expression"><i class="fa fa-check"></i><b>4.2</b> Basic Differential Expression</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#group-comparison"><i class="fa fa-check"></i><b>4.2.1</b> Group Comparison</a></li>
<li class="chapter" data-level="4.2.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#understanding-the-model"><i class="fa fa-check"></i><b>4.2.2</b> Understanding the Model</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#model-results"><i class="fa fa-check"></i><b>4.3</b> Model Results</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#key-columns-in-results"><i class="fa fa-check"></i><b>4.3.1</b> Key Columns in Results</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#finding-significant-proteins"><i class="fa fa-check"></i><b>4.4</b> Finding Significant Proteins</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#filter-by-fdr-threshold"><i class="fa fa-check"></i><b>4.4.1</b> Filter by FDR Threshold</a></li>
<li class="chapter" data-level="4.4.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#upregulated-vs-downregulated"><i class="fa fa-check"></i><b>4.4.2</b> Upregulated vs Downregulated</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#volcano-plots"><i class="fa fa-check"></i><b>4.5</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#inflammation-vs-normal"><i class="fa fa-check"></i><b>4.5.1</b> Inflammation vs Normal</a></li>
<li class="chapter" data-level="4.5.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#understanding-volcano-plots"><i class="fa fa-check"></i><b>4.5.2</b> Understanding Volcano Plots</a></li>
<li class="chapter" data-level="4.5.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#cancer-vs-normal"><i class="fa fa-check"></i><b>4.5.3</b> Cancer vs Normal</a></li>
<li class="chapter" data-level="4.5.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#saving-volcano-plots"><i class="fa fa-check"></i><b>4.5.4</b> Saving Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#visualization-of-significant-targets"><i class="fa fa-check"></i><b>4.6</b> Visualization of Significant Targets</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#inflammation-vs-normal-1"><i class="fa fa-check"></i><b>4.6.1</b> Inflammation vs Normal</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#multiple-comparisons"><i class="fa fa-check"></i><b>4.7</b> Multiple Comparisons</a></li>
<li class="chapter" data-level="4.8" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#model-considerations"><i class="fa fa-check"></i><b>4.8</b> Model Considerations</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#reference-level"><i class="fa fa-check"></i><b>4.8.1</b> Reference Level</a></li>
<li class="chapter" data-level="4.8.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#including-covariates"><i class="fa fa-check"></i><b>4.8.2</b> Including Covariates</a></li>
<li class="chapter" data-level="4.8.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#model-formula-options"><i class="fa fa-check"></i><b>4.8.3</b> Model Formula Options</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#interpreting-results"><i class="fa fa-check"></i><b>4.9</b> Interpreting Results</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#effect-size-coefficient"><i class="fa fa-check"></i><b>4.9.1</b> Effect Size (Coefficient)</a></li>
<li class="chapter" data-level="4.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#p-values"><i class="fa fa-check"></i><b>4.9.2</b> P-values</a></li>
<li class="chapter" data-level="4.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#significance-thresholds"><i class="fa fa-check"></i><b>4.9.3</b> Significance Thresholds</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#exporting-results"><i class="fa fa-check"></i><b>4.10</b> Exporting Results</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#save-results-table"><i class="fa fa-check"></i><b>4.10.1</b> Save Results Table</a></li>
<li class="chapter" data-level="4.10.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#create-summary-table"><i class="fa fa-check"></i><b>4.10.2</b> Create Summary Table</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#complete-differential-expression-workflow"><i class="fa fa-check"></i><b>4.11</b> Complete Differential Expression Workflow</a></li>
<li class="chapter" data-level="4.12" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#best-practices"><i class="fa fa-check"></i><b>4.12</b> Best Practices</a></li>
<li class="chapter" data-level="4.13" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#common-issues"><i class="fa fa-check"></i><b>4.13</b> Common Issues</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html"><i class="fa fa-check"></i><b>5</b> Longitudinal Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="longitudinal-analysis.html"><a href="longitudinal-analysis.html#overview-1"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="predictive-modeling.html"><a href="predictive-modeling.html"><i class="fa fa-check"></i><b>6</b> Predictive Modeling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#overview-2"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="predictive-modeling.html"><a href="predictive-modeling.html#continuous-outcomes"><i class="fa fa-check"></i><b>6.2</b> Continuous Outcomes</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#example-predicting-a-clinical-score"><i class="fa fa-check"></i><b>6.2.1</b> Example: Predicting a Clinical Score</a></li>
<li class="chapter" data-level="6.2.2" data-path="predictive-modeling.html"><a href="predictive-modeling.html#understanding-the-model-1"><i class="fa fa-check"></i><b>6.2.2</b> Understanding the Model</a></li>
<li class="chapter" data-level="6.2.3" data-path="predictive-modeling.html"><a href="predictive-modeling.html#examining-results"><i class="fa fa-check"></i><b>6.2.3</b> Examining Results</a></li>
<li class="chapter" data-level="6.2.4" data-path="predictive-modeling.html"><a href="predictive-modeling.html#key-columns"><i class="fa fa-check"></i><b>6.2.4</b> Key Columns</a></li>
<li class="chapter" data-level="6.2.5" data-path="predictive-modeling.html"><a href="predictive-modeling.html#finding-predictive-proteins"><i class="fa fa-check"></i><b>6.2.5</b> Finding Predictive Proteins</a></li>
<li class="chapter" data-level="6.2.6" data-path="predictive-modeling.html"><a href="predictive-modeling.html#interpreting-coefficients"><i class="fa fa-check"></i><b>6.2.6</b> Interpreting Coefficients</a></li>
<li class="chapter" data-level="6.2.7" data-path="predictive-modeling.html"><a href="predictive-modeling.html#visualizing-predictions"><i class="fa fa-check"></i><b>6.2.7</b> Visualizing Predictions</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="predictive-modeling.html"><a href="predictive-modeling.html#binary-outcomes"><i class="fa fa-check"></i><b>6.3</b> Binary Outcomes</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#example-predicting-disease-status"><i class="fa fa-check"></i><b>6.3.1</b> Example: Predicting Disease Status</a></li>
<li class="chapter" data-level="6.3.2" data-path="predictive-modeling.html"><a href="predictive-modeling.html#understanding-the-model-2"><i class="fa fa-check"></i><b>6.3.2</b> Understanding the Model</a></li>
<li class="chapter" data-level="6.3.3" data-path="predictive-modeling.html"><a href="predictive-modeling.html#examining-results-1"><i class="fa fa-check"></i><b>6.3.3</b> Examining Results</a></li>
<li class="chapter" data-level="6.3.4" data-path="predictive-modeling.html"><a href="predictive-modeling.html#key-columns-1"><i class="fa fa-check"></i><b>6.3.4</b> Key Columns</a></li>
<li class="chapter" data-level="6.3.5" data-path="predictive-modeling.html"><a href="predictive-modeling.html#finding-predictive-proteins-1"><i class="fa fa-check"></i><b>6.3.5</b> Finding Predictive Proteins</a></li>
<li class="chapter" data-level="6.3.6" data-path="predictive-modeling.html"><a href="predictive-modeling.html#interpreting-odds-ratios"><i class="fa fa-check"></i><b>6.3.6</b> Interpreting Odds Ratios</a></li>
<li class="chapter" data-level="6.3.7" data-path="predictive-modeling.html"><a href="predictive-modeling.html#visualizing-binary-predictions"><i class="fa fa-check"></i><b>6.3.7</b> Visualizing Binary Predictions</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="predictive-modeling.html"><a href="predictive-modeling.html#comparing-approaches"><i class="fa fa-check"></i><b>6.4</b> Comparing Approaches</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#when-to-use-each-function"><i class="fa fa-check"></i><b>6.4.1</b> When to Use Each Function</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="predictive-modeling.html"><a href="predictive-modeling.html#advanced-model-comparisons"><i class="fa fa-check"></i><b>6.5</b> Advanced: Model Comparisons</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#compare-model-performance"><i class="fa fa-check"></i><b>6.5.1</b> Compare Model Performance</a></li>
<li class="chapter" data-level="6.5.2" data-path="predictive-modeling.html"><a href="predictive-modeling.html#multi-protein-models"><i class="fa fa-check"></i><b>6.5.2</b> Multi-protein Models</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="predictive-modeling.html"><a href="predictive-modeling.html#model-validation"><i class="fa fa-check"></i><b>6.6</b> Model Validation</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="predictive-modeling.html"><a href="predictive-modeling.html#cross-validation"><i class="fa fa-check"></i><b>6.6.1</b> Cross-Validation</a></li>
<li class="chapter" data-level="6.6.2" data-path="predictive-modeling.html"><a href="predictive-modeling.html#test-set-validation"><i class="fa fa-check"></i><b>6.6.2</b> Test Set Validation</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="predictive-modeling.html"><a href="predictive-modeling.html#exporting-results-1"><i class="fa fa-check"></i><b>6.7</b> Exporting Results</a></li>
<li class="chapter" data-level="6.8" data-path="predictive-modeling.html"><a href="predictive-modeling.html#complete-predictive-modeling-workflow"><i class="fa fa-check"></i><b>6.8</b> Complete Predictive Modeling Workflow</a></li>
<li class="chapter" data-level="6.9" data-path="predictive-modeling.html"><a href="predictive-modeling.html#best-practices-1"><i class="fa fa-check"></i><b>6.9</b> Best Practices</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html"><i class="fa fa-check"></i><b>7</b> Complete Workflow: <br> Identifying Proteomic Signatures in Lupus and Rheumatoid Arthritis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#overview-3"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#data-preparation"><i class="fa fa-check"></i><b>7.2</b> Data Preparation</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#load-required-libraries"><i class="fa fa-check"></i><b>7.2.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="7.2.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#import-data"><i class="fa fa-check"></i><b>7.2.2</b> Import Data</a></li>
<li class="chapter" data-level="7.2.3" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#prepare-sample-metadata"><i class="fa fa-check"></i><b>7.2.3</b> Prepare Sample Metadata</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#apply-detectability-threshold"><i class="fa fa-check"></i><b>7.3</b> Apply Detectability Threshold</a></li>
<li class="chapter" data-level="7.4" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#global-data-visualization"><i class="fa fa-check"></i><b>7.4</b> Global Data Visualization</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#prepare-metadata-for-visualization"><i class="fa fa-check"></i><b>7.4.1</b> Prepare Metadata for Visualization</a></li>
<li class="chapter" data-level="7.4.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>7.4.2</b> Hierarchical Clustering Heatmap</a></li>
<li class="chapter" data-level="7.4.3" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.4.3</b> Principal Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#statistical-testing"><i class="fa fa-check"></i><b>7.5</b> Statistical Testing</a></li>
<li class="chapter" data-level="7.6" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#systemic-lupus-erythematosus-sle-analysis"><i class="fa fa-check"></i><b>7.6</b> Systemic Lupus Erythematosus (SLE) Analysis</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#background"><i class="fa fa-check"></i><b>7.6.1</b> Background</a></li>
<li class="chapter" data-level="7.6.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#volcano-plot"><i class="fa fa-check"></i><b>7.6.2</b> Volcano Plot</a></li>
<li class="chapter" data-level="7.6.3" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#visualizations"><i class="fa fa-check"></i><b>7.6.3</b> Visualizations</a></li>
<li class="chapter" data-level="7.6.4" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#directional-analysis"><i class="fa fa-check"></i><b>7.6.4</b> Directional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#rheumatoid-arthritis-ra-analysis"><i class="fa fa-check"></i><b>7.7</b> Rheumatoid Arthritis (RA) Analysis</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#background-1"><i class="fa fa-check"></i><b>7.7.1</b> Background</a></li>
<li class="chapter" data-level="7.7.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#volcano-plot-1"><i class="fa fa-check"></i><b>7.7.2</b> Volcano Plot</a></li>
<li class="chapter" data-level="7.7.3" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#visualizations-1"><i class="fa fa-check"></i><b>7.7.3</b> Visualizations</a></li>
<li class="chapter" data-level="7.7.4" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#directional-analysis-1"><i class="fa fa-check"></i><b>7.7.4</b> Directional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#comparing-disease-signatures"><i class="fa fa-check"></i><b>7.8</b> Comparing Disease Signatures</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#overlap-analysis"><i class="fa fa-check"></i><b>7.8.1</b> Overlap Analysis</a></li>
<li class="chapter" data-level="7.8.2" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#summary-statistics-table"><i class="fa fa-check"></i><b>7.8.2</b> Summary Statistics Table</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#next-steps"><i class="fa fa-check"></i><b>7.9</b> Next Steps</a></li>
<li class="chapter" data-level="7.10" data-path="complete-workflow-1.html"><a href="complete-workflow-1.html#complete-workflow-example-code"><i class="fa fa-check"></i><b>7.10</b> Complete Workflow Example Code</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i>Additional Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#nulisaseq-resources"><i class="fa fa-check"></i>NULISAseq Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#package-documentation"><i class="fa fa-check"></i>Package Documentation</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#github-repository"><i class="fa fa-check"></i>GitHub Repository</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#support-and-issues"><i class="fa fa-check"></i>Support and Issues</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#nulisa-analysis-software-nas"><i class="fa fa-check"></i>NULISA Analysis Software (NAS)</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#technical-documentation"><i class="fa fa-check"></i>Technical Documentation</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#r-resources"><i class="fa fa-check"></i>R Programming Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#r-install"><i class="fa fa-check"></i>Installing R and RStudio</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#tidyverse-resources"><i class="fa fa-check"></i>Tidyverse Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#visualization-resources"><i class="fa fa-check"></i>Data Visualization Resources</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#ggplot2"><i class="fa fa-check"></i>ggplot2</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#complexheatmap"><i class="fa fa-check"></i>ComplexHeatmap</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#pcatools"><i class="fa fa-check"></i>PCAtools</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citations"><i class="fa fa-check"></i>Citing This Book and Packages</a>
<ul>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citing-this-book"><i class="fa fa-check"></i>Citing This Book</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#citing-r-and-packages"><i class="fa fa-check"></i>Citing R and Packages</a></li>
<li class="chapter" data-level="" data-path="additional-resources.html"><a href="additional-resources.html#key-citations"><i class="fa fa-check"></i>Key Citations</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/Alamar-Biosciences/NULISAseqR" target="_blank">GitHub Repository</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NULISAseqR User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="predictive-modeling" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Predictive Modeling<a href="predictive-modeling.html#predictive-modeling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter covers using protein expression as predictors to model outcomes, including both continuous and binary responses.</p>
<div id="overview-2" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Overview<a href="predictive-modeling.html#overview-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While differential expression asks <strong>“What proteins differ between groups?”</strong>, predictive modeling asks <strong>“Which proteins predict an outcome?”</strong></p>
<p>NULISAseqR provides two functions:</p>
<ul>
<li><strong><code>lmNULISAseq_predict()</code></strong>: For continuous outcomes (linear regression)</li>
<li><strong><code>glmNULISAseq_predict()</code></strong>: For binary outcomes (logistic regression)</li>
</ul>
</div>
<div id="continuous-outcomes" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Continuous Outcomes<a href="predictive-modeling.html#continuous-outcomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Use <code>lmNULISAseq_predict()</code> when your outcome is continuous (e.g., disease severity score, biomarker level, clinical measurement).</p>
<div id="example-predicting-a-clinical-score" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Example: Predicting a Clinical Score<a href="predictive-modeling.html#example-predicting-a-clinical-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="predictive-modeling.html#cb109-1" tabindex="-1"></a><span class="co"># Add a continuous outcome to metadata (e.g., disease severity score)</span></span>
<span id="cb109-2"><a href="predictive-modeling.html#cb109-2" tabindex="-1"></a><span class="co"># In real analysis, this would be actual clinical data</span></span>
<span id="cb109-3"><a href="predictive-modeling.html#cb109-3" tabindex="-1"></a>metadata<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="at">n =</span> <span class="fu">nrow</span>(metadata), <span class="at">min =</span> <span class="dv">1</span>, <span class="at">max =</span> <span class="dv">100</span>)</span>
<span id="cb109-4"><a href="predictive-modeling.html#cb109-4" tabindex="-1"></a></span>
<span id="cb109-5"><a href="predictive-modeling.html#cb109-5" tabindex="-1"></a><span class="co"># Fit models with each protein predicting the score</span></span>
<span id="cb109-6"><a href="predictive-modeling.html#cb109-6" tabindex="-1"></a>lmpt <span class="ot">&lt;-</span> <span class="fu">lmNULISAseq_predict</span>(</span>
<span id="cb109-7"><a href="predictive-modeling.html#cb109-7" tabindex="-1"></a>  <span class="at">data =</span> data<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[, sample_list],</span>
<span id="cb109-8"><a href="predictive-modeling.html#cb109-8" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SampleName <span class="sc">%in%</span> sample_list),</span>
<span id="cb109-9"><a href="predictive-modeling.html#cb109-9" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;SampleName&quot;</span>,</span>
<span id="cb109-10"><a href="predictive-modeling.html#cb109-10" tabindex="-1"></a>  <span class="at">response_var =</span> <span class="st">&quot;score&quot;</span>,</span>
<span id="cb109-11"><a href="predictive-modeling.html#cb109-11" tabindex="-1"></a>  <span class="at">modelFormula =</span> <span class="st">&quot;age&quot;</span>  <span class="co"># Adjust for age as covariate</span></span>
<span id="cb109-12"><a href="predictive-modeling.html#cb109-12" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="understanding-the-model-1" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Understanding the Model<a href="predictive-modeling.html#understanding-the-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each protein, the model is:</p>
<pre><code>score ~ protein_expression + age</code></pre>
<p>This tests: <strong>“Does this protein predict the score, after accounting for age?”</strong></p>
</div>
<div id="examining-results" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Examining Results<a href="predictive-modeling.html#examining-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="predictive-modeling.html#cb111-1" tabindex="-1"></a><span class="co"># View structure</span></span>
<span id="cb111-2"><a href="predictive-modeling.html#cb111-2" tabindex="-1"></a><span class="fu">names</span>(lmpt)</span>
<span id="cb111-3"><a href="predictive-modeling.html#cb111-3" tabindex="-1"></a></span>
<span id="cb111-4"><a href="predictive-modeling.html#cb111-4" tabindex="-1"></a><span class="co"># View results table</span></span>
<span id="cb111-5"><a href="predictive-modeling.html#cb111-5" tabindex="-1"></a><span class="fu">head</span>(lmpt<span class="sc">$</span>modelStats)</span></code></pre></div>
</div>
<div id="key-columns" class="section level3 hasAnchor" number="6.2.4">
<h3><span class="header-section-number">6.2.4</span> Key Columns<a href="predictive-modeling.html#key-columns" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong><code>target</code></strong>: Protein name</li>
<li><strong><code>target_coef</code></strong>: Effect size (how much score changes per unit protein)</li>
<li><strong><code>target_pval</code></strong>: Raw p-value</li>
<li><strong><code>target_pval_FDR</code></strong>: FDR-adjusted p-value</li>
<li><strong><code>r_squared</code></strong>: Model R² (proportion of variance explained)</li>
</ul>
</div>
<div id="finding-predictive-proteins" class="section level3 hasAnchor" number="6.2.5">
<h3><span class="header-section-number">6.2.5</span> Finding Predictive Proteins<a href="predictive-modeling.html#finding-predictive-proteins" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="predictive-modeling.html#cb112-1" tabindex="-1"></a><span class="co"># Find proteins that significantly predict the score</span></span>
<span id="cb112-2"><a href="predictive-modeling.html#cb112-2" tabindex="-1"></a>predictive_proteins <span class="ot">&lt;-</span> lmpt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb112-3"><a href="predictive-modeling.html#cb112-3" tabindex="-1"></a>  <span class="fu">filter</span>(target_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb112-4"><a href="predictive-modeling.html#cb112-4" tabindex="-1"></a>  <span class="fu">arrange</span>(target_pval_FDR) <span class="sc">%&gt;%</span></span>
<span id="cb112-5"><a href="predictive-modeling.html#cb112-5" tabindex="-1"></a>  <span class="fu">select</span>(target, target_coef, target_pval_FDR, r_squared)</span>
<span id="cb112-6"><a href="predictive-modeling.html#cb112-6" tabindex="-1"></a></span>
<span id="cb112-7"><a href="predictive-modeling.html#cb112-7" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Proteins predicting score:&quot;</span>, <span class="fu">nrow</span>(predictive_proteins), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb112-8"><a href="predictive-modeling.html#cb112-8" tabindex="-1"></a><span class="fu">head</span>(predictive_proteins, <span class="dv">10</span>)</span></code></pre></div>
</div>
<div id="interpreting-coefficients" class="section level3 hasAnchor" number="6.2.6">
<h3><span class="header-section-number">6.2.6</span> Interpreting Coefficients<a href="predictive-modeling.html#interpreting-coefficients" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong>Positive coefficient</strong>: Higher protein expression → higher score</li>
<li><strong>Negative coefficient</strong>: Higher protein expression → lower score</li>
<li><strong>Magnitude</strong>: Size of effect (units of score per unit protein expression)</li>
</ul>
</div>
<div id="visualizing-predictions" class="section level3 hasAnchor" number="6.2.7">
<h3><span class="header-section-number">6.2.7</span> Visualizing Predictions<a href="predictive-modeling.html#visualizing-predictions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="predictive-modeling.html#cb113-1" tabindex="-1"></a><span class="co"># Plot top predictive protein</span></span>
<span id="cb113-2"><a href="predictive-modeling.html#cb113-2" tabindex="-1"></a>top_protein <span class="ot">&lt;-</span> predictive_proteins<span class="sc">$</span>target[<span class="dv">1</span>]</span>
<span id="cb113-3"><a href="predictive-modeling.html#cb113-3" tabindex="-1"></a></span>
<span id="cb113-4"><a href="predictive-modeling.html#cb113-4" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> data_long <span class="sc">%&gt;%</span></span>
<span id="cb113-5"><a href="predictive-modeling.html#cb113-5" tabindex="-1"></a>  <span class="fu">filter</span>(target <span class="sc">==</span> top_protein, SampleName <span class="sc">%in%</span> sample_list) <span class="sc">%&gt;%</span></span>
<span id="cb113-6"><a href="predictive-modeling.html#cb113-6" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleName, score), <span class="at">by =</span> <span class="st">&quot;SampleName&quot;</span>)</span>
<span id="cb113-7"><a href="predictive-modeling.html#cb113-7" tabindex="-1"></a></span>
<span id="cb113-8"><a href="predictive-modeling.html#cb113-8" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> NPQ, <span class="at">y =</span> score)) <span class="sc">+</span></span>
<span id="cb113-9"><a href="predictive-modeling.html#cb113-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb113-10"><a href="predictive-modeling.html#cb113-10" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">se =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb113-11"><a href="predictive-modeling.html#cb113-11" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb113-12"><a href="predictive-modeling.html#cb113-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(top_protein, <span class="st">&quot;vs Clinical Score&quot;</span>),</span>
<span id="cb113-13"><a href="predictive-modeling.html#cb113-13" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">paste</span>(top_protein, <span class="st">&quot;Expression (NPQ)&quot;</span>),</span>
<span id="cb113-14"><a href="predictive-modeling.html#cb113-14" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Clinical Score&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="binary-outcomes" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Binary Outcomes<a href="predictive-modeling.html#binary-outcomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Use <code>glmNULISAseq_predict()</code> when your outcome is binary (e.g., disease vs healthy, responder vs non-responder).</p>
<div id="example-predicting-disease-status" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Example: Predicting Disease Status<a href="predictive-modeling.html#example-predicting-disease-status" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="predictive-modeling.html#cb114-1" tabindex="-1"></a><span class="co"># Add a binary outcome to metadata</span></span>
<span id="cb114-2"><a href="predictive-modeling.html#cb114-2" tabindex="-1"></a><span class="co"># In real analysis, this would be actual outcome data</span></span>
<span id="cb114-3"><a href="predictive-modeling.html#cb114-3" tabindex="-1"></a>metadata<span class="sc">$</span>outcome <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n =</span> <span class="fu">nrow</span>(metadata), <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob =</span> <span class="fl">0.5</span>)</span>
<span id="cb114-4"><a href="predictive-modeling.html#cb114-4" tabindex="-1"></a></span>
<span id="cb114-5"><a href="predictive-modeling.html#cb114-5" tabindex="-1"></a><span class="co"># Fit logistic regression models</span></span>
<span id="cb114-6"><a href="predictive-modeling.html#cb114-6" tabindex="-1"></a>glmt <span class="ot">&lt;-</span> <span class="fu">glmNULISAseq_predict</span>(</span>
<span id="cb114-7"><a href="predictive-modeling.html#cb114-7" tabindex="-1"></a>  <span class="at">data =</span> data<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[, sample_list],</span>
<span id="cb114-8"><a href="predictive-modeling.html#cb114-8" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SampleName <span class="sc">%in%</span> sample_list),</span>
<span id="cb114-9"><a href="predictive-modeling.html#cb114-9" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;SampleName&quot;</span>,</span>
<span id="cb114-10"><a href="predictive-modeling.html#cb114-10" tabindex="-1"></a>  <span class="at">response_var =</span> <span class="st">&quot;outcome&quot;</span>,</span>
<span id="cb114-11"><a href="predictive-modeling.html#cb114-11" tabindex="-1"></a>  <span class="at">modelFormula =</span> <span class="st">&quot;disease_type + sex + age&quot;</span></span>
<span id="cb114-12"><a href="predictive-modeling.html#cb114-12" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="understanding-the-model-2" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Understanding the Model<a href="predictive-modeling.html#understanding-the-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each protein, the model is:</p>
<pre><code>log(odds of outcome) ~ protein_expression + disease_type + sex + age</code></pre>
<p>This tests: <strong>“Does this protein predict the outcome, after accounting for disease type, sex, and age?”</strong></p>
</div>
<div id="examining-results-1" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Examining Results<a href="predictive-modeling.html#examining-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="predictive-modeling.html#cb116-1" tabindex="-1"></a><span class="co"># View results table</span></span>
<span id="cb116-2"><a href="predictive-modeling.html#cb116-2" tabindex="-1"></a><span class="fu">head</span>(glmt<span class="sc">$</span>modelStats)</span></code></pre></div>
</div>
<div id="key-columns-1" class="section level3 hasAnchor" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Key Columns<a href="predictive-modeling.html#key-columns-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong><code>target</code></strong>: Protein name</li>
<li><strong><code>target_coef</code></strong>: Log odds ratio</li>
<li><strong><code>target_OR</code></strong>: Odds ratio (exponentiated coefficient)</li>
<li><strong><code>target_pval</code></strong>: Raw p-value (Wald test)</li>
<li><strong><code>target_pval_FDR</code></strong>: FDR-adjusted p-value</li>
<li><strong><code>AIC</code></strong>: Model fit statistic (lower is better)</li>
</ul>
</div>
<div id="finding-predictive-proteins-1" class="section level3 hasAnchor" number="6.3.5">
<h3><span class="header-section-number">6.3.5</span> Finding Predictive Proteins<a href="predictive-modeling.html#finding-predictive-proteins-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="predictive-modeling.html#cb117-1" tabindex="-1"></a><span class="co"># Find proteins that significantly predict the outcome</span></span>
<span id="cb117-2"><a href="predictive-modeling.html#cb117-2" tabindex="-1"></a>predictive_binary <span class="ot">&lt;-</span> glmt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb117-3"><a href="predictive-modeling.html#cb117-3" tabindex="-1"></a>  <span class="fu">filter</span>(target_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb117-4"><a href="predictive-modeling.html#cb117-4" tabindex="-1"></a>  <span class="fu">arrange</span>(target_pval_FDR) <span class="sc">%&gt;%</span></span>
<span id="cb117-5"><a href="predictive-modeling.html#cb117-5" tabindex="-1"></a>  <span class="fu">select</span>(target, target_coef, target_OR, target_pval_FDR, AIC)</span>
<span id="cb117-6"><a href="predictive-modeling.html#cb117-6" tabindex="-1"></a></span>
<span id="cb117-7"><a href="predictive-modeling.html#cb117-7" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Proteins predicting outcome:&quot;</span>, <span class="fu">nrow</span>(predictive_binary), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb117-8"><a href="predictive-modeling.html#cb117-8" tabindex="-1"></a><span class="fu">head</span>(predictive_binary, <span class="dv">10</span>)</span></code></pre></div>
</div>
<div id="interpreting-odds-ratios" class="section level3 hasAnchor" number="6.3.6">
<h3><span class="header-section-number">6.3.6</span> Interpreting Odds Ratios<a href="predictive-modeling.html#interpreting-odds-ratios" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong>OR = 1</strong>: No association</li>
<li><strong>OR &gt; 1</strong>: Higher protein expression → higher odds of outcome = 1</li>
<li><strong>OR &lt; 1</strong>: Higher protein expression → lower odds of outcome = 1</li>
<li><strong>Example</strong>: OR = 2 means doubling protein expression doubles the odds of the outcome</li>
</ul>
</div>
<div id="visualizing-binary-predictions" class="section level3 hasAnchor" number="6.3.7">
<h3><span class="header-section-number">6.3.7</span> Visualizing Binary Predictions<a href="predictive-modeling.html#visualizing-binary-predictions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="predictive-modeling.html#cb118-1" tabindex="-1"></a><span class="co"># Plot top predictive protein</span></span>
<span id="cb118-2"><a href="predictive-modeling.html#cb118-2" tabindex="-1"></a>top_protein_binary <span class="ot">&lt;-</span> predictive_binary<span class="sc">$</span>target[<span class="dv">1</span>]</span>
<span id="cb118-3"><a href="predictive-modeling.html#cb118-3" tabindex="-1"></a></span>
<span id="cb118-4"><a href="predictive-modeling.html#cb118-4" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> data_long <span class="sc">%&gt;%</span></span>
<span id="cb118-5"><a href="predictive-modeling.html#cb118-5" tabindex="-1"></a>  <span class="fu">filter</span>(target <span class="sc">==</span> top_protein_binary, SampleName <span class="sc">%in%</span> sample_list) <span class="sc">%&gt;%</span></span>
<span id="cb118-6"><a href="predictive-modeling.html#cb118-6" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleName, outcome), <span class="at">by =</span> <span class="st">&quot;SampleName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb118-7"><a href="predictive-modeling.html#cb118-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">outcome =</span> <span class="fu">factor</span>(outcome, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), </span>
<span id="cb118-8"><a href="predictive-modeling.html#cb118-8" tabindex="-1"></a>                         <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Negative&quot;</span>, <span class="st">&quot;Positive&quot;</span>)))</span>
<span id="cb118-9"><a href="predictive-modeling.html#cb118-9" tabindex="-1"></a></span>
<span id="cb118-10"><a href="predictive-modeling.html#cb118-10" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> outcome, <span class="at">y =</span> NPQ, <span class="at">fill =</span> outcome)) <span class="sc">+</span></span>
<span id="cb118-11"><a href="predictive-modeling.html#cb118-11" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb118-12"><a href="predictive-modeling.html#cb118-12" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb118-13"><a href="predictive-modeling.html#cb118-13" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb118-14"><a href="predictive-modeling.html#cb118-14" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(top_protein_binary, <span class="st">&quot;Expression by Outcome&quot;</span>),</span>
<span id="cb118-15"><a href="predictive-modeling.html#cb118-15" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Outcome&quot;</span>, <span class="at">y =</span> <span class="fu">paste</span>(top_protein_binary, <span class="st">&quot;Expression (NPQ)&quot;</span>)) <span class="sc">+</span></span>
<span id="cb118-16"><a href="predictive-modeling.html#cb118-16" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="comparing-approaches" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Comparing Approaches<a href="predictive-modeling.html#comparing-approaches" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="when-to-use-each-function" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> When to Use Each Function<a href="predictive-modeling.html#when-to-use-each-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Differential Expression (<code>lmNULISAseq</code>)</strong></p>
<ul>
<li>Goal: Find proteins that differ between groups</li>
<li>Proteins are outcomes, groups are predictors</li>
<li>Example: “Which proteins differ in cancer vs normal?”</li>
</ul>
<p><strong>Continuous Prediction (<code>lmNULISAseq_predict</code>)</strong></p>
<ul>
<li>Goal: Identify proteins that predict a continuous outcome</li>
<li>Proteins are predictors, continuous variable is outcome</li>
<li>Example: “Which proteins predict disease severity score?”</li>
</ul>
<p><strong>Binary Prediction (<code>glmNULISAseq_predict</code>)</strong></p>
<ul>
<li>Goal: Identify proteins that predict a binary outcome</li>
<li>Proteins are predictors, binary variable is outcome</li>
<li>Example: “Which proteins predict treatment response (yes/no)?”</li>
</ul>
</div>
</div>
<div id="advanced-model-comparisons" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Advanced: Model Comparisons<a href="predictive-modeling.html#advanced-model-comparisons" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="compare-model-performance" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> Compare Model Performance<a href="predictive-modeling.html#compare-model-performance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="predictive-modeling.html#cb119-1" tabindex="-1"></a><span class="co"># For continuous outcomes: Compare R² values</span></span>
<span id="cb119-2"><a href="predictive-modeling.html#cb119-2" tabindex="-1"></a>top_10_continuous <span class="ot">&lt;-</span> lmpt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb119-3"><a href="predictive-modeling.html#cb119-3" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(r_squared)) <span class="sc">%&gt;%</span></span>
<span id="cb119-4"><a href="predictive-modeling.html#cb119-4" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb119-5"><a href="predictive-modeling.html#cb119-5" tabindex="-1"></a>  <span class="fu">select</span>(target, r_squared, target_pval_FDR)</span>
<span id="cb119-6"><a href="predictive-modeling.html#cb119-6" tabindex="-1"></a></span>
<span id="cb119-7"><a href="predictive-modeling.html#cb119-7" tabindex="-1"></a><span class="fu">print</span>(top_10_continuous)</span>
<span id="cb119-8"><a href="predictive-modeling.html#cb119-8" tabindex="-1"></a></span>
<span id="cb119-9"><a href="predictive-modeling.html#cb119-9" tabindex="-1"></a><span class="co"># For binary outcomes: Compare AIC values (lower is better)</span></span>
<span id="cb119-10"><a href="predictive-modeling.html#cb119-10" tabindex="-1"></a>top_10_binary <span class="ot">&lt;-</span> glmt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb119-11"><a href="predictive-modeling.html#cb119-11" tabindex="-1"></a>  <span class="fu">arrange</span>(AIC) <span class="sc">%&gt;%</span></span>
<span id="cb119-12"><a href="predictive-modeling.html#cb119-12" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb119-13"><a href="predictive-modeling.html#cb119-13" tabindex="-1"></a>  <span class="fu">select</span>(target, AIC, target_OR, target_pval_FDR)</span>
<span id="cb119-14"><a href="predictive-modeling.html#cb119-14" tabindex="-1"></a></span>
<span id="cb119-15"><a href="predictive-modeling.html#cb119-15" tabindex="-1"></a><span class="fu">print</span>(top_10_binary)</span></code></pre></div>
</div>
<div id="multi-protein-models" class="section level3 hasAnchor" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> Multi-protein Models<a href="predictive-modeling.html#multi-protein-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For building models with multiple proteins:</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="predictive-modeling.html#cb120-1" tabindex="-1"></a><span class="co"># Select top predictive proteins</span></span>
<span id="cb120-2"><a href="predictive-modeling.html#cb120-2" tabindex="-1"></a>top_proteins <span class="ot">&lt;-</span> predictive_proteins<span class="sc">$</span>target[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb120-3"><a href="predictive-modeling.html#cb120-3" tabindex="-1"></a></span>
<span id="cb120-4"><a href="predictive-modeling.html#cb120-4" tabindex="-1"></a><span class="co"># Create wide format with selected proteins</span></span>
<span id="cb120-5"><a href="predictive-modeling.html#cb120-5" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> data_long <span class="sc">%&gt;%</span></span>
<span id="cb120-6"><a href="predictive-modeling.html#cb120-6" tabindex="-1"></a>  <span class="fu">filter</span>(target <span class="sc">%in%</span> top_proteins, SampleName <span class="sc">%in%</span> sample_list) <span class="sc">%&gt;%</span></span>
<span id="cb120-7"><a href="predictive-modeling.html#cb120-7" tabindex="-1"></a>  <span class="fu">select</span>(SampleName, target, NPQ) <span class="sc">%&gt;%</span></span>
<span id="cb120-8"><a href="predictive-modeling.html#cb120-8" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> target, <span class="at">values_from =</span> NPQ) <span class="sc">%&gt;%</span></span>
<span id="cb120-9"><a href="predictive-modeling.html#cb120-9" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleName, score, age), <span class="at">by =</span> <span class="st">&quot;SampleName&quot;</span>)</span>
<span id="cb120-10"><a href="predictive-modeling.html#cb120-10" tabindex="-1"></a></span>
<span id="cb120-11"><a href="predictive-modeling.html#cb120-11" tabindex="-1"></a><span class="co"># Fit multi-protein model</span></span>
<span id="cb120-12"><a href="predictive-modeling.html#cb120-12" tabindex="-1"></a>multi_model <span class="ot">&lt;-</span> <span class="fu">lm</span>(score <span class="sc">~</span> ., <span class="at">data =</span> model_data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>SampleName))</span>
<span id="cb120-13"><a href="predictive-modeling.html#cb120-13" tabindex="-1"></a><span class="fu">summary</span>(multi_model)</span></code></pre></div>
</div>
</div>
<div id="model-validation" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Model Validation<a href="predictive-modeling.html#model-validation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="cross-validation" class="section level3 hasAnchor" number="6.6.1">
<h3><span class="header-section-number">6.6.1</span> Cross-Validation<a href="predictive-modeling.html#cross-validation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For robust predictions, use cross-validation:</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="predictive-modeling.html#cb121-1" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb121-2"><a href="predictive-modeling.html#cb121-2" tabindex="-1"></a></span>
<span id="cb121-3"><a href="predictive-modeling.html#cb121-3" tabindex="-1"></a><span class="co"># Prepare data for top protein</span></span>
<span id="cb121-4"><a href="predictive-modeling.html#cb121-4" tabindex="-1"></a>top_protein <span class="ot">&lt;-</span> predictive_proteins<span class="sc">$</span>target[<span class="dv">1</span>]</span>
<span id="cb121-5"><a href="predictive-modeling.html#cb121-5" tabindex="-1"></a>cv_data <span class="ot">&lt;-</span> data<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ_long <span class="sc">%&gt;%</span></span>
<span id="cb121-6"><a href="predictive-modeling.html#cb121-6" tabindex="-1"></a>  <span class="fu">filter</span>(target <span class="sc">==</span> top_protein, SampleName <span class="sc">%in%</span> sample_list) <span class="sc">%&gt;%</span></span>
<span id="cb121-7"><a href="predictive-modeling.html#cb121-7" tabindex="-1"></a>  <span class="fu">select</span>(SampleName, NPQ) <span class="sc">%&gt;%</span></span>
<span id="cb121-8"><a href="predictive-modeling.html#cb121-8" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleName, score, age), <span class="at">by =</span> <span class="st">&quot;SampleName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb121-9"><a href="predictive-modeling.html#cb121-9" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb121-10"><a href="predictive-modeling.html#cb121-10" tabindex="-1"></a></span>
<span id="cb121-11"><a href="predictive-modeling.html#cb121-11" tabindex="-1"></a><span class="co"># 10-fold cross-validation</span></span>
<span id="cb121-12"><a href="predictive-modeling.html#cb121-12" tabindex="-1"></a>train_control <span class="ot">&lt;-</span> <span class="fu">trainControl</span>(<span class="at">method =</span> <span class="st">&quot;cv&quot;</span>, <span class="at">number =</span> <span class="dv">10</span>)</span>
<span id="cb121-13"><a href="predictive-modeling.html#cb121-13" tabindex="-1"></a></span>
<span id="cb121-14"><a href="predictive-modeling.html#cb121-14" tabindex="-1"></a>model_cv <span class="ot">&lt;-</span> <span class="fu">train</span>(</span>
<span id="cb121-15"><a href="predictive-modeling.html#cb121-15" tabindex="-1"></a>  score <span class="sc">~</span> NPQ <span class="sc">+</span> age,</span>
<span id="cb121-16"><a href="predictive-modeling.html#cb121-16" tabindex="-1"></a>  <span class="at">data =</span> cv_data,</span>
<span id="cb121-17"><a href="predictive-modeling.html#cb121-17" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;lm&quot;</span>,</span>
<span id="cb121-18"><a href="predictive-modeling.html#cb121-18" tabindex="-1"></a>  <span class="at">trControl =</span> train_control</span>
<span id="cb121-19"><a href="predictive-modeling.html#cb121-19" tabindex="-1"></a>)</span>
<span id="cb121-20"><a href="predictive-modeling.html#cb121-20" tabindex="-1"></a></span>
<span id="cb121-21"><a href="predictive-modeling.html#cb121-21" tabindex="-1"></a><span class="fu">print</span>(model_cv)</span></code></pre></div>
</div>
<div id="test-set-validation" class="section level3 hasAnchor" number="6.6.2">
<h3><span class="header-section-number">6.6.2</span> Test Set Validation<a href="predictive-modeling.html#test-set-validation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="predictive-modeling.html#cb122-1" tabindex="-1"></a><span class="co"># Split data into train/test</span></span>
<span id="cb122-2"><a href="predictive-modeling.html#cb122-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb122-3"><a href="predictive-modeling.html#cb122-3" tabindex="-1"></a>train_idx <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(cv_data), <span class="at">size =</span> <span class="fl">0.7</span> <span class="sc">*</span> <span class="fu">nrow</span>(cv_data))</span>
<span id="cb122-4"><a href="predictive-modeling.html#cb122-4" tabindex="-1"></a></span>
<span id="cb122-5"><a href="predictive-modeling.html#cb122-5" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> cv_data[train_idx, ]</span>
<span id="cb122-6"><a href="predictive-modeling.html#cb122-6" tabindex="-1"></a>test_data <span class="ot">&lt;-</span> cv_data[<span class="sc">-</span>train_idx, ]</span>
<span id="cb122-7"><a href="predictive-modeling.html#cb122-7" tabindex="-1"></a></span>
<span id="cb122-8"><a href="predictive-modeling.html#cb122-8" tabindex="-1"></a><span class="co"># Train model</span></span>
<span id="cb122-9"><a href="predictive-modeling.html#cb122-9" tabindex="-1"></a>train_model <span class="ot">&lt;-</span> <span class="fu">lm</span>(score <span class="sc">~</span> NPQ <span class="sc">+</span> age, <span class="at">data =</span> train_data)</span>
<span id="cb122-10"><a href="predictive-modeling.html#cb122-10" tabindex="-1"></a></span>
<span id="cb122-11"><a href="predictive-modeling.html#cb122-11" tabindex="-1"></a><span class="co"># Test predictions</span></span>
<span id="cb122-12"><a href="predictive-modeling.html#cb122-12" tabindex="-1"></a>predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(train_model, <span class="at">newdata =</span> test_data)</span>
<span id="cb122-13"><a href="predictive-modeling.html#cb122-13" tabindex="-1"></a><span class="fu">cor</span>(predictions, test_data<span class="sc">$</span>score)<span class="sc">^</span><span class="dv">2</span>  <span class="co"># R² on test set</span></span></code></pre></div>
</div>
</div>
<div id="exporting-results-1" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Exporting Results<a href="predictive-modeling.html#exporting-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="predictive-modeling.html#cb123-1" tabindex="-1"></a><span class="co"># Save continuous outcome results</span></span>
<span id="cb123-2"><a href="predictive-modeling.html#cb123-2" tabindex="-1"></a><span class="fu">write_csv</span>(lmpt<span class="sc">$</span>modelStats, <span class="st">&quot;results/continuous_prediction_results.csv&quot;</span>)</span>
<span id="cb123-3"><a href="predictive-modeling.html#cb123-3" tabindex="-1"></a><span class="fu">write_csv</span>(predictive_proteins, <span class="st">&quot;results/significant_predictors_continuous.csv&quot;</span>)</span>
<span id="cb123-4"><a href="predictive-modeling.html#cb123-4" tabindex="-1"></a></span>
<span id="cb123-5"><a href="predictive-modeling.html#cb123-5" tabindex="-1"></a><span class="co"># Save binary outcome results</span></span>
<span id="cb123-6"><a href="predictive-modeling.html#cb123-6" tabindex="-1"></a><span class="fu">write_csv</span>(glmt<span class="sc">$</span>modelStats, <span class="st">&quot;results/binary_prediction_results.csv&quot;</span>)</span>
<span id="cb123-7"><a href="predictive-modeling.html#cb123-7" tabindex="-1"></a><span class="fu">write_csv</span>(predictive_binary, <span class="st">&quot;results/significant_predictors_binary.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="complete-predictive-modeling-workflow" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> Complete Predictive Modeling Workflow<a href="predictive-modeling.html#complete-predictive-modeling-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="predictive-modeling.html#cb124-1" tabindex="-1"></a><span class="co"># 1. Continuous outcome prediction</span></span>
<span id="cb124-2"><a href="predictive-modeling.html#cb124-2" tabindex="-1"></a>lmpt <span class="ot">&lt;-</span> <span class="fu">lmNULISAseq_predict</span>(</span>
<span id="cb124-3"><a href="predictive-modeling.html#cb124-3" tabindex="-1"></a>  <span class="at">data =</span> data<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[, sample_list],</span>
<span id="cb124-4"><a href="predictive-modeling.html#cb124-4" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SampleName <span class="sc">%in%</span> sample_list),</span>
<span id="cb124-5"><a href="predictive-modeling.html#cb124-5" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;SampleName&quot;</span>,</span>
<span id="cb124-6"><a href="predictive-modeling.html#cb124-6" tabindex="-1"></a>  <span class="at">response_var =</span> <span class="st">&quot;score&quot;</span>,</span>
<span id="cb124-7"><a href="predictive-modeling.html#cb124-7" tabindex="-1"></a>  <span class="at">modelFormula =</span> <span class="st">&quot;age&quot;</span></span>
<span id="cb124-8"><a href="predictive-modeling.html#cb124-8" tabindex="-1"></a>)</span>
<span id="cb124-9"><a href="predictive-modeling.html#cb124-9" tabindex="-1"></a></span>
<span id="cb124-10"><a href="predictive-modeling.html#cb124-10" tabindex="-1"></a><span class="co"># 2. Find significant predictors</span></span>
<span id="cb124-11"><a href="predictive-modeling.html#cb124-11" tabindex="-1"></a>sig_continuous <span class="ot">&lt;-</span> lmpt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb124-12"><a href="predictive-modeling.html#cb124-12" tabindex="-1"></a>  <span class="fu">filter</span>(target_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb124-13"><a href="predictive-modeling.html#cb124-13" tabindex="-1"></a>  <span class="fu">arrange</span>(target_pval_FDR)</span>
<span id="cb124-14"><a href="predictive-modeling.html#cb124-14" tabindex="-1"></a></span>
<span id="cb124-15"><a href="predictive-modeling.html#cb124-15" tabindex="-1"></a><span class="co"># 3. Binary outcome prediction</span></span>
<span id="cb124-16"><a href="predictive-modeling.html#cb124-16" tabindex="-1"></a>glmt <span class="ot">&lt;-</span> <span class="fu">glmNULISAseq_predict</span>(</span>
<span id="cb124-17"><a href="predictive-modeling.html#cb124-17" tabindex="-1"></a>  <span class="at">data =</span> data<span class="sc">$</span>merged<span class="sc">$</span>Data_NPQ[, sample_list],</span>
<span id="cb124-18"><a href="predictive-modeling.html#cb124-18" tabindex="-1"></a>  <span class="at">sampleInfo =</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SampleName <span class="sc">%in%</span> sample_list),</span>
<span id="cb124-19"><a href="predictive-modeling.html#cb124-19" tabindex="-1"></a>  <span class="at">sampleName_var =</span> <span class="st">&quot;SampleName&quot;</span>,</span>
<span id="cb124-20"><a href="predictive-modeling.html#cb124-20" tabindex="-1"></a>  <span class="at">response_var =</span> <span class="st">&quot;outcome&quot;</span>,</span>
<span id="cb124-21"><a href="predictive-modeling.html#cb124-21" tabindex="-1"></a>  <span class="at">modelFormula =</span> <span class="st">&quot;disease_type + sex + age&quot;</span></span>
<span id="cb124-22"><a href="predictive-modeling.html#cb124-22" tabindex="-1"></a>)</span>
<span id="cb124-23"><a href="predictive-modeling.html#cb124-23" tabindex="-1"></a></span>
<span id="cb124-24"><a href="predictive-modeling.html#cb124-24" tabindex="-1"></a><span class="co"># 4. Find significant predictors</span></span>
<span id="cb124-25"><a href="predictive-modeling.html#cb124-25" tabindex="-1"></a>sig_binary <span class="ot">&lt;-</span> glmt<span class="sc">$</span>modelStats <span class="sc">%&gt;%</span></span>
<span id="cb124-26"><a href="predictive-modeling.html#cb124-26" tabindex="-1"></a>  <span class="fu">filter</span>(target_pval_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb124-27"><a href="predictive-modeling.html#cb124-27" tabindex="-1"></a>  <span class="fu">arrange</span>(target_pval_FDR)</span>
<span id="cb124-28"><a href="predictive-modeling.html#cb124-28" tabindex="-1"></a></span>
<span id="cb124-29"><a href="predictive-modeling.html#cb124-29" tabindex="-1"></a><span class="co"># 5. Export results</span></span>
<span id="cb124-30"><a href="predictive-modeling.html#cb124-30" tabindex="-1"></a><span class="fu">write_csv</span>(sig_continuous, <span class="st">&quot;results/predictors_continuous.csv&quot;</span>)</span>
<span id="cb124-31"><a href="predictive-modeling.html#cb124-31" tabindex="-1"></a><span class="fu">write_csv</span>(sig_binary, <span class="st">&quot;results/predictors_binary.csv&quot;</span>)</span>
<span id="cb124-32"><a href="predictive-modeling.html#cb124-32" tabindex="-1"></a></span>
<span id="cb124-33"><a href="predictive-modeling.html#cb124-33" tabindex="-1"></a><span class="co"># 6. Summary</span></span>
<span id="cb124-34"><a href="predictive-modeling.html#cb124-34" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Predictive Modeling Summary:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb124-35"><a href="predictive-modeling.html#cb124-35" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Continuous outcome predictors:&quot;</span>, <span class="fu">nrow</span>(sig_continuous), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb124-36"><a href="predictive-modeling.html#cb124-36" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Binary outcome predictors:&quot;</span>, <span class="fu">nrow</span>(sig_binary), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<div id="best-practices-1" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> Best Practices<a href="predictive-modeling.html#best-practices-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Model Design</strong></p>
<ul>
<li>Include relevant covariates</li>
<li>Consider non-linear relationships</li>
<li>Check for interactions if biologically relevant</li>
</ul>
<p><strong>Validation</strong></p>
<ul>
<li>Use cross-validation for small datasets</li>
<li>Reserve test set for larger datasets</li>
<li>Validate key findings independently</li>
</ul>
<p><strong>Interpretation</strong></p>
<ul>
<li>Consider both statistical significance and effect size</li>
<li>Assess model fit (R², AIC)</li>
<li>Check model assumptions</li>
</ul>
<p><strong>Reporting</strong></p>
<ul>
<li>Document model formulas</li>
<li>Report validation metrics</li>
<li>Acknowledge limitations</li>
</ul>
<p><strong>Continue to:</strong> <a href="complete-workflow-1.html#complete-workflow-1">Chapter 7: Complete Workflow: Identifying Proteomic Signatures in SLE and RA</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="longitudinal-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="complete-workflow-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": {},
    "google": false
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  },
  "toolbar": {
    "position": "fixed"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
