<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate Heatmap for NULISAseq Data with ComplexHeatmap — generate_heatmap • NULISAseqR</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate Heatmap for NULISAseq Data with ComplexHeatmap — generate_heatmap"><meta property="og:description" content="Draws a heatmap for a set of samples and targets based on sample and target metadata.
Uses ComplexHeatmap for generating plots. Colors of covariates are automatically generated
from palettes in RColorBrewer, or can be user-specified. Supports both standard and transposed
orientations, row and column annotations, and various clustering options."><meta property="og:image" content="https://nulisaseqr.alamarbio.com/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">NULISAseqR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="https://nulisaseqr.alamarbio.com/user-guide">User Guide</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Alamar-Biosciences/NULISAseqR" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate Heatmap for NULISAseq Data with ComplexHeatmap</h1>
    <small class="dont-index">Source: <a href="https://github.com/Alamar-Biosciences/NULISAseqR/blob/HEAD/R/generate_heatmap.R" class="external-link"><code>R/generate_heatmap.R</code></a></small>
    <div class="hidden name"><code>generate_heatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Draws a heatmap for a set of samples and targets based on sample and target metadata.
Uses ComplexHeatmap for generating plots. Colors of covariates are automatically generated
from palettes in RColorBrewer, or can be user-specified. Supports both standard and transposed
orientations, row and column annotations, and various clustering options.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">generate_heatmap</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">sampleInfo</span>,</span>
<span>  targetInfo <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">sampleName_var</span>,</span>
<span>  targetName_var <span class="op">=</span> <span class="st">"Target"</span>,</span>
<span>  sample_subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  target_subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_sample_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_target_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  column_split_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  row_split_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  row_fontsize <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  col_fontsize <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  column_split_num <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"Z-Score"</span>,</span>
<span>  row_title_rot <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  clustering_method_columns <span class="op">=</span> <span class="st">"ward.D2"</span>,</span>
<span>  row_split <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_column_slices <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_row_slices <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  transpose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  numeric_color_palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span>, <span class="st">"green"</span>, <span class="st">"purple"</span>, <span class="st">"orange"</span>, <span class="st">"brown"</span><span class="op">)</span>,</span>
<span>  sample_colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_width <span class="op">=</span> <span class="fl">14</span>,</span>
<span>  plot_height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>A matrix with targets in rows, samples in columns.
Row names should be the target names, and column names are the sample names.
It is assumed that data has already been transformed using <code>log2(x + 1)</code>
for each NULISAseq normalized count value <code>x</code>, i.e. NPQ.</p></dd>


<dt id="arg-sampleinfo">sampleInfo<a class="anchor" aria-label="anchor" href="#arg-sampleinfo"></a></dt>
<dd><p>A data frame with sample metadata. Rows are samples,
columns are sample metadata variables. Heatmap will only show the samples in
<code>sample_subset</code> from <code>sampleInfo</code>.</p></dd>


<dt id="arg-targetinfo">targetInfo<a class="anchor" aria-label="anchor" href="#arg-targetinfo"></a></dt>
<dd><p>A data frame with target metadata. Rows are targets,
columns are target metadata variables. Required if <code>annotate_target_by</code>
or <code>row_split_by</code> are specified; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-samplename-var">sampleName_var<a class="anchor" aria-label="anchor" href="#arg-samplename-var"></a></dt>
<dd><p>Character string specifying the name of the column in <code>sampleInfo</code>
that matches the column names of <code>data</code>. This variable will be used for subsetting
samples from <code>sampleInfo</code>.</p></dd>


<dt id="arg-targetname-var">targetName_var<a class="anchor" aria-label="anchor" href="#arg-targetname-var"></a></dt>
<dd><p>Character string specifying the name of the column in <code>targetInfo</code>
that matches the row names of <code>data</code>; defaults to <code>"Target"</code>.</p></dd>


<dt id="arg-sample-subset">sample_subset<a class="anchor" aria-label="anchor" href="#arg-sample-subset"></a></dt>
<dd><p>Vector of sample names for selected samples to show in the heatmap,
should match the existing column names of <code>data</code>; defaults to <code>NULL</code> (all samples).</p></dd>


<dt id="arg-target-subset">target_subset<a class="anchor" aria-label="anchor" href="#arg-target-subset"></a></dt>
<dd><p>Vector of target names for selected targets to show in the heatmap,
should match the existing row names of <code>data</code>; defaults to <code>NULL</code> (all targets).</p></dd>


<dt id="arg-annotate-sample-by">annotate_sample_by<a class="anchor" aria-label="anchor" href="#arg-annotate-sample-by"></a></dt>
<dd><p>Character vector of column names from <code>sampleInfo</code> that will
be used for sample annotations (shown as colored bars); defaults to <code>NULL</code>.</p></dd>


<dt id="arg-annotate-target-by">annotate_target_by<a class="anchor" aria-label="anchor" href="#arg-annotate-target-by"></a></dt>
<dd><p>Character vector of column names from <code>targetInfo</code> that will
be used for target annotations (shown as colored bars on the left or top depending on orientation);
defaults to <code>NULL</code>.</p></dd>


<dt id="arg-column-split-by">column_split_by<a class="anchor" aria-label="anchor" href="#arg-column-split-by"></a></dt>
<dd><p>Character string specifying the name of the column from <code>sampleInfo</code>
that will be used for supervised clustering of columns (samples). This creates separate column
slices in the heatmap; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-row-split-by">row_split_by<a class="anchor" aria-label="anchor" href="#arg-row-split-by"></a></dt>
<dd><p>Character string specifying the name of the column from <code>targetInfo</code>
that will be used for supervised clustering of rows (targets). This creates separate row
slices in the heatmap; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-row-fontsize">row_fontsize<a class="anchor" aria-label="anchor" href="#arg-row-fontsize"></a></dt>
<dd><p>Numeric value for the text size of the row labels in heatmap;
defaults to 4.</p></dd>


<dt id="arg-col-fontsize">col_fontsize<a class="anchor" aria-label="anchor" href="#arg-col-fontsize"></a></dt>
<dd><p>Numeric value for the text size of the column labels in heatmap;
defaults to 4.</p></dd>


<dt id="arg-column-split-num">column_split_num<a class="anchor" aria-label="anchor" href="#arg-column-split-num"></a></dt>
<dd><p>Integer specifying the number of slices that the columns are split into
via unsupervised clustering; defaults to <code>NULL</code>. Ignored if <code>column_split_by</code> is specified.</p></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p>Character string used as the title of the heatmap legend;
defaults to <code>"Z-Score"</code>.</p></dd>


<dt id="arg-row-title-rot">row_title_rot<a class="anchor" aria-label="anchor" href="#arg-row-title-rot"></a></dt>
<dd><p>Numeric value for rotation of row titles in degrees.
Only 0, 90, 270 are allowed; defaults to 0.</p></dd>


<dt id="arg-clustering-method-columns">clustering_method_columns<a class="anchor" aria-label="anchor" href="#arg-clustering-method-columns"></a></dt>
<dd><p>Character string specifying the method to perform
hierarchical clustering, passed to <code>hclust</code>; defaults to <code>"ward.D2"</code>.</p></dd>


<dt id="arg-row-split">row_split<a class="anchor" aria-label="anchor" href="#arg-row-split"></a></dt>
<dd><p>Integer specifying the number of slices that the rows are split into
via unsupervised clustering; defaults to <code>NULL</code>. Ignored if <code>row_split_by</code> is specified.
When <code>NULL</code> and <code>targetInfo</code> is not provided, no row splitting is performed.</p></dd>


<dt id="arg-cluster-rows">cluster_rows<a class="anchor" aria-label="anchor" href="#arg-cluster-rows"></a></dt>
<dd><p>Logical indicating whether to cluster rows (targets). If <code>NULL</code> (default),
clustering is enabled when <code>targetInfo</code> is provided, and disabled when <code>targetInfo</code> is <code>NULL</code>.</p></dd>


<dt id="arg-cluster-column-slices">cluster_column_slices<a class="anchor" aria-label="anchor" href="#arg-cluster-column-slices"></a></dt>
<dd><p>Logical indicating whether to perform clustering on column slices
if columns are split; defaults to <code>FALSE</code>.</p></dd>


<dt id="arg-cluster-row-slices">cluster_row_slices<a class="anchor" aria-label="anchor" href="#arg-cluster-row-slices"></a></dt>
<dd><p>Logical indicating whether to perform clustering on row slices
if rows are split; defaults to <code>TRUE</code>.</p></dd>


<dt id="arg-transpose">transpose<a class="anchor" aria-label="anchor" href="#arg-transpose"></a></dt>
<dd><p>Logical indicating whether to transpose the heatmap (samples in rows,
targets in columns); defaults to <code>FALSE</code>.</p></dd>


<dt id="arg-numeric-color-palette">numeric_color_palette<a class="anchor" aria-label="anchor" href="#arg-numeric-color-palette"></a></dt>
<dd><p>Character vector of colors to use for numeric annotations.
Each color will be used with a white-to-color gradient for continuous variables;
defaults to <code>c("blue", "red", "green", "purple", "orange", "brown")</code>.</p></dd>


<dt id="arg-sample-colors">sample_colors<a class="anchor" aria-label="anchor" href="#arg-sample-colors"></a></dt>
<dd><p>Named list of custom colors for categorical sample annotations.
List names should match column names in <code>annotate_sample_by</code>. Each element should be
a named vector where names are category levels and values are color codes; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-output-dir">output_dir<a class="anchor" aria-label="anchor" href="#arg-output-dir"></a></dt>
<dd><p>Character string specifying the directory path to save the plot.
If <code>NULL</code>, the plot is not saved; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-plot-name">plot_name<a class="anchor" aria-label="anchor" href="#arg-plot-name"></a></dt>
<dd><p>Character string specifying the filename for the saved plot,
including file extension (.pdf, .png, .jpg, or .svg). Required if <code>output_dir</code> is specified;
defaults to <code>NULL</code>.</p></dd>


<dt id="arg-plot-title">plot_title<a class="anchor" aria-label="anchor" href="#arg-plot-title"></a></dt>
<dd><p>Character string for the title of the heatmap; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-plot-width">plot_width<a class="anchor" aria-label="anchor" href="#arg-plot-width"></a></dt>
<dd><p>Numeric value for the width of the saved plot in inches; defaults to 14.</p></dd>


<dt id="arg-plot-height">plot_height<a class="anchor" aria-label="anchor" href="#arg-plot-height"></a></dt>
<dd><p>Numeric value for the height of the saved plot in inches; defaults to 7.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">ComplexHeatmap::Heatmap</a></code> function.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing:</p><dl><dt>targets_used</dt>
<dd><p>Character vector of target names used in the heatmap after filtering.</p></dd>

  <dt>heatmap</dt>
<dd><p>The ComplexHeatmap object.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The function performs the following steps:</p><ol><li><p>Filters data to specified samples and targets</p></li>
<li><p>Removes targets with all zero values</p></li>
<li><p>Scales data by row (Z-score transformation)</p></li>
<li><p>Removes rows with NA, NaN, or Inf values after scaling</p></li>
<li><p>Optionally transposes the matrix</p></li>
<li><p>Generates or uses custom colors for annotations</p></li>
<li><p>Creates the heatmap with specified annotations and clustering</p></li>
<li><p>Optionally saves to file</p></li>
</ol></div>
    <div id="custom-colors">
    <h2>Custom Colors</h2>


<p>To specify custom colors for categorical sample annotations, use the <code>sample_colors</code> parameter:</p><div class="sourceCode"><pre><code><span></span>
<span><span class="va">my_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  Group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span> <span class="op">=</span> <span class="st">"#FF0000"</span>, <span class="st">"Treatment"</span> <span class="op">=</span> <span class="st">"#0000FF"</span><span class="op">)</span>,</span>
<span>  Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch1"</span> <span class="op">=</span> <span class="st">"#FFA500"</span>, <span class="st">"Batch2"</span> <span class="op">=</span> <span class="st">"#800080"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>

    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Basic heatmap with sample annotations (no row clustering by default)</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_heatmap</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Heatmap with row clustering enabled</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_heatmap</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  row_split <span class="op">=</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Heatmap with target annotations and custom colors</span></span></span>
<span class="r-in"><span><span class="va">custom_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  Group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span> <span class="op">=</span> <span class="st">"blue"</span>, <span class="st">"Treatment"</span> <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_heatmap</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  targetInfo <span class="op">=</span> <span class="va">target_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  targetName_var <span class="op">=</span> <span class="st">"Target"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  annotate_target_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pathway"</span>, <span class="st">"Domain"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  sample_colors <span class="op">=</span> <span class="va">custom_colors</span>,</span></span>
<span class="r-in"><span>  row_split_by <span class="op">=</span> <span class="st">"Pathway"</span>,</span></span>
<span class="r-in"><span>  column_split_by <span class="op">=</span> <span class="st">"Group"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Save heatmap to file (supports PDF, PNG, JPG, SVG)</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_heatmap</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">expr_matrix</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"Sample_ID"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  output_dir <span class="op">=</span> <span class="st">"output/figures"</span>,</span></span>
<span class="r-in"><span>  plot_name <span class="op">=</span> <span class="st">"expression_heatmap.svg"</span>,</span></span>
<span class="r-in"><span>  plot_title <span class="op">=</span> <span class="st">"Gene Expression Heatmap"</span>,</span></span>
<span class="r-in"><span>  plot_width <span class="op">=</span> <span class="fl">12</span>,</span></span>
<span class="r-in"><span>  plot_height <span class="op">=</span> <span class="fl">8</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alamar Biosciences Bioinformatics Team.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

