<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate PCA Biplot for NULISAseq Data with PCAtools — generate_pca • NULISAseqR</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate PCA Biplot for NULISAseq Data with PCAtools — generate_pca"><meta property="og:description" content="Performs Principal Component Analysis (PCA) and generates a biplot for visualizing
sample relationships based on gene expression data. Uses PCAtools for analysis
and ggplot2 for visualization. Colors are automatically generated from RColorBrewer
palettes."><meta property="og:image" content="https://nulisaseqr.alamarbio.com/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">NULISAseqR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.4.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="https://nulisaseqr.alamarbio.com/user-guide">User Guide</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Alamar-Biosciences/NULISAseqR" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate PCA Biplot for NULISAseq Data with PCAtools</h1>
    <small class="dont-index">Source: <a href="https://github.com/Alamar-Biosciences/NULISAseqR/blob/HEAD/R/generate_pca.R" class="external-link"><code>R/generate_pca.R</code></a></small>
    <div class="hidden name"><code>generate_pca.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Performs Principal Component Analysis (PCA) and generates a biplot for visualizing
sample relationships based on gene expression data. Uses PCAtools for analysis
and ggplot2 for visualization. Colors are automatically generated from RColorBrewer
palettes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">generate_pca</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">sampleInfo</span>,</span>
<span>  <span class="va">sampleName_var</span>,</span>
<span>  annotate_sample_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_points <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sample_subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  target_subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shape_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ellipse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ellipseType <span class="op">=</span> <span class="st">"t"</span>,</span>
<span>  ellipseAlpha <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>  ellipseFill <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ellipseLineSize <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  sample_colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  components <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_width <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  plot_height <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>A matrix with targets in rows, samples in columns.
Row names should be the target names, and column names are the sample names.
It is assumed that data has already been transformed using <code>log2(x + 1)</code>
for each NULISAseq normalized count value <code>x</code>, i.e. NPQ.</p></dd>


<dt id="arg-sampleinfo">sampleInfo<a class="anchor" aria-label="anchor" href="#arg-sampleinfo"></a></dt>
<dd><p>A data frame with sample metadata. Rows are samples,
columns are sample metadata variables.</p></dd>


<dt id="arg-samplename-var">sampleName_var<a class="anchor" aria-label="anchor" href="#arg-samplename-var"></a></dt>
<dd><p>Character string specifying the name of the column in <code>sampleInfo</code>
that matches the column names of <code>data</code>.</p></dd>


<dt id="arg-annotate-sample-by">annotate_sample_by<a class="anchor" aria-label="anchor" href="#arg-annotate-sample-by"></a></dt>
<dd><p>Character string specifying the column name from <code>sampleInfo</code>
to use for coloring points. Only one variable is allowed; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-label-points">label_points<a class="anchor" aria-label="anchor" href="#arg-label-points"></a></dt>
<dd><p>Logical indicating whether to add sample labels to the plot;
defaults to <code>FALSE</code>.</p></dd>


<dt id="arg-sample-subset">sample_subset<a class="anchor" aria-label="anchor" href="#arg-sample-subset"></a></dt>
<dd><p>Vector of sample names for selected samples to include in PCA,
should match the existing column names of <code>data</code>; defaults to <code>NULL</code> (all samples).</p></dd>


<dt id="arg-target-subset">target_subset<a class="anchor" aria-label="anchor" href="#arg-target-subset"></a></dt>
<dd><p>Vector of target names for selected targets to include in PCA,
should match the existing row names of <code>data</code>; defaults to <code>NULL</code> (all targets).</p></dd>


<dt id="arg-shape-by">shape_by<a class="anchor" aria-label="anchor" href="#arg-shape-by"></a></dt>
<dd><p>Character string specifying the column name from <code>sampleInfo</code>
to use for point shapes; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-ellipse">ellipse<a class="anchor" aria-label="anchor" href="#arg-ellipse"></a></dt>
<dd><p>Logical indicating whether to draw ellipses around groups;
defaults to <code>TRUE</code>.</p></dd>


<dt id="arg-ellipsetype">ellipseType<a class="anchor" aria-label="anchor" href="#arg-ellipsetype"></a></dt>
<dd><p>Character string specifying the type of ellipse. Options include
<code>"t"</code> for t-distribution and <code>"norm"</code> for normal distribution;
defaults to <code>"t"</code>.</p></dd>


<dt id="arg-ellipsealpha">ellipseAlpha<a class="anchor" aria-label="anchor" href="#arg-ellipsealpha"></a></dt>
<dd><p>Numeric value between 0 and 1 for ellipse transparency;
defaults to <code>0.15</code>.</p></dd>


<dt id="arg-ellipsefill">ellipseFill<a class="anchor" aria-label="anchor" href="#arg-ellipsefill"></a></dt>
<dd><p>Logical indicating whether to fill the ellipses;
defaults to <code>TRUE</code>.</p></dd>


<dt id="arg-ellipselinesize">ellipseLineSize<a class="anchor" aria-label="anchor" href="#arg-ellipselinesize"></a></dt>
<dd><p>Numeric value for the ellipse border line width;
defaults to <code>0</code> (no border).</p></dd>


<dt id="arg-sample-colors">sample_colors<a class="anchor" aria-label="anchor" href="#arg-sample-colors"></a></dt>
<dd><p>Named vector of custom colors for sample groups.
Names should match the levels in <code>annotate_sample_by</code>; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-components">components<a class="anchor" aria-label="anchor" href="#arg-components"></a></dt>
<dd><p>Integer vector of length 2 specifying which principal components to plot.
For example, <code>c(1, 2)</code> plots PC1 vs PC2, <code>c(2, 3)</code> plots PC2 vs PC3;
defaults to <code>c(1, 2)</code>.</p></dd>


<dt id="arg-output-dir">output_dir<a class="anchor" aria-label="anchor" href="#arg-output-dir"></a></dt>
<dd><p>Character string specifying the directory path to save the plot.
If <code>NULL</code>, the plot is not saved; defaults to <code>NULL</code>. If provided without
<code>plot_name</code>, a default filename with timestamp will be generated.</p></dd>


<dt id="arg-plot-name">plot_name<a class="anchor" aria-label="anchor" href="#arg-plot-name"></a></dt>
<dd><p>Character string specifying the filename for the saved plot,
including file extension (.pdf, .png, .jpg, or .svg). If <code>NULL</code> and <code>output_dir</code>
is provided, a default filename with timestamp will be used; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-plot-title">plot_title<a class="anchor" aria-label="anchor" href="#arg-plot-title"></a></dt>
<dd><p>Character string for the title of the PCA plot; defaults to <code>NULL</code>.</p></dd>


<dt id="arg-plot-width">plot_width<a class="anchor" aria-label="anchor" href="#arg-plot-width"></a></dt>
<dd><p>Numeric value for the width of the saved plot in inches; defaults to 10.</p></dd>


<dt id="arg-plot-height">plot_height<a class="anchor" aria-label="anchor" href="#arg-plot-height"></a></dt>
<dd><p>Numeric value for the height of the saved plot in inches; defaults to 8.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to <code><a href="https://rdrr.io/pkg/PCAtools/man/biplot.html" class="external-link">PCAtools::biplot</a></code> function.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing:</p><dl><dt>targets_used</dt>
<dd><p>Character vector of target names used in the PCA after filtering.</p></dd>

  <dt>pca_results</dt>
<dd><p>The PCAtools PCA object containing all PCA results.</p></dd>

  <dt>rotated</dt>
<dd><p>Data frame containing the PC scores (PC1, PC2, etc.) for each sample.</p></dd>

  <dt>plot</dt>
<dd><p>The ggplot2 object of the PCA biplot.</p></dd>

  <dt>output_path</dt>
<dd><p>Character string of the full path to the saved file, or <code>NULL</code> if not saved.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The function performs the following steps:</p><ol><li><p>Filters data to specified samples and targets</p></li>
<li><p>Removes targets with all zero values</p></li>
<li><p>Scales data by row (Z-score transformation)</p></li>
<li><p>Removes rows with NA, NaN, or Inf values after scaling</p></li>
<li><p>Performs PCA using PCAtools</p></li>
<li><p>Generates biplot with specified aesthetics</p></li>
<li><p>Optionally saves to file</p></li>
</ol></div>
    <div id="custom-colors">
    <h2>Custom Colors</h2>


<p>To specify custom colors for sample groups, use the <code>sample_colors</code> parameter:</p><div class="sourceCode"><pre><code><span></span>
<span><span class="va">my_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span> <span class="op">=</span> <span class="st">"#FF0000"</span>, <span class="st">"Treatment"</span> <span class="op">=</span> <span class="st">"#0000FF"</span><span class="op">)</span></span></code></pre></div>

    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Basic PCA plot</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_pca</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="st">"Group"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># PCA with sample labels and custom shapes</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_pca</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="st">"Group"</span>,</span></span>
<span class="r-in"><span>  shape_by <span class="op">=</span> <span class="st">"Batch"</span>,</span></span>
<span class="r-in"><span>  label_points <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># PCA with custom colors</span></span></span>
<span class="r-in"><span><span class="va">custom_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span> <span class="op">=</span> <span class="st">"blue"</span>, <span class="st">"Treatment"</span> <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_pca</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="st">"Group"</span>,</span></span>
<span class="r-in"><span>  sample_colors <span class="op">=</span> <span class="va">custom_colors</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Save PCA plot to file</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">generate_pca</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">Data_NPQ</span>,</span></span>
<span class="r-in"><span>  sampleInfo <span class="op">=</span> <span class="va">sample_metadata</span>,</span></span>
<span class="r-in"><span>  sampleName_var <span class="op">=</span> <span class="st">"SampleName"</span>,</span></span>
<span class="r-in"><span>  annotate_sample_by <span class="op">=</span> <span class="st">"Group"</span>,</span></span>
<span class="r-in"><span>  output_dir <span class="op">=</span> <span class="st">"output/figures"</span>,</span></span>
<span class="r-in"><span>  plot_name <span class="op">=</span> <span class="st">"pca_analysis.pdf"</span>,</span></span>
<span class="r-in"><span>  plot_title <span class="op">=</span> <span class="st">"PCA Analysis of Expression Data"</span>,</span></span>
<span class="r-in"><span>  plot_width <span class="op">=</span> <span class="fl">12</span>,</span></span>
<span class="r-in"><span>  plot_height <span class="op">=</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alamar Biosciences Bioinformatics Team.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

